---
title: "TE Visualisations"
author: "Chung-Wing Ko"
date: "28 March 2025"
output: 
  html_document:
    toc: true
    toc_depth: 4
    toc_float: true
    code_folding: show
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, message = FALSE, warning = FALSE)
```

## Set up

### load packages/colors

```{r}
rm (list = ls())
par(ask = FALSE)
```


```{r}
#install
#devtools::install_github("KarstensLab/microshades")
#devtools::install_github("mikemc/speedyseq")
#BiocManager::install("microbiome")

#pkgs <- c("phyloseq", "ALDEx2", "SummarizedExperiment", "Biobase", "devtools", 
     #     "ComplexHeatmap", "BiocGenerics", "BiocManager", "metagenomeSeq", 
     #     "Maaslin2", "edgeR", "lefser", "limma", "KEGGREST", "DESeq2")
#for (pkg in pkgs) {
 # if (!requireNamespace(pkg, quietly = TRUE))
 #   BiocManager::install(pkg)
#}
#devtools::install_github("cafferychen777/ggpicrust2")

#load libraries
library(tidyr)
library(dplyr)
library(tidyverse)
library(readr)
library(ggplot2)
library(ggrepel)
library(ggpubr)
library(microshades) #compositional plots
library(speedyseq)
library(vegan) #NMDS
library(BiodiversityR)
library(tibble)
library(metacoder) #taxonomy sorting
library(microbiome)
library(cowplot)
library(lme4) #LMMs
library(lmerTest) #LMMs
library(emmeans) #emmeans
library(agricolae) #tukey test
library(performance) #checking assumptions
library(see)
library(treemapify) #treemap
library(iNEXT) #rarefaction curves
library(forcats)
library(reshape) #melt function

#for ggpicrust2
library(ggpicrust2)
library(tibble)
library(ggprism)
library(patchwork)
library(ggh4x) #metacyc
library(stats)
library(aplot)
library(ggplotify)

#colour palettes
library(RColorBrewer)
library(MetBrewer)
library(paletteer)

setwd("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots")
path <- setwd("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots")

#change colours
taxa_colors <- c("#4E79A7FF", "#B07AA1FF","#F28E2BFF", "#499894FF", "#FABFD2FF", "#59A14FFF", "#F1CE63FF", "#8CD17DFF", "#E15759FF", "#86BCB6FF",  "#79706EFF", "#F8E898FF", "#9D7660FF", "#FF9D9AFF", "#D7B5A6FF", "#D4A6C8FF", "#FAB255", "#B6992DFF", "#D37295FF", "#206078FF", "#D8D8E8FF", "#C4E4B0", "#DD5129", "#805840FF", "#E0A858FF","#505058FF", "#43B284", "#4c2f61", "#90D0E0FF", "#884058FF", "#BAB0ACFF", "#E0A858FF", "#D8D8E8FF", "#3898B0FF", "#000000FF", "#F0F0F8FF", "#505058FF", "#805840FF", "#70B8D0FF", "#B88838FF", "#90D0E0FF", "#884058FF", "#9898A8FF")

plot_colors <- c("#74c8c3","#5a97c1","#0a2e57", "#d8d97a", "#669d62")
depth_colors <- c("#e7e5cc", "#c2d6a4", "#669d62", "#192813")
rainbow_colors <- c("#DD5129", "#FAB255", "#0F7BA2", "#43B284", "#4c2f61")

#nice places to find colours: https://emilhvitfeldt.github.io/r-color-palettes/
#colourblind checker: https://davidmathlogic.com/colorblind/#%23D81B60-%231E88E5-%23FFC107-%23004D40
```

### load environmental data

```{r}
#import soil data
env_data <- read.csv("total_env_data.csv") 

#convert to factors/levels
env_data$Transect <- as.factor(env_data$Transect)

env_data$Depth <- as.factor(env_data$Depth)
levels(env_data$Depth) <- c("0-10","10-50", "50-100", "100-150")
levels(env_data$Depth)

env_data$Plot <- as.factor(env_data$Plot)
levels(env_data$Plot) <- c("30", "75", "250", "900", "1900")
levels(env_data$Plot)

env_data <- env_data %>% dplyr::rename(SoilMoisture = soil_moisture, N = N_percent, 
                                       C = C_percent, H = H_percent, P = P_ww_mgkg, 
                                       Al = Al_ww_mg.kg, Ca = Ca_avg_ww_mg.kg, Fe = Fe_avg_ww_mg.kg,
                                       Mg = Mg_avg_ww_mg.kg, Mn = Mn_avg_ww_mg.kg, K = K_ww_mg.kg, 
                                       NH4 = ammonium, NO3 = nitrate)
summary(env_data)
env_data <- env_data %>% dplyr::select(-H, -X, -ecmcol, -PERr, -Phenoxr, -PERb)

#separate into datasets
bact_env_data <- env_data[,-c(2:3)] %>% dplyr::rename(Sample = X16S_filename)
arc.remove.list <- paste(c("AAC", "AAD", "ABC", "ABD", "ACB", "ACD", "BAD", "BCB", "CAD", "CBC", "CBD", "CCC",
            "CDD", "CEC", "CED", "DAD", "DDC", "DDD", "DED", "ECA", "ECB", "ECC", "ECD", "EDD", "EEC"), collapse = '|')
arc_env_data <- env_data[,-c(1,3)] %>% dplyr::rename(Sample = Archaea_filename) %>% filter(!str_detect(Sample, arc.remove.list))
fun_env_data <- env_data[,-c(1,2)] %>% dplyr::rename(Sample = ITS_filename) %>% filter(!str_detect(Sample, "BCA"))
```

### load tax/abund data

```{r}
#bacteria
bact_phyloseq_rare <- readRDS("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/physeq_rarefied_16S")
bact_phyloseq_filt <- readRDS("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/physeq_pruned_16S")

#convert phyloseq object to dataframe + add env_data
bact_phyloseq_rare.df <- psmelt(bact_phyloseq_rare)
bact_phyloseq_rare.df <- bact_phyloseq_rare.df %>% subset(Kingdom == "Bacteria") #filter only bacteria
bact_phyloseq_filt.df <- psmelt(bact_phyloseq_filt)
bact_phyloseq_filt.df <- bact_phyloseq_filt.df %>% subset(Kingdom == "Bacteria") #filter only bacteria
length(unique(bact_phyloseq_rare.df$OTU))

#archaea
arc_phyloseq_rare <- readRDS("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/Arch_physeq_rare.RDS")
arc_phyloseq_filt <- readRDS("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/Arch_physeq_notrare.RDS")

#convert phyloseq object to dataframe + add env_data
arc_phyloseq_rare.df <- psmelt(arc_phyloseq_rare)
arc_phyloseq_rare.df <- arc_phyloseq_rare.df %>% dplyr::select(-Transect, -Plot, -Depth) %>% dplyr::right_join(arc_env_data)
arc_phyloseq_rare.df <- arc_phyloseq_rare.df %>% subset(Kingdom == "Archaea") #filter only archaea
arc_phyloseq_filt.df <- psmelt(arc_phyloseq_filt)
arc_phyloseq_filt.df <- arc_phyloseq_filt.df %>% dplyr::select(-Transect, -Plot, -Depth) %>% dplyr::right_join(arc_env_data)
length(unique(arc_phyloseq_rare.df$OTU))

#fungi
fun_phyloseq_rare <- readRDS("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_ITS/Post_tax/physeq_rarefied_ITS.Rds")
fun_phyloseq_filt <- readRDS("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_ITS/Post_tax/ITS_phyloseq_filtered.Rds")

#convert phyloseq object to dataframe
fun_phyloseq_rare.df <- psmelt(fun_phyloseq_rare)
fun_phyloseq_filt.df <- psmelt(fun_phyloseq_filt)
length(unique(fun_phyloseq_rare.df$OTU))
```

## Bacteria

### PERMANOVA

```{r}
#log transform
bact_physeq.t <- microbiome::transform(bact_phyloseq_rare,transform="log10")
#calculating bray curtis distances
bact_bray.dist<-phyloseq::distance(bact_physeq.t,"bray") 
bact_samp.data<-sample_data(bact_phyloseq_rare)

#modeling
bact_physeq.model.plot <-adonis2(bact_bray.dist~bact_samp.data$Plot); bact_physeq.model.plot
bact_physeq.model.depth <-adonis2(bact_bray.dist~bact_samp.data$Depth); bact_physeq.model.depth
bact_physeq.model.trans <-adonis2(bact_bray.dist~bact_samp.data$Transect); bact_physeq.model.trans

#check for homogeneity
dis <- vegdist(bact_bray.dist)
#plot
dispersion_model <- betadisper(dis, bact_samp.data$Plot)
permutest(dispersion_model, pairwise = TRUE)
plot(dispersion_model)
#transect
dispersion_model <- betadisper(dis, bact_samp.data$Transect)
permutest(dispersion_model, pairwise = TRUE)
plot(dispersion_model)
#depth
dispersion_model <- betadisper(dis, bact_samp.data$Depth)
permutest(dispersion_model, pairwise = TRUE)
plot(dispersion_model)
```

### NMDS w/ envfit

```{r}
bact_community <- bact_phyloseq_rare.df %>% select(c(1:3),) %>% pivot_wider(names_from = "OTU", values_from = "Abundance") %>% column_to_rownames("Sample")
bact_community.matrix <- as.matrix(bact_community)

#nmds code
set.seed(123)
bact_nmds = metaMDS(bact_community.matrix, distance = "bray"); bact_nmds

summary(bact_env_data)

bact_en = envfit(bact_nmds, bact_env_data, permutations = 999, na.rm = TRUE)

plot(bact_nmds)
plot(bact_en)

#extract NMDS scores (x and y coordinates) for sites from newer versions of vegan package
bact_data.scores = as.data.frame(scores(bact_nmds)$sites)
bact_data.scores <- tibble::rownames_to_column(bact_data.scores, "Sample")
bact_data.scores <- bact_data.scores %>% dplyr::inner_join(bact_env_data) %>% select(c(1:6),) %>% column_to_rownames("Sample")

bact_en_coord_cont = as.data.frame(scores(bact_en, "vectors")) * ordiArrowMul(bact_en)
bact_en_coord_cat = as.data.frame(scores(bact_en, "factors")) * ordiArrowMul(bact_en)

nmds_colors <- c("#e7e5cc","#74c8c3","#0a2e57", "#d8d97a", "#669d62", "#FAB255", "#74c8c3", "#DD5129", "#4c2f61")

bact_nmds_plot <- ggplot(data = bact_data.scores, aes(x = NMDS1, y = NMDS2)) + 
  stat_ellipse(aes(fill = Plot), geom = "polygon", alpha = 0.25, show.legend = FALSE) +
  geom_point(data = bact_data.scores, aes(fill = Plot, shape = Depth), size = 5, alpha = 1, show.legend = TRUE) + 
  scale_shape_manual(values = c(22, 21, 24, 25)) +
  scale_fill_manual(values = plot_colors)  + 
  scale_color_manual(values=plot_colors)  +
    # geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2), 
      # data = bact_en_coord_cont, alpha = 0.5) +
   #  geom_text_repel(data = bact_en_coord_cont, aes(x = NMDS1, y = NMDS2), 
      # fontface = "bold", label = row.names(bact_en_coord_cont), size = 8) + 
     theme(axis.title = element_text(size = 20), 
       panel.background = element_blank(), panel.border = element_rect(fill = NA), 
       axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), 
       legend.title = element_text(size = 25), 
       legend.text = element_text(size =20)) + 
  #guides(fill = guide_legend(title = "Group"), shape = guide_legend(title = "Group")) +
    labs(fill = "Plot") #+ theme(legend.position = "none")
bact_nmds_plot

bact_nmds_plot <- bact_nmds_plot + #theme(plot.background = element_rect(fill = "#C4E4B0")) +
  theme(axis.text = element_text(size=15), 
        axis.title=element_text(size=25)) +
  theme(panel.background = element_rect(fill = 'white', color = 'black'),
        panel.grid.major = element_line(size = 0.07, linetype = 'solid', colour = "black"))

tiff("Figures/Plot2D.tiff", width = 7, height = 4, units = 'in', res = 300)
bact_nmds_plot
dev.off()
```

### Richness

```{r}
bact_DIV <- estimate_richness(bact_phyloseq_rare); bact_DIV
summary(bact_DIV)

#join env data
bact_DIV <- rownames_to_column(bact_DIV) %>% dplyr::rename(Sample = rowname)

bact_plot <- bact_phyloseq_rare.df %>% select("Sample", "Transect", "Plot", "Depth") %>% unique()
rownames(bact_plot) <- NULL
bact_DIV <- bact_DIV %>% dplyr::right_join(bact_plot) %>% dplyr::rename(Bacteria_Observed = Observed)
levels(bact_DIV$Depth)

#anova
bact_obvs_anova <- lmer((Bacteria_Observed) ~ Plot+Depth + (1|Transect), data = bact_DIV)
check_model(bact_obvs_anova)
anova(bact_obvs_anova, type = 2)
emmeans(bact_obvs_anova, list(pairwise ~Plot), adjust = "tukey")
bact_obvs_emmeans <- emmeans(bact_obvs_anova, ~Plot*Depth)
bact_obvs_emmeans <- as.data.frame(bact_obvs_emmeans)
bact_obvs_emmeans_plot <- emmeans(bact_obvs_anova, ~Plot)
bact_obvs_emmeans_plot <- as.data.frame(bact_obvs_emmeans_plot)

# PLOT x DEPTH
bact_obvs_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = bact_obvs_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = bact_obvs_emmeans) +
  geom_jitter(aes(x = Plot, y = Bacteria_Observed, color = Depth), width = 0.25, alpha = 0.5, data = bact_DIV) + 
  labs(y = "Observed richness", x = "Plot") + scale_color_manual(values = depth_colors) + theme_classic() + theme(legend.position = "none") + theme(axis.text = element_text(size=15), axis.title=element_text(size=20))

tiff("Figures/PlotS2A.tiff", width = 6, height = 4, units = 'in', res = 300)
bact_obvs_plot
dev.off()

## PLOT alone
ggplot() + geom_point(aes(x = Plot, y = emmean, color = Plot), 
                                       position = position_dodge(width = 0.5), size = 5, data = bact_obvs_emmeans_plot) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Plot), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = bact_obvs_emmeans_plot) +
  geom_jitter(aes(x = Plot, y = Bacteria_Observed, color = Plot), width = 0.25, alpha = 0.5, data = bact_DIV) + 
  labs(y = "Observed bacterial richness", x = "Plot") + scale_color_manual(values = plot_colors) + theme_classic() +
  theme(legend.position = "none")
```

### Shannon's diversity

```{r}
#anova
bact_shannon_anova <- lmer((Shannon) ~ Plot*Depth + (1|Transect), data = bact_DIV)
check_model(bact_shannon_anova)
anova(bact_shannon_anova, type = 2)
bact_shannon_emmeans <- emmeans(bact_shannon_anova, ~Plot*Depth)
bact_shannon_emmeans <- as.data.frame(bact_shannon_emmeans)

## PLOT alone
bact_shannon_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = bact_shannon_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = bact_shannon_emmeans) +
  geom_jitter(aes(x = Plot, y = Shannon, color = Depth), width = 0.25, alpha = 0.5, data = bact_DIV) + 
  labs(y = "Bacteria Shannon diversity", x = "Plot") + scale_color_manual(values = taxa_colors) + theme_classic() +
  theme(legend.position = "none")
```

### Taxonomic composition

#### microshades

```{r}
#agglomerate, normalize, and melt the phyloseq object
mdf_object<-merge_samples(bact_phyloseq_filt, "Plot")
bact_mdf_prep<-prep_mdf(mdf_object,subgroup_level = "Family")

#bact_mdf_prep$ID <- bact_mdf_prep$Sample
#bact_mdf_prep$ID <-gsub(pattern = "TE16S", replacement = "", x = bact_mdf_prep$ID)
#bact_mdf_prep <- bact_mdf_prep %>%
  #mutate(newID = paste0(substr(ID, nchar(ID), nchar(ID)), 
                                 #substr(ID, 1, nchar(ID) - 1)))

bact_mdf_prep$newID <- bact_mdf_prep$Plot

#create a color object for the specified data
tail(names(sort(table(bact_mdf_prep$Phylum))), 6)

bact_color_obj <- create_color_dfs(bact_mdf_prep, group_level = "Phylum", subgroup_level = "Family", selected_groups=c('Proteobacteria', 'Actinobacteriota', 'Acidobacteriota', 'Firmicutes'), cvd = FALSE)

#extract
bact_mdf <- bact_color_obj$mdf
bact_cdf <- bact_color_obj$cdf

bact_cdf <- color_reassign(bact_cdf, "Firmicutes", "micro_cvd_orange", group_level = "Phylum")
bact_cdf <- color_reassign(bact_cdf, "Acidobacteriota", "micro_green", group_level = "Phylum")
bact_cdf <- color_reassign(bact_cdf, "Actinobacteriota", rev("micro_blue"), group_level = "Phylum")
bact_cdf <- color_reassign(bact_cdf, "Proteobacteria", "micro_purple", group_level = "Phylum")

#reorder
sample_order <- bact_mdf_prep$newID
color_reorder <- reorder_samples_by(bact_mdf, bact_cdf, sample_variable = "newID")

#extract
bact_mdf <- color_reorder$mdf
bact_mdf$Family <-gsub(pattern = "Unknown", replacement = "Other", x = bact_mdf$Family)
bact_cdf <- color_reorder$cdf

#plot
bact_microshades <- plot_microshades(bact_mdf, bact_cdf, x = "newID") + scale_y_continuous(labels = scales::percent, expand = expansion(0)) + theme_classic() +
  theme(legend.position = "none") +
  theme(plot.margin = margin(6,10,6,6)) +
  xlab("Plot") + #facet_grid(~Plot, scales = "free_x") +
  theme(axis.text = element_text(size=12), axis.title=element_text(size=15)) 

tiff("Figures/PlotS6A1.tiff", width = 5, height = 4, units = 'in', res = 300)
bact_microshades
dev.off()


#legend
bact_micro_legend <- custom_legend(bact_mdf, bact_cdf, group_level = "Phylum",
  subgroup_level = "Family",legend_key_size = 1, legend_text_size = 15,
                                 legend_orientation = "horizontal")

tiff("Figures/PlotS6A2.tiff", width = 10, height = 4, units = 'in', res = 300)
bact_micro_legend
dev.off()

center_legend <- plot_grid(NULL, bact_micro_legend, nrow=1, rel_widths=c(0.04, 0.96))

#final microshades plot
plot_grid(bact_microshades, bact_micro_legend, nrow = 2, rel_heights = c(1, 0.3)) 
```

#### by phylum (filtered, not rarefied - just replace bact_phyloseq_filt)

```{r}
#create ASV-TAX table
taxa_names(bact_phyloseq_rare) <- paste0("Seq", seq(ntaxa(bact_phyloseq_rare)))
bact_ASV <- data.frame(otu_table(bact_phyloseq_rare))
bact_TAX <- data.frame(tax_table(bact_phyloseq_rare))
bact_t_ASV <- t(bact_ASV)
bact_ASV_TAX <- merge(bact_t_ASV, bact_TAX, by = "row.names")

## create the combined OTU table plus taxonomy string metacoder needs

#read it in using phyloseq object
tax_table(bact_phyloseq_rare)<-tax_table(bact_phyloseq_rare)[,1:7]
bact_x1<-parse_phyloseq(bact_phyloseq_rare) #this takes ages

#calculate relative abundance
bact_x1$data$abund_data <- calc_obs_props(bact_x1, "otu_table")

#calculate total abundance by forest type
bact_x1$data$Plot<-calc_taxon_abund(bact_x1, "abund_data",groups=bact_x1$data$sample_data$Plot) #this takes a while

### make taxonomy plots
# make sure we use functions from correct package
transform <- microbiome::transform
# merge rare taxa to speed up examples
bact_pseq <- microbiome::transform(bact_phyloseq_filt, "compositional")
bact_pseq <- aggregate_rare(bact_pseq, level = "Phylum", detection = 1/100, prevalence = 2/100)

#create dataframe from phyloseq 
bact_df <- psmelt(bact_pseq)

### construct a stacked barplot showing relative abundance at 'Phylum' level
#expand color palettes to fit data
colourCount <- length(unique(bact_df$Phylum))
getPalette <- colorRampPalette(brewer.pal(9, "Paired"))

#filter taxa by Abundance + merge "Unknown and Other" groups
bact_df$Phylum[bact_df$Abundance < 0.03] <- "Taxa < 3% Abundant"
bact_df$Phylum[bact_df$Phylum == "Unknown" | bact_df$Phylum == "Other" | bact_df$Phylum == "Taxa < 3% Abundant"] <- "Taxa < 3% Abundant"

#change relevel <3% to be at the bottom
#bact_df$Phylum <- relevel(factor(bact_df$Phylum), ref = "Taxa < 3% Abundant")
bact_df$Phylum <- factor(bact_df$Phylum)
lapply(bact_df, levels) #check levels
bact_df$Phylum <- fct_relevel(bact_df$Phylum, "Taxa < 3% Abundant", after = Inf)

#plot the relative abundance by Phylum
ggplot(bact_df, aes(fill=Phylum, y=Abundance, x=Depth)) + geom_bar(position="fill", stat="identity") + theme_minimal() + facet_grid(~Plot) + ylab("Relative Abundance") + theme(axis.text.x = element_text(angle=45)) + scale_y_continuous(labels = scales::percent) + scale_fill_manual(values = taxa_colors)
```

#### across family, across plot

```{r, fig.fullwidth=TRUE, fig.height=6, fig.width=15}
# merge rare taxa to speed up examples
bact_pseq_fam <- microbiome::transform(bact_phyloseq_filt, "compositional")
bact_pseq_fam <- aggregate_rare(bact_pseq_fam, level = "Family", detection = 1/100, prevalence = 2/100)

#create dataframe from phyloseq 
bact_df_fam <- psmelt(bact_pseq_fam)

#filter taxa by Abundance + merge "Unknown and Other" groups
bact_df_fam$Family[bact_df$Abundance < 0.05] <- "Taxa < 5% Abundant"
bact_df_fam$Family[bact_df_fam$Family == "Unknown Family" | bact_df_fam$Family == "Other" | bact_df_fam$Family == "Taxa < 5% Abundant"] <- "Taxa < 5% Abundant"

#plot the relative abundance by Phylum
bact_phylum_plot_facet <- ggplot(bact_df_fam, aes(fill=Family, y=Abundance, x=Depth)) + geom_bar(position="fill", stat="identity") + theme_minimal() + facet_grid(~Plot) + ylab("Relative Abundance") + theme(axis.text.x = element_text(angle=45)) + scale_y_continuous(labels = scales::percent) + scale_fill_manual(values = taxa_colors)

###PHYLUM####

# merge rare taxa to speed up examples
bact_pseq_phyla <- microbiome::transform(bact_phyloseq_filt, "compositional")
bact_pseq_phyla <- aggregate_rare(bact_pseq_phyla, level = "Phylum", detection = 1/100, prevalence = 2/100)

#create dataframe from phyloseq 
bact_df_phyla <- psmelt(bact_pseq_phyla)

#filter taxa by Abundance + merge "Unknown and Other" groups
bact_df_phyla$Phylum[bact_df$Abundance < 0.03] <- "Taxa < 3% Abundant"
bact_df_phyla$Phylum[bact_df_phyla$Phylum == "Unknown Phylum" | bact_df_phyla$Phylum == "Other" | bact_df_phyla$Phylum == "Taxa < 3% Abundant"] <- "Taxa < 3% Abundant"
bact_df_phyla$Phylum <- fct_relevel(bact_df_phyla$Phylum, "Taxa < 3% Abundant", after = Inf)

#plot the relative abundance by Phylum
bact_phylum_plot_facet <- ggplot(bact_df_phyla, aes(fill=Phylum, y=Abundance, x=Depth)) + geom_bar(position="fill", stat="identity") + theme_minimal() + facet_grid(~Plot) + ylab("Relative Abundance") + theme(axis.text.x = element_text(angle=45)) + scale_y_continuous(labels = scales::percent) + scale_fill_manual(values = taxa_colors)

tiff("Figures/PlotS5A.tiff", width = 6, height = 4, units = 'in', res = 300)
bact_phylum_plot_facet
dev.off()

#merge samples by category
bact_samples<-merge_samples(bact_phyloseq_rare, "Plot")
sample_data(bact_samples)$Plot = sample_names(bact_samples)
bact_samples@sam_data$Plot <- factor(bact_samples@sam_data$Plot, levels = c("30", "75", "250", "900", "1900"))

#aggregate samples by taxonomic rank (family here)
bact_family <- bact_samples %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at genus level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #transform to rel. abundance
  psmelt() %>%                                         #melt to long format
  filter(Abundance > 0.01) %>%                         #filter out low abundance taxa
  arrange(Abundance)                                     #sort data frame by abundance

#make an other group
bact_family.plot = bact_family |>
  mutate(Family = if_else(Abundance < 0.05, "Others (< 5%)", Family))

bact_family.plot$Family <- as.factor(bact_family.plot$Family)
levels(bact_family.plot$Family)
bact_family.plot$Family = factor(bact_family.plot$Family, levels = c("Acidothermaceae", "Alicyclobacillaceae", "Bacillaceae", "Paenibacillaceae", "Sporomusaceae", "Thermacetogeniaceae", "Xanthobacteraceae", "Others (< 5%)"))

bactgen_colors <- c("#43B284", "#C4E4B0", "#90D0E0FF","#E0A858FF",  "#499894FF", "#F8E898FF", "#206078FF", "#D3D3D3")

#making the plot using ggplots
bact_fam_plot <- ggplot(bact_family.plot, aes(x = Plot, y = Abundance, fill = Family)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = bactgen_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Family > 1%) \n") +
  xlab("Plot") +
  ggtitle("Composition of Bacterial Families") + scale_y_continuous(labels = scales::percent) + theme_bw() + #theme(axis.text = element_text(size=15), 
      #  axis.title=element_text(size=15)) +
   theme(axis.title.y = element_text(hjust=0.5, vjust = -4)) + guides(fill=guide_legend(title="Bacterial Families"))
```

### FAPROTAX

```{r}
bact_asv.function.table <- read.table("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/FAPROTAX/functional_table.tsv", sep="\t", header=TRUE)
bact_rel.asv.fn.df <- data.frame(bact_asv.function.table)

bact_asv.melt <- melt(bact_rel.asv.fn.df, id.vars = "group")
bact_asv.melt <- subset(bact_asv.melt, value > 0)
bact_asv.melt$variable <- as.factor(bact_asv.melt$variable)
bact_asv.melt <- bact_asv.melt %>% dplyr::rename(Sample = variable)
bact_asv.melt <- bact_asv.melt %>% dplyr::right_join(bact_env_data)
```

#### across plots

```{r}
#bubble graph
ggplot(bact_asv.melt[which(bact_asv.melt$value > 0),], aes(x=Plot, y=group)) + geom_point(aes(colour= Depth, size = value)) + theme(axis.text.x = element_blank()) + ylab("") + theme_minimal() + xlab("")

#heat map
ggplot(bact_asv.melt[which(bact_asv.melt$value > 0),], aes(x=Plot, y=group, fill = value)) + geom_tile() + theme_minimal() + ylab("Function") + scale_color_gradientn(colors=met.brewer("VanGogh3"))
```

#### compositional bar graphs

```{r}
bact_biofunction_16S <- bact_asv.melt[!is.na(bact_asv.melt$group),] 
bact_biofunction_16S$group<-gsub("_"," ",as.character(bact_biofunction_16S$group))

# put small abund stuff tgt under others
bact_df.plot <- bact_biofunction_16S |>
  mutate(group = if_else(value < 0.02, "Other (< 2%)", group))

bact_df.plot$group <- factor(bact_df.plot$group)
unique(bact_df.plot$group)
bact_df.plot$group <- fct_relevel(bact_df.plot$group, "Other (< 2%)", after = Inf)
bact_df.plot$Depth = factor(bact_df.plot$Depth, levels = c("100-150", "50-100", "10-50", "0-10"))

#making the plot using ggplots
ggplot(bact_df.plot, aes(y = Depth, x = value, fill = group)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = taxa_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  xlab("Relative Abundance (Function > 2%) \n") +
  ylab("Depth") +
  theme_minimal() + facet_grid(Plot~.) +
  ggtitle("Composition of Bacterial Functional Groups Along Degradation Transects in Brunei") + scale_x_continuous(labels = scales::percent) + # can change y axis to percentages (100%)
  theme_bw() + theme(axis.text.y = element_text(angle=45))

#to look at less common functional grps
strings_to_drop <- c("aerobic chemoheterotrophy", "chemoheterotrophy", "cellulolysis")
bact_df.plot_filt <- bact_df.plot %>% filter(!grepl(paste(strings_to_drop, collapse = "|"), group))

ggplot(bact_df.plot_filt, aes(x = Plot, y = value, fill = group)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = taxa_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  xlab("Relative Abundance (Function > 2%) \n") +
  ylab("Plot") +
  theme_minimal() + 
  #facet_grid(Transect~.) +
  ggtitle("Composition of Bacterial Functional Groups Along Degradation Transects in Brunei") + 
  scale_y_continuous(labels = scales::percent) + # can change y axis to percentages (100%)
  theme_bw() #+ theme(axis.text.y = element_text(angle=45))

#methane pathway
bact_methane_group <- bact_biofunction_16S %>% filter(stringr::str_starts(group, "meth"))
bact_methane_group$group <- factor(bact_methane_group$group)
bact_methane_group$Depth = factor(bact_methane_group$Depth, levels = c("0-10", "10-50", "50-100", "100-150"))

ggplot(bact_methane_group, aes(y = value, x = Plot, fill = group)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values=met.brewer("Hiroshige", 6)) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance") +
  xlab("Plot") +
  theme_minimal() + 
  ggtitle("Methane Dynamics Along Degradation Transects in Brunei") + scale_y_continuous(labels = scales::percent) +
  theme(axis.text.x = element_text(margin = margin(t = 10))) +
  theme(axis.text = element_text(size=12), 
        axis.title=element_text(size=15)) + 
   theme(legend.title = element_text(size = 15), 
       legend.text = element_text(size =12)) + guides(fill=guide_legend(title="Biofunction"))

#nitrogen pathway
bact_nitrogen_group <- bact_biofunction_16S %>% filter(stringr::str_detect(group, "nitr"))
bact_nitrogen_group$group <- factor(bact_nitrogen_group$group)
bact_nitrogen_group$Depth = factor(bact_nitrogen_group$Depth, levels = c("0-10", "10-50", "50-100", "100-150"))

#group
denitrification <- paste(c("denitrification", "nitrate denitrification", "nitrite denitrification"), collapse = '|')
bact_nitrogen_group <- bact_nitrogen_group %>% mutate(group = gsub(denitrification, "Denitrification", group)) 

respiration <- paste(c("nitrate respiration", "nitrite respiration", "nitrogen respiration"), collapse = '|')
bact_nitrogen_group <- bact_nitrogen_group %>% mutate(group = gsub(respiration, "Respiration", group)) 
bact_nitrogen_group <- bact_nitrogen_group %>%
  group_by(Transect, Plot, Depth, Sample, group) %>%  # Group by columns that should remain unchanged
  summarise(value = sum(value), .groups = "drop")  # Summarize the 'value' column by taking the sum


#relevel
bact_nitrogen_group$group = factor(bact_nitrogen_group$group, levels = 
                                c("nitrate reduction", "Denitrification", "Respiration", "nitrogen fixation"))

nitrogen_colors <- c("#b9b9b8","#4692b0", "#1e3d14", "#a9845b")

ggplot(bact_nitrogen_group, aes(y = value, x = Plot, fill = group)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values=nitrogen_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance") +
  xlab("Plot") +
  theme_minimal() +
  ggtitle("Nitrogen Dynamics Along Degradation Transects in Brunei") + scale_y_continuous(labels = scales::percent) +
  theme_bw() + theme(axis.text.x = element_text(margin = margin(t = 10))) +
  theme(axis.text = element_text(size=12), 
        axis.title=element_text(size=15)) + 
   theme(legend.title = element_text(size = 15), 
       legend.text = element_text(size =12)) + guides(fill=guide_legend(title="Biofunction"))
```

### picrust2

#### pathways relative abund

```{r}
#import data (predicted pathways)
bact_pathways <- read_tsv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/picrust2-2.5.2/picrust2_out_pipeline/pathways_out/path_abun_unstrat.tsv")

bact_pathways <- bact_pathways %>% pivot_longer(!pathway, names_to = "Sample", values_to = "abund")
bact_pathways$Sample <-gsub(pattern = "_S.*", replacement = "", x = bact_pathways$Sample)
bact_pathways <- bact_pathways %>% dplyr::left_join(bact_env_data)

#remove data w NA in group
bact_picrust <- bact_pathways[!is.na(bact_pathways$pathway),] 
bact_picrust$pathway<-gsub("_"," ",as.character(bact_picrust$pathway))

bact_picrust.df <- bact_picrust %>% group_by(Sample) %>% mutate(relabund = abund/sum(abund))

# put small abund stuff tgt under others
bact_df.picrust.plot = bact_picrust.df |>
  mutate(pathway = if_else(relabund < 0.008, NA, pathway)) #making small abundance NA for plot

bact_df.picrust.plot$pathway <- factor(bact_df.picrust.plot$pathway)
unique(bact_df.picrust.plot$pathway)

#making the plot using ggplots
bact_df.picrust.plot <- bact_df.picrust.plot %>% subset(!is.na(pathway))

bact_metacyc_pathways <- read.csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/MetaCyc_pathways .csv") %>% dplyr::rename(pathway = Bacteria)
bact_picrust_clean <- bact_df.picrust.plot %>% dplyr::left_join(bact_metacyc_pathways) 
bact_picrust_clean <- bact_picrust_clean %>% select(-1,) %>% dplyr::rename(Pathway = Re.label, Biofunction = Group)
unique(bact_picrust_clean$Pathway)

bact_picrust_clean$Pathway = factor(bact_picrust_clean$Pathway, levels = c(
  "Amino acid biosynthesis", "L-isoleucine biosynthesis", "L-valine biosynthesis", "Aerobic respiration", "TCA cycle", 
  "Isobutanol biosynthesis", "Gondoate biosynthesis (anaerobic)", "Fatty acid elongation", "GDP-mannose biosynthesis", "cis-Vaccenate biosynthesis", "Pyrimidine nucleobases salvage I", "Romatic biogenic amine degradation", "Peptidoglycan maturation"
))

bact_pathway_colors <- c("#2d223c","#90719f","#dec5da","#92351e","#ecb27d", "#fbe3c2","#b9563f", "#134b73", "#72aeb6", "#2f70a1", "#abc9c8", "#0a3351", "#4692b0")

bact_function_plot <- ggplot(data = bact_picrust_clean, aes(y = relabund, x = Plot, fill = Pathway)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = bact_pathway_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Function > 0.8%) \n") +
  xlab("Plot") +
  theme_bw()+
  theme(plot.title = element_blank()) + 
  scale_y_continuous(labels = scales::percent) + # can change y axis to percentages (100%) 
   theme(axis.text.x = element_text(margin = margin(t = 5))) +
  theme(axis.text = element_text(size=10), 
        axis.title=element_text(size=12)) + 
  theme(axis.title.y = element_text(hjust=0.5, vjust = -4)) +
   theme(legend.title = element_text(size = 12), 
       legend.text = element_text(size =20)) + theme(legend.position = "none")

tiff("Figures/Plot3D.tiff", width = 3, height = 4, units = 'in', res = 300)
bact_function_plot
dev.off()
```

#### ANOVAs for function

```{r}
bact_picrust_clean_anova <- bact_picrust.df
bact_picrust_clean_anova$pathway <- factor(bact_picrust_clean_anova$pathway)
unique(bact_picrust_clean_anova$pathway)

# put small abund stuff tgt under others
bact_picrust_clean_anova = bact_picrust.df |>
  mutate(pathway = if_else(relabund < 0.008, "Other", pathway)) #making small abundance NA for plot

#making the plot using ggplots
bact_picrust_clean_anova <- bact_picrust_clean_anova %>% subset(!is.na(pathway))
bact_picrust_clean_anova <- bact_picrust_clean_anova %>% dplyr::left_join(bact_metacyc_pathways) 
bact_picrust_clean_anova <- bact_picrust_clean_anova %>% select(-1,) %>% dplyr::rename(Pathway = Re.label, Biofunction = Group)
unique(bact_picrust_clean_anova$Pathway)

bact_picrust_clean_anova$Pathway = factor(bact_picrust_clean_anova$Pathway, levels = c(
  "Amino acid biosynthesis", "L-isoleucine biosynthesis", "L-valine biosynthesis", "Aerobic respiration", "TCA cycle", 
  "Isobutanol biosynthesis", "Gondoate biosynthesis (anaerobic)", "Fatty acid elongation", "GDP-mannose biosynthesis", "cis-Vaccenate biosynthesis", "Pyrimidine nucleobases salvage I", "Romatic biogenic amine degradation", "Peptidoglycan maturation"
))

head(bact_picrust_clean_anova)
bact_function_bare_pathway <- bact_picrust_clean_anova %>% dplyr:: select(Sample, Transect, Plot, Depth, relabund, Pathway) 
bact_function_bare_pathway <- bact_function_bare_pathway %>% group_by(Sample, Transect, Plot, Depth, Pathway) %>% mutate(totalrelabund = sum(relabund)) %>% select(-relabund) %>% unique() %>% pivot_wider(names_from = Pathway, values_from = totalrelabund)
summary(bact_function_bare_pathway)
bact_function_bare_pathway[is.na(bact_function_bare_pathway)] <- 0

bact_function_bare_biofunction <- bact_picrust_clean_anova %>% dplyr:: select(Sample, Transect, Plot, Depth, relabund, Biofunction) 
bact_function_bare_biofunction <- bact_function_bare_biofunction %>% group_by(Sample, Transect, Plot, Depth, Biofunction) %>% mutate(totalrelabund = sum(relabund)) %>% select(-relabund) %>% unique() %>% pivot_wider(names_from = Biofunction, values_from = totalrelabund)
bact_function_bare_biofunction[is.na(bact_function_bare_biofunction)] <- 0

#ANOVAs
library(glmmTMB)

hist(log(bact_function_bare_pathway$`Aerobic respiration`))
##aerobic respiration (significantly higher for top depth compared to other 3 depths)
function_anova <- lmer(log(`Aerobic respiration`) ~ Plot+Depth + (1|Transect), data = bact_function_bare_pathway)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Depth), adjust = "tukey")

hist((bact_function_bare_pathway$`cis-Vaccenate biosynthesis`))
##cis-Vaccenate biosynthesis (significant between degraded and most forested - A,A,AB,B,B - lower for degraded)
function_anova <- glmmTMB((`cis-Vaccenate biosynthesis`) ~ Plot+Depth + (1|Transect), data = bact_function_bare_pathway, family = beta_family(link = "logit"), ziformula = ~1)
library(DHARMa)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist(log(bact_function_bare_pathway$`Isobutanol biosynthesis`))
##`Isobutanol biosynthesis` (significant between degraded and all forested - A,A,B,B,B - lower for degraded)
function_anova <- lmer(log(`Isobutanol biosynthesis`) ~ Plot+Depth + (1|Transect), data = bact_function_bare_pathway)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

hist(log(bact_function_bare_pathway$`Gondoate biosynthesis (anaerobic)`))
##Gondoate biosynthesis (anaerobic)
function_anova <- glmmTMB((`Gondoate biosynthesis (anaerobic)`) ~ Plot+Depth + (1|Transect), data = bact_function_bare_pathway, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist(log(bact_function_bare_pathway$`L-valine biosynthesis`))
##L-valine biosynthesis (significant between degraded and all forested - A,A,B,B,B - lower for degraded)
function_anova <- glmmTMB((`L-valine biosynthesis`) ~ Plot+Depth + (1|Transect), data = bact_function_bare_pathway, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist(log(bact_function_bare_pathway$`Amino acid biosynthesis`))
##amino acid biosynthesis (total) (significant between degraded and all forested - A,A,B,B,B - lower for degraded)
function_anova <- glmmTMB((`Amino acid biosynthesis`) ~ Plot+Depth + (1|Transect), data = bact_function_bare_pathway, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist((bact_function_bare_biofunction$`Structural bacterial function`))
##`Structural bacterial function`
function_anova <- glmmTMB((`Structural bacterial function`) ~ Plot+Depth + (1|Transect), data = bact_function_bare_biofunction, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist((bact_function_bare_biofunction$`Energy production`))
##Energy production (specifically 900 + 1900 top layer is higher than every other layer - 250 is transitional
function_anova <- lmer((`Energy production`) ~ Plot*Depth + (1|Transect), data = bact_function_bare_biofunction)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot*Depth), adjust = "tukey")
function_anova_emmeans <- emmeans(function_anova, ~Plot*Depth)
function_anova_emmeans <- as.data.frame(function_anova_emmeans)

# PLOT x DEPTH
ggplot() + geom_point(aes(x = Plot, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = function_anova_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = function_anova_emmeans) +
  geom_jitter(aes(x = Plot, y = `Energy production`, color = Depth), width = 0.25, alpha = 0.5, data = bact_function_bare_biofunction) + 
  labs(y = "Energy Production", x = "Plot") + scale_color_manual(values = depth_colors) + theme_classic()

##Fatty acids (energy)
function_anova <- glmmTMB((`Fatty acids (energy)`) ~ Plot+Depth + (1|Transect), data = bact_function_bare_biofunction, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
```

#### manually running ggpicrust2

##### KEGG pathway

```{r, eval = FALSE}
#load metadata
bact_metadata <- as_tibble(bact_env_data)

#load KEGG pathway abundance
bact_kegg_abundance <- ko2kegg_abundance("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/picrust2-2.5.2/picrust2_out_pipeline/KO_metagenome_out/pred_metagenome_unstrat.tsv") 
names(bact_kegg_abundance) = gsub(pattern = "_S.*", replacement = "", x = names(bact_kegg_abundance))

#pathway differential abundance analysis (DAA) using DESeq2 (multiple factors)
bact_daa_results <- pathway_daa(abundance = bact_kegg_abundance, metadata = bact_metadata, group = "Plot", daa_method = "DESeq2", select = NULL, reference = "1900") 

#filter results
unique(bact_daa_results$method)
bact_daa_results <- bact_daa_results[bact_daa_results$method == "DESeq2", ]
bact_daa_sig_results <- bact_daa_results[bact_daa_results$p_adjust< 0.05, ]
bact_daa_sig_results <- bact_daa_sig_results %>% drop_na(feature)

#split for practical reasons
daa_sig_results1 <- bact_daa_sig_results[1:75,]
daa_sig_results2 <- bact_daa_sig_results[76:150,]
daa_sig_results3 <- bact_daa_sig_results[151:225,]
daa_sig_results4 <- bact_daa_sig_results[226:300,]
daa_sig_results5 <- bact_daa_sig_results[301:375,]
daa_sig_results6 <- bact_daa_sig_results[376:450,]
daa_sig_results7 <- bact_daa_sig_results[451:525,]
daa_sig_results8 <- bact_daa_sig_results[526:600,]
daa_sig_results9 <- bact_daa_sig_results[601:675,]
daa_sig_results10 <- bact_daa_sig_results[676:750,]
daa_sig_results11 <- bact_daa_sig_results[751:825,]
daa_sig_results12 <- bact_daa_sig_results[826:900,]
daa_sig_results13 <- bact_daa_sig_results[901:975,]
daa_sig_results14 <- bact_daa_sig_results[976:1050,]
daa_sig_results15 <- bact_daa_sig_results[1051:1125,]
daa_sig_results16 <- bact_daa_sig_results[1126:1200,]
daa_sig_results17 <- bact_daa_sig_results[1201:1275,]
daa_sig_results18 <- bact_daa_sig_results[1276:1317,]

#annotate pathway results using KO to KEGG conversion
daa_annotated_sig_results1 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results1, ko_to_kegg = TRUE)
daa_annotated_sig_results2 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results2, ko_to_kegg = TRUE)
daa_annotated_sig_results3 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results3, ko_to_kegg = TRUE)
daa_annotated_sig_results4 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results4, ko_to_kegg = TRUE)
daa_annotated_sig_results5 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results5, ko_to_kegg = TRUE)
daa_annotated_sig_results6 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results6, ko_to_kegg = TRUE)
daa_annotated_sig_results7 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results7, ko_to_kegg = TRUE)
daa_annotated_sig_results8 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results8, ko_to_kegg = TRUE)
daa_annotated_sig_results9 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results9, ko_to_kegg = TRUE)
daa_annotated_sig_results10 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results10, ko_to_kegg = TRUE)
daa_annotated_sig_results11 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results11, ko_to_kegg = TRUE)
daa_annotated_sig_results12 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results12, ko_to_kegg = TRUE)
daa_annotated_sig_results13 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results13, ko_to_kegg = TRUE)
daa_annotated_sig_results14 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results14, ko_to_kegg = TRUE)
daa_annotated_sig_results15 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results15, ko_to_kegg = TRUE)
daa_annotated_sig_results16 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results16, ko_to_kegg = TRUE)
daa_annotated_sig_results17 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results17, ko_to_kegg = TRUE)
daa_annotated_sig_results18 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results18, ko_to_kegg = TRUE)

#bind full dataset
bact_daa_results_full <- rbind(daa_annotated_sig_results1, daa_annotated_sig_results2, 
                          daa_annotated_sig_results3, daa_annotated_sig_results4,
                          daa_annotated_sig_results5, daa_annotated_sig_results6,
                          daa_annotated_sig_results7, daa_annotated_sig_results8, 
                          daa_annotated_sig_results9, daa_annotated_sig_results10,
                          daa_annotated_sig_results11, daa_annotated_sig_results12,
                          daa_annotated_sig_results13, daa_annotated_sig_results14, 
                          daa_annotated_sig_results15, daa_annotated_sig_results16,
                          daa_annotated_sig_results17, daa_annotated_sig_results18)
bact_daa_results_full <- bact_daa_results_full[!is.na(bact_daa_results_full$pathway_name),]
#choose levels to compare
bact_daa_results_30 <- bact_daa_results_full[bact_daa_results_full$group1 == "30", ]
bact_daa_results_30 <- bact_daa_results_30[bact_daa_results_30$group2 == "1900", ]

#filter
bact_daa_results_30$p_adjust <- round(bact_daa_results_30$p_adjust,5)
bact_kegg <- bact_kegg_abundance
bact_kegg$avgabund <- rowMeans(bact_kegg, na.rm = TRUE)
bact_kegg <- bact_kegg %>% rownames_to_column() %>% select(rowname, avgabund) %>% dplyr::rename(feature = rowname)
bact_daa_results_30 <- bact_daa_results_30 %>% dplyr::left_join(bact_kegg)

bact_low_p_feature <- bact_daa_results_30[order(bact_daa_results_30$avgabund, decreasing = TRUE),]$feature[1:20]
```

##### pathway error bar plot

```{r, eval = FALSE}
ggpicrust2::pathway_errorbar(abundance = bact_kegg_abundance, daa_results_df = bact_daa_results_30, 
                             Group = bact_metadata$Plot, ko_to_kegg = TRUE, p_values_threshold = 0.05, 
                             order = "pathway_class", select = bact_low_p_feature, p_value_bar = TRUE, 
                             colors = NULL, x_lab = "pathway_name")
```

##### EC pathway error barplot
```{r, eval = FALSE}
bact_ec_abundance <- read_tsv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/picrust2-2.5.2/picrust2_out_pipeline/EC_metagenome_out/pred_metagenome_unstrat.tsv")
names(bact_ec_abundance) = gsub(pattern = "_S.*", replacement = "", x = names(bact_ec_abundance))

#DAA
bact_ec_daa_results <- pathway_daa(abundance = bact_ec_abundance %>% column_to_rownames("function"), metadata = bact_metadata, group = "Plot", daa_method = "DESeq2")

#annotate pathway results without KO to KEGG conversion
bact_ec_daa_annotated_results <- pathway_annotation(pathway = "EC", daa_results_df = bact_ec_daa_results, ko_to_kegg = FALSE)
#filter
bact_ec_daa_annotated_results <- bact_ec_daa_annotated_results[!is.na(bact_ec_daa_annotated_results$feature),]
bact_ec_daa_annotated_results <- bact_ec_daa_annotated_results[bact_ec_daa_annotated_results$group1 == "30", ]
bact_ec_daa_annotated_results <- bact_ec_daa_annotated_results[bact_ec_daa_annotated_results$group2 == "1900", ]
bact_ec_daa_annotated_results$p_adjust <- round(bact_ec_daa_annotated_results$p_adjust,5)
bact_low_p_feature_ec <- bact_ec_daa_annotated_results[order(bact_ec_daa_annotated_results$p_adjust), ]$feature[1:20]

#pathway error bar plot
pathway_errorbar(abundance = bact_ec_abundance %>% column_to_rownames("function"), daa_results_df = bact_ec_daa_annotated_results, Group = bact_metadata$Plot, ko_to_kegg = FALSE, p_values_threshold = 0.05, order = "group", select = bact_low_p_feature_ec, p_value_bar = TRUE, colors = NULL, x_lab = "description")
```

##### MetaCyc pathway analysis

```{r, eval = FALSE}
#load MetaCyc pathway abundance and metadata
bact_metacyc_abundance <- read_tsv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/picrust2-2.5.2/picrust2_out_pipeline/pathways_out/path_abun_unstrat.tsv")
names(bact_metacyc_abundance) = gsub(pattern = "_S.*", replacement = "", x = names(bact_metacyc_abundance))

#pathway DAA using DESeq2 method
bact_metacyc_daa_results <- pathway_daa(abundance = bact_metacyc_abundance %>% column_to_rownames("pathway"), metadata = bact_metadata, group = "Plot", daa_method = "DESeq2")

#annotate MetaCyc pathway results without KO to KEGG conversion
bact_metacyc_daa_annotated_results <- pathway_annotation(pathway = "MetaCyc", daa_results_df = bact_metacyc_daa_results, ko_to_kegg = FALSE)

bact_metacyc_daa_annotated_results <- bact_metacyc_daa_annotated_results[!is.na(bact_metacyc_daa_annotated_results$feature),]
bact_metacyc_daa_annotated_results <- bact_metacyc_daa_annotated_results[bact_metacyc_daa_annotated_results$group1 == "250", ]
bact_metacyc_daa_annotated_results <- bact_metacyc_daa_annotated_results[bact_metacyc_daa_annotated_results$group2 == "1900", ]
bact_metacyc_daa_annotated_results$p_adjust <- round(bact_metacyc_daa_annotated_results$p_adjust,5)
bact_low_p_feature_metacyc <- bact_metacyc_daa_annotated_results[order(bact_metacyc_daa_annotated_results$p_adjust), ]$feature[1:20]

#pathway error bar plot
pathway_errorbar(abundance = bact_metacyc_abundance %>% column_to_rownames("pathway"), daa_results_df = bact_metacyc_daa_annotated_results, Group = bact_metadata$Plot, ko_to_kegg = FALSE, p_values_threshold = 0.05, order = "p_values", select = bact_low_p_feature_metacyc, p_value_bar = TRUE, colors = NULL, x_lab = "description")
bact_metacyc_abundance_subset <- bact_metacyc_abundance[1:20,]

#pathway heatmap
pathway_heatmap(abundance = bact_metacyc_abundance_subset %>% filter(pathway %in% bact_metacyc_abundance_subset$pathway) %>% column_to_rownames("pathway"), metadata = bact_metadata, group = "Plot")
```

##### picrust2 PCA

```{r, eval = FALSE}
#pathway PCA plot
bact_abundance <- bact_metacyc_abundance %>% column_to_rownames("pathway")

#manual function for PCA
pathway_pca <- function(abundance, metadata, group){
# due to NSE notes in R CMD check
PC1 = PC2 = Group = NULL
# Perform PCA on the abundance data, keeping the first two principal components
pca_axis <- stats::prcomp(t(bact_abundance), center = TRUE, scale = TRUE)$x[,1:2]

#calculate the proportion of total variation explained by each PC
pca_proportion <- stats::prcomp(t(bact_abundance), center = TRUE, scale = TRUE)$sdev[1:2]/sum(stats::prcomp(t(bact_abundance), center = TRUE, scale = TRUE)$sdev)*100

#combine the PCA results with the metadata information
pca <- cbind(pca_axis, bact_metadata %>% select(c(Plot),))
pca$Plot

levels <- length(levels(factor(pca$Plot)))

#create a ggplot object for the PCA scatter plot
Fig1a.taxa.pca <- ggplot2::ggplot(pca,aes(x = PC1,y = PC2))+
 ggplot2::geom_point(size=5,ggplot2::aes(color=Plot),show.legend = T) +
 ggplot2::scale_color_manual(values=plot_colors) +
  ggplot2::stat_ellipse(ggplot2::aes(color = Plot, fill = Plot),geom = "polygon",
              level=0.95, alpha = 0.15, show.legend = F) + 
   ggplot2::scale_fill_manual(values=plot_colors) +
 ggplot2::labs(x=paste0("PC1(",round(pca_proportion[1],1),"%)"),y=paste0("PC2(",round(pca_proportion[2],1),"%)"))+
 ggplot2::theme_bw() +
 ggplot2::theme(axis.line=ggplot2::element_line(colour = "black"),
       axis.title=ggplot2::element_text(color="black",size = 16, face = "bold"),
       panel.grid.major = ggplot2::element_blank(),
       panel.grid.minor = ggplot2::element_blank(),
       panel.background = ggplot2::element_blank(),
       axis.text = ggplot2::element_text(color="black",size=10,face = "bold"),
       legend.text = ggplot2::element_text(size = 16),
       legend.title = ggplot2::element_text(size = 16, face = "bold"))

#create a ggplot object for the density plot of PC1
#plot the density of PC1 colored by group
Fig1a.taxa.pc1.density <-
 ggplot2::ggplot(pca) +
 ggplot2::geom_density(ggplot2::aes(x=PC1, group=Plot, fill=Plot), # Plot the density of PC1
              color="black", alpha=0.5, position = 'identity',show.legend = F) +
 ggplot2::scale_fill_manual(values=plot_colors) + # Manually set the fill color for each group
 ggplot2::theme_classic() + # Set the classic theme
 ggplot2::scale_y_discrete(expand = c(0,0.001)) + # Scale the y-axis with a small expansion to improve appearance
 ggplot2::labs(x=NULL, y=NULL) + # Remove x and y labels
 ggplot2::theme(axis.text.x=ggplot2::element_blank(), # Remove x axis text
       axis.ticks.x = ggplot2::element_blank())

# Plot the density of PC2 colored by Plot
Fig1a.taxa.pc2.density <-
 ggplot2::ggplot(pca) +
 ggplot2::geom_density(ggplot2::aes(x=PC2, group=Plot, fill=Plot), # Plot the density of PC2
              color="black", alpha=0.5, position = 'identity',show.legend = F) +
 ggplot2::scale_fill_manual(values=plot_colors) + # Manually set the fill color for each group
 ggplot2::theme_classic() + # Set the classic theme
 ggplot2::scale_y_discrete(expand = c(0,0.001)) + # Scale the y-axis with a small expansion to improve appearance
 ggplot2::labs(x=NULL, y=NULL) + # Remove x and y labels
 ggplot2::theme(axis.text.y = ggplot2::element_blank(), # Remove y axis text
       axis.ticks.y = ggplot2::element_blank()) +
 ggplot2::coord_flip() # Flip the plot to change it from a horizontal to a vertical orientation

#combine the two plots into a single plot with the PC1 plot on top and the PC2 plot on the right
Fig1a.taxa.pca %>%
 aplot::insert_top(Fig1a.taxa.pc1.density, height = 0.3) %>% # Insert the PC1 plot on top with a height of 0.3
 aplot::insert_right(Fig1a.taxa.pc2.density, width=0.3) %>% # Insert the PC2 plot on the right with a width of 0.3
 ggplotify::as.ggplot() # Convert the combined plot to a ggplot object
}

#plot
bact_pca_picrust <- pathway_pca(abundance = abundance, 
            metadata = bact_metadata, group = "Plot")
bact_pca_picrust
```

### Major shifts

```{r}
#richness and shannon div
bact_DIV_merge <- bact_DIV %>% dplyr::select(Sample, Bacteria_Observed, Shannon, Transect, Plot, Depth)

#abundance of major phyla
bact_df_merge <- bact_df %>% dplyr::select(Sample, Phylum, Abundance, Transect, Plot, Depth)
bact_phyla_list <- paste(c("Firmicutes", "Actinobact", "Acidobacter", "Proteobacter"), collapse = '|')
bact_df_merge <- bact_df_merge %>% filter(stringr::str_detect(Phylum, bact_phyla_list))
bact_df_merge <- bact_df_merge %>% group_by(Phylum) %>%
  pivot_wider(names_from = Phylum, values_from = Abundance)
bact_df_merge[is.na(bact_df_merge)] <- 0

#abundance of major families
bact_df_fam <- psmelt(bact_pseq_fam)
unique(bact_df_fam$Family)
bact_fam_merge <- bact_df_fam %>% dplyr::select(Sample, Family, Abundance, Transect, Plot, Depth)
bact_fam_list <- paste(c("Alicyclobacillaceae", "Paenibacillaceae", "Acidothermaceae", "Sporomusaceae", "Mesorhizobium", "Thermacetogeniaceae", "Xanthobacteraceae"), collapse = '|')
bact_fam_merge <- bact_fam_merge %>% filter(stringr::str_detect(Family, bact_fam_list))
bact_fam_merge <- bact_fam_merge %>% group_by(Family) %>%
  pivot_wider(names_from = Family, values_from = Abundance)

#abundance of functional grps (faprotax)
bact_func_merge <- bact_df.plot %>% dplyr::select(Sample, group, value, Transect, Plot, Depth)
bact_func_list <- paste(c("cellulolysis", "chemohetero", "ferment"), collapse = '|')
bact_func_merge <- bact_func_merge %>% filter(stringr::str_detect(group, bact_func_list))
bact_func_merge <- bact_func_merge %>% group_by(group) %>%
  pivot_wider(names_from = group, values_from = value)

bact_major_merge <- bact_DIV_merge %>% dplyr::right_join(bact_df_merge) %>% dplyr::right_join(bact_fam_merge) %>% dplyr::right_join(bact_func_merge)
bact_major_merge <- bact_major_merge %>% dplyr::rename(Bacteria_Shannon = Shannon, cellulolysis_bact = cellulolysis, 
                                       `aerobic_chemoheterotrophy_bact` = `aerobic chemoheterotrophy`, 
                                       chemoheterotrophy_bact = chemoheterotrophy, fermentation_bact = fermentation) %>% 
  dplyr::select(-Sample) %>% select(Transect, Plot, Depth, everything())
head(bact_major_merge)

#make new datasets
bact_df_merge_sum <- bact_df_merge %>% group_by(Plot) %>% summarise_if(is.numeric, mean, na.rm = TRUE)
bact_df_merge_sum <- bact_df_merge_sum %>% pivot_longer(!c(Plot), names_to = "Group", values_to = "Abundance")
bact_df_merge_sum$Plot <- as.numeric(bact_df_merge_sum$Plot)
bact_df_merge_sum$Group <- factor(bact_df_merge_sum$Group, levels = c("Firmicutes", "Proteobacteria", "Actinobacteriota", "Acidobacteriota"))

bact_df_merge_new <- bact_df_merge %>% dplyr::select(-Sample) %>% pivot_longer(!c(Plot, Transect, Depth), names_to = "Group", values_to = "Abundance")
bact_df_merge_new$Plot <- as.numeric(bact_df_merge_new$Plot)
bact_df_merge_new$Group <- factor(bact_df_merge_new$Group, levels = c("Firmicutes", "Proteobacteria", "Actinobacteriota", "Acidobacteriota"))

#family
bact_fam_merge_sum <- bact_fam_merge %>% group_by(Plot) %>% summarise_if(is.numeric, mean, na.rm = TRUE)
bact_fam_merge_sum <- bact_fam_merge_sum %>% pivot_longer(!c(Plot), names_to = "Group", values_to = "Abundance")
bact_fam_merge_sum$Plot <- as.numeric(bact_fam_merge_sum$Plot)
bact_fam_merge_sum$Group <- factor(bact_fam_merge_sum$Group, levels = c("Xanthobacteraceae","Acidothermaceae","Paenibacillaceae","Thermacetogeniaceae", "Sporomusaceae", "Alicyclobacillaceae"))

bact_fam_merge_new <- bact_fam_merge %>% dplyr::select(-Sample) %>% pivot_longer(!c(Plot, Transect, Depth), names_to = "Group", values_to = "Abundance")
bact_fam_merge_new$Plot <- as.numeric(bact_fam_merge_new$Plot)
bact_fam_merge_new$Group <- factor(bact_fam_merge_new$Group, levels = c("Xanthobacteraceae","Acidothermaceae","Paenibacillaceae","Thermacetogeniaceae", "Sporomusaceae", "Alicyclobacillaceae"))

bact_taxshift_plot <- ggplot() + theme_classic() + 
  geom_line(aes(x = Plot, y = Abundance, color = Group), data = bact_df_merge_sum) + 
  geom_point(aes(x = Plot, y = Abundance, color = Group), size = 6, data = bact_df_merge_sum) +
  geom_jitter(aes(x = Plot, y = Abundance, color = Group), 
             alpha = 0.5, width = 0.15, data = bact_df_merge_new) + 
  #geom_line(aes(x = Plot, y = Abundance, color = Group), linetype="dotted", data = bact_fam_merge_sum) + 
  #geom_point(aes(x = Plot, y = Abundance, color = Group), size = 1, data = bact_fam_merge_sum) +
  ylab("Relative Abundance") + scale_color_manual(values = c("#DD5129", "#FAB255", "#0F7BA2", "#43B284", "#4c2f61"))

tiff("Figures/Plot3A.tiff", width = 6, height = 4, units = 'in', res = 300)
bact_taxshift_plot
dev.off()
```

### CCA
 
#### Prep data
 
```{r, eval = FALSE}
bact_community_cca <- bact_community[!(row.names(bact_community) %in% c("TE16SBED","TE16SCAC", "TE16SECD", "TE16SCCC","TE16SDAD", "TE16SDEC", "TE16SAAC", "TE16SDAA", "TE16SDDA", "TE16SDDB", "TE16SEBA")),]

bact_env_cca <- bact_env_data %>% tibble::column_to_rownames("Sample") %>% dplyr::select(-Transect, -Plot, -Depth)
bact_env_cca <- bact_env_cca[!(row.names(bact_env_cca) %in% c("TE16SBED","TE16SCAC", "TE16SECD", "TE16SCCC","TE16SDAD", "TE16SDEC", "TE16SAAC", "TE16SDAA", "TE16SDDA", "TE16SDDB", "TE16SEBA")),]
 
#Hellinger transformation to species matrix due to rare sp.
bact_community_trans <- decostand(bact_community_cca, "hellinger")
#scale and center soil parameters due to diff units
bact_env_trans <- decostand(bact_env_cca, method = "standardize")
#variables are now centered around a mean of 0
round(apply(bact_env_trans, 2, mean), 1)
#and scaled to have a standard deviation of 1
apply(bact_env_trans, 2, sd)
```
 
#### Run CCA
 
```{r, eval = FALSE}
#run CCA
bact_CCA <- cca(bact_community_trans~., bact_env_trans)
plot(bact_CCA) #first visualisation
summary(bact_CCA) #look at model
 
#stepwise model
bact_finalmodel<- ordistep(bact_CCA, scope=formula(bact_CCA))
vif<-vif.cca(bact_finalmodel)
vif #go to original dataset and remove redundant terms (VIF > 10)
bact_finalmodel
#14.4% inertia explained
plot(bact_finalmodel)
capture.output(vif, file = "vif.txt")
summary(bact_finalmodel)
screeplot(bact_finalmodel)
 
#significance tests
anova.cca(bact_finalmodel)
anova.cca(bact_finalmodel, by="terms")
anova.cca(bact_finalmodel, by="axis")
 
#colors for plot
bact_env_data$color <- bact_env_data$Plot %>% as.character()
bact_env_data$color[bact_env_data$color == "30"] <- "#74c8c3"
bact_env_data$color[bact_env_data$color == "75"] <- "#5a97c1" 
bact_env_data$color[bact_env_data$color == "250"] <- "#0a2e57"
bact_env_data$color[bact_env_data$color == "900"] <- "#d8d97a"
bact_env_data$color[bact_env_data$color == "1900"] <- "#669d62"
summary(bact_env_data)
 
#generate biplot for sites and soil var
plot(c(-2.5, 2), c(-1.5, 1.5), xlab="CCA1 (30.0%)", ylab="CCA2 (15.8%)", type="n")
abline(h = 0, v = 0, lty = 2, lwd =.5)
points(bact_finalmodel, dis="lc", pch=21, col="black", bg=bact_env_data$color, cex=1.5, scaling = "sites")
text(bact_finalmodel, dis="cn", cex = 1.5, scaling = "sites")
bact_cca_plot <- recordPlot()
 
#generate biplot of sites and species
plot(bact_finalmodel, scaling = "sites", type="n")
points(bact_finalmodel, "species", pch=1, col="forestgreen", cex=0.8, scaling = "sites")
text(bact_finalmodel, "species", col="black", cex=0.4, scaling = "sites")

#colors for depth
bact_env_data$color <- bact_env_data$Depth %>% as.character()
bact_env_data$color[bact_env_data$color == "0-10"] <- "#e7e5cc"
bact_env_data$color[bact_env_data$color == "10-50"] <- "#c2d6a4" 
bact_env_data$color[bact_env_data$color == "50-100"] <- "#669d62"
bact_env_data$color[bact_env_data$color == "100-150"] <- "#192813"
summary(bact_env_data)
 
#generate biplot for sites and soil var
plot(c(-2.5, 2), c(-1.5, 1.5), xlab="CCA1 (30.0%)", ylab="CCA2 (15.8%)", type="n")
abline(h = 0, v = 0, lty = 2, lwd =.5)
points(bact_finalmodel, dis="lc", pch=21, col="black", bg=bact_env_data$color, cex=1.5, scaling = "sites")
text(bact_finalmodel, dis="cn", cex = 1.5, scaling = "sites")
bact_cca_plot_depth <- recordPlot()
 
#generate biplot of sites and species
plot(bact_finalmodel, scaling = "sites", type="n")
points(bact_finalmodel, "species", pch=1, col="forestgreen", cex=0.8, scaling = "sites")
text(bact_finalmodel, "species", col="black", cex=0.4, scaling = "sites")
```

## Archaea

### PERMANOVA

```{r}
#log transform
arc_physeq.t <- microbiome::transform(arc_phyloseq_rare,transform="log10")
#calculating bray curtis distances
arc_bray.dist<-phyloseq::distance(arc_physeq.t,"bray") 
arc_samp.data<-sample_data(arc_phyloseq_rare)

#modeling
arc_physeq.model.plot <-adonis2(arc_bray.dist~arc_samp.data$Plot); arc_physeq.model.plot
arc_physeq.model.depth <-adonis2(arc_bray.dist~arc_samp.data$Depth); arc_physeq.model.depth
arc_physeq.model.trans <-adonis2(arc_bray.dist~arc_samp.data$Transect); arc_physeq.model.trans

#check for homogeneity
dis <- vegdist(arc_bray.dist)
#plot
dispersion_model <- betadisper(dis, arc_samp.data$Plot)
permutest(dispersion_model, pairwise = TRUE)
plot(dispersion_model)
#transect
dispersion_model <- betadisper(dis, arc_samp.data$Transect)
permutest(dispersion_model, pairwise = TRUE)
plot(dispersion_model)
#depth
dispersion_model <- betadisper(dis, arc_samp.data$Depth)
permutest(dispersion_model, pairwise = TRUE)
plot(dispersion_model)
```

### NMDS w/ envfit

```{r}
arc_community <- arc_phyloseq_rare.df %>% select(c(1:3),) %>% pivot_wider(names_from = "OTU", values_from = "Abundance") %>% column_to_rownames("Sample")
arc_community.matrix <- as.matrix(arc_community)

#nmds code
set.seed(123)
arc_nmds = metaMDS(arc_community.matrix, distance = "bray"); arc_nmds

arc_en = envfit(arc_nmds, arc_env_data, permutations = 999, na.rm = TRUE)

plot(arc_nmds)
plot(arc_en)

#extract NMDS scores (x and y coordinates) for sites from newer versions of vegan package
arc_data.scores = as.data.frame(scores(arc_nmds)$sites)
arc_data.scores <- tibble::rownames_to_column(arc_data.scores, "Sample")
arc_data.scores <- arc_data.scores %>% dplyr::inner_join(arc_env_data) %>% select(c(1:6),) %>% column_to_rownames("Sample")

arc_en_coord_cont = as.data.frame(scores(arc_en, "vectors")) * ordiArrowMul(arc_en)
arc_en_coord_cat = as.data.frame(scores(arc_en, "factors")) * ordiArrowMul(arc_en)

arc_nmds_plot <- ggplot(data = arc_data.scores, aes(x = NMDS1, y = NMDS2)) + 
  stat_ellipse(aes(fill = Plot), geom = "polygon", alpha = 0.25, show.legend = FALSE) +
  geom_point(data = arc_data.scores, aes(fill = Plot, shape = Depth), size = 5, alpha = 1, show.legend = TRUE) + 
  scale_shape_manual(values = c(22, 21, 24, 25)) +
  scale_fill_manual(values = plot_colors)  + 
  scale_color_manual(values=plot_colors)  +
    # geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2), 
      # data = arc_en_coord_cont, alpha = 0.5) +
   #  geom_text_repel(data = arc_en_coord_cont, aes(x = NMDS1, y = NMDS2), 
      # fontface = "bold", label = row.names(arc_en_coord_cont), size = 8) + 
     theme(axis.title = element_text(size = 20), 
       panel.background = element_blank(), panel.border = element_rect(fill = NA), 
       axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), 
       legend.title = element_text(size = 25), 
       legend.text = element_text(size =20)) + 
  #guides(fill = guide_legend(title = "Group"), shape = guide_legend(title = "Group")) +
    labs(fill = "Plot") #+ theme(legend.position = "none")
arc_nmds_plot

arc_nmds_plot <- arc_nmds_plot + #theme(plot.background = element_rect(fill = "#C4E4B0")) +
  theme(axis.text = element_text(size=15), 
        axis.title=element_text(size=25)) +
  theme(panel.background = element_rect(fill = 'white', color = 'black'),
        panel.grid.major = element_line(size = 0.07, linetype = 'solid', colour = "black"))

tiff("Figures/Plot2E.tiff", width = 7, height = 4, units = 'in', res = 300)
arc_nmds_plot
dev.off()
```

### Richness

```{r}
arc_DIV <- estimate_richness(arc_phyloseq_rare); arc_DIV
summary(arc_DIV)

#join env data
arc_DIV <- rownames_to_column(arc_DIV) %>% dplyr::rename(Sample = rowname)

arc_plot <- arc_phyloseq_rare.df %>% select("Sample", "Transect", "Plot", "Depth") %>% unique()
rownames(arc_plot) <- NULL
arc_DIV <- arc_DIV %>% dplyr::right_join(arc_plot) %>% dplyr::rename(Archaea_Observed = Observed)
levels(arc_DIV$Depth)

#anova
arc_obvs_anova <- lmer((Archaea_Observed) ~ Plot+Depth + (1|Transect), data = arc_DIV)
check_model(arc_obvs_anova)
anova(arc_obvs_anova, type = 2)
emmeans(arc_obvs_anova, list(pairwise ~Plot), adjust = "tukey")
emmeans(arc_obvs_anova, list(pairwise ~Depth), adjust = "tukey")
emmeans(arc_obvs_anova, list(pairwise ~Plot*Depth), adjust = "tukey")
arc_obvs_emmeans <- emmeans(arc_obvs_anova, ~Plot*Depth)
arc_obvs_emmeans <- as.data.frame(arc_obvs_emmeans)
arc_obvs_emmeans_depth <- emmeans(arc_obvs_anova, ~Depth)
arc_obvs_emmeans_depth <- as.data.frame(arc_obvs_emmeans_depth)

# PLOT x DEPTH
arc_obvs_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = arc_obvs_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = arc_obvs_emmeans) +
  geom_jitter(aes(x = Plot, y = Archaea_Observed, color = Depth), width = 0.25, alpha = 0.5, data = arc_DIV) + 
  labs(y = "Observed richness", x = "Plot") + scale_color_manual(values = depth_colors) + theme_classic() + theme(axis.text = element_text(size=15), axis.title=element_text(size=20)) +
  theme(legend.position = "none")

tiff("Figures/PlotS2B.tiff", width = 6, height = 4, units = 'in', res = 500)
arc_obvs_plot
dev.off()

## DEPTH alone
ggplot() + geom_point(aes(x = Depth, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = arc_obvs_emmeans_depth) +
  geom_errorbar(aes(x = Depth, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = arc_obvs_emmeans_depth) +
  geom_jitter(aes(x = Depth, y = Archaea_Observed, color = Depth), width = 0.25, alpha = 0.5, data = arc_DIV) + 
  labs(y = "Observed archaeal richness", x = "Depth") + scale_color_manual(values = plot_colors) + theme_classic() +
  theme(legend.position = "none")
```

### Shannon's diversity

```{r}
#anova
arc_shannon_anova <- lmer((Shannon) ~ Plot*Depth + (1|Transect), data = arc_DIV)
check_model(arc_shannon_anova)
anova(arc_shannon_anova, type = 2)
arc_shannon_emmeans <- emmeans(arc_shannon_anova, ~Plot*Depth)
arc_shannon_emmeans <- as.data.frame(arc_shannon_emmeans)

# PLOT x DEPTH
arc_shannon_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = arc_shannon_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = arc_shannon_emmeans) +
  geom_jitter(aes(x = Plot, y = Shannon, color = Depth), width = 0.25, alpha = 0.5, data = arc_DIV) + 
  labs(y = "Archaea Shannon diversity", x = "Plot") + scale_color_manual(values = depth_colors) + theme_classic() +
  theme(legend.position = "none")
```

### Taxonomic composition

#### microshades

```{r}
#agglomerate, normalize, and melt the phyloseq object
mdf_object<-merge_samples(arc_phyloseq_filt, "Plot")
arc_mdf_prep<-prep_mdf(mdf_object,subgroup_level = "Family")

#arc_mdf_prep$ID <- arc_mdf_prep$Sample
#arc_mdf_prep$ID <-gsub(pattern = "TE16S", replacement = "", x = arc_mdf_prep$ID)
#arc_mdf_prep <- arc_mdf_prep %>%
  #mutate(newID = paste0(substr(ID, nchar(ID), nchar(ID)), 
                                 #substr(ID, 1, nchar(ID) - 1)))

arc_mdf_prep$newID <- arc_mdf_prep$Plot

#create a color object for the specified data
tail(names(sort(table(arc_mdf_prep$Phylum))), 6)

arc_color_obj <- create_color_dfs(arc_mdf_prep, group_level = "Phylum", subgroup_level = "Family", selected_groups=c('Halobacterota', 'Thermoplasmatota', 'Crenarchaeota'), cvd = FALSE)

#extract
arc_mdf <- arc_color_obj$mdf
arc_cdf <- arc_color_obj$cdf

arc_cdf <- color_reassign(arc_cdf, "Halobacterota", ("micro_orange"), group_level = "Phylum")
arc_cdf <- color_reassign(arc_cdf, "Thermoplasmatota", ("micro_brown"), group_level = "Phylum")
arc_cdf <- color_reassign(arc_cdf, "Crenarchaeota", "micro_cvd_blue", group_level = "Phylum")

#reorder
sample_order <- arc_mdf_prep$newID
color_reorder <- reorder_samples_by(arc_mdf, arc_cdf, sample_variable = "newID")

#extract
arc_mdf <- color_reorder$mdf
arc_mdf$Family <-gsub(pattern = "Unknown", replacement = "Other", x = arc_mdf$Family)
arc_cdf <- color_reorder$cdf

#plot
arc_microshades <- plot_microshades(arc_mdf, arc_cdf, x = "newID") + scale_y_continuous(labels = scales::percent, expand = expansion(0)) + theme_classic() +
  theme(legend.position = "none") +
  theme(plot.margin = margin(6,10,6,6)) +
  xlab("Plot") + #facet_grid(~Plot, scales = "free_x") +
  theme(axis.text = element_text(size=12), axis.title=element_text(size=15)) 

tiff("Figures/PlotS6B1.tiff", width = 5, height = 4, units = 'in', res = 300)
arc_microshades
dev.off()

#legend
arc_micro_legend <- custom_legend(arc_mdf, arc_cdf, group_level = "Phylum",
  subgroup_level = "Family",legend_key_size = 1, legend_text_size = 15,
                                 legend_orientation = "horizontal")

tiff("Figures/PlotS6B2.tiff", width = 10, height = 4, units = 'in', res = 300)
arc_micro_legend
dev.off()

center_legend <- plot_grid(NULL, arc_micro_legend, nrow=1, rel_widths=c(0.04, 0.96))

#final microshades plot
plot_grid(arc_microshades, arc_micro_legend, nrow = 2, rel_heights = c(1, 0.3))
```

#### by phyla (filtered, not rarefied - just replace arc_phyloseq_filt)

```{r}
#create ASV-TAX table
taxa_names(arc_phyloseq_rare) <- paste0("Seq", seq(ntaxa(arc_phyloseq_rare)))
arc_ASV <- data.frame(otu_table(arc_phyloseq_rare))
arc_TAX <- data.frame(tax_table(arc_phyloseq_rare))
arc_t_ASV <- t(arc_ASV)
arc_ASV_TAX <- merge(arc_t_ASV, arc_TAX, by = "row.names")

## create the combined OTU table plus taxonomy string metacoder needs

#read it in using phyloseq object
tax_table(arc_phyloseq_rare)<-tax_table(arc_phyloseq_rare)[,1:7]
arc_x1<-parse_phyloseq(arc_phyloseq_rare) #this takes ages

#calculate relative abundance
arc_x1$data$abund_data <- calc_obs_props(arc_x1, "otu_table")

#calculate total abundance by forest type
arc_x1$data$Plot<-calc_taxon_abund(arc_x1, "abund_data",groups=arc_x1$data$sample_data$Plot) #this takes a while

### make taxonomy plots
# make sure we use functions from correct package
transform <- microbiome::transform
# merge rare taxa to speed up examples
arc_pseq <- microbiome::transform(arc_phyloseq_filt, "compositional")
arc_pseq_phylum <- aggregate_rare(arc_pseq, level = "Phylum", detection = 1/100, prevalence = 2/100)

#create dataframe from phyloseq 
arc_df <- psmelt(arc_pseq_phylum)

### construct a stacked barplot showing relative abundance at 'Family' level
#expand color palettes to fit data
colourCount <- length(unique(arc_df$Phylum))
getPalette <- colorRampPalette(brewer.pal(9, "Paired"))

#filter taxa by Abundance + merge "Unknown and Other" groups
arc_df$Phylum[arc_df$Abundance < 0.02] <- "Taxa < 2% Abundant"
arc_df$Phylum[arc_df$Phylum == "Unknown" | arc_df$Phylum == "Other" | arc_df$Phylum == "Taxa < 2% Abundant"] <- "Taxa < 2% Abundant"

#change relevel <3% to be at the bottom
#arc_df$Phylum <- relevel(factor(arc_df$Phylum), ref = "Taxa < 2% Abundant")
arc_df$Phylum <- factor(arc_df$Phylum)
lapply(arc_df, levels) #check levels
arc_df$Phylum <- fct_relevel(arc_df$Phylum, "Taxa < 2% Abundant", after = Inf)

#plot the relative abundance by Phylum
arc_phylum_plot_facet <- ggplot(arc_df, aes(fill=Phylum, y=Abundance, x=Depth)) + geom_bar(position="fill", stat="identity") + theme_minimal() + facet_grid(~Plot) + 
  ylab("Relative Abundance") + theme(axis.text.x = element_text(angle=45)) + scale_y_continuous(labels = scales::percent) + scale_fill_manual(values = taxa_colors)

tiff("Figures/PlotS5B.tiff", width = 6, height = 4, units = 'in', res = 300)
arc_phylum_plot_facet
dev.off()
```

#### across family, across plot

```{r, fig.fullwidth=TRUE, fig.height=6, fig.width=15}
# merge rare taxa to speed up examples
arc_pseq <- microbiome::transform(arc_phyloseq_filt, "compositional")
arc_pseq_fam <- aggregate_rare(arc_pseq, level = "Family", detection = 1/100, prevalence = 2/100)

#create dataframe from phyloseq 
arc_df_fam <- psmelt(arc_pseq_fam)

#filter taxa by Abundance + merge "Unknown and Other" groups
arc_df_fam$Family[arc_df_fam$Abundance < 0.02] <- "Taxa < 2% Abundant"
unique(arc_df_fam$Family)
arc_df_fam$Family[arc_df_fam$Family == "Unknown" | arc_df_fam$Family == "Other" | arc_df_fam$Family == "Taxa < 2% Abundant"] <- "Taxa < 2% Abundant"
arc_df_fam$Family <- factor(arc_df_fam$Family)
lapply(arc_df_fam, levels) #check levels
arc_df_fam$Family <- fct_relevel(arc_df_fam$Family, "Taxa < 2% Abundant", after = Inf)

#plot the relative abundance by Family
fun_phylum_plot_facet <- ggplot(arc_df_fam, aes(fill=Family, y=Abundance, x=Depth)) + geom_bar(position="fill", stat="identity") + theme_minimal() + facet_grid(~Plot) + 
  ylab("Relative Abundance") + theme(axis.text.x = element_text(angle=45)) + scale_y_continuous(labels = scales::percent) + scale_fill_manual(values = taxa_colors)

##other way
#merge samples by category
arc_samples<-merge_samples(arc_phyloseq_rare, "Plot")
sample_data(arc_samples)$Plot = sample_names(arc_samples)
arc_samples@sam_data$Plot <- factor(arc_samples@sam_data$Plot, levels = c("30", "75", "250", "900", "1900"))

#aggregate samples by taxonomic rank (family here)
arc_family <- arc_samples %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at genus level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #transform to rel. abundance
  psmelt() %>%                                         #melt to long format
  filter(Abundance > 0.01) %>%                         #filter out low abundance taxa
  arrange(Abundance)                                     #sort data frame alphabetically by phylum

#make an other group
arc_family.plot = arc_family |>
  mutate(Family = if_else(Abundance < 0.05, "Others (< 5%)", Family))

arc_family.plot$Family <- as.factor(arc_family.plot$Family)
levels(arc_family.plot$Family)
arc_family.plot$Family = factor(arc_family.plot$Family, levels = c("Methanocellaceae", "Methanomassiliicoccaceae", "Nitrososphaeraceae", "Nitrosotaleaceae", "Others (< 5%)"))

arcgen_colors <- c("#DD5129", "#FAB255","#4c2f61",  "#70B8D0FF", "#D3D3D3")

#making the plot using ggplots
arc_fam_plot <- ggplot(arc_family.plot, aes(x = Plot, y = Abundance, fill = Family)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = arcgen_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Family > 1%) \n") +
  xlab("Plot") +
  ggtitle("Composition of Archaeal Families") + scale_y_continuous(labels = scales::percent) + theme_bw() + #theme(axis.text = element_text(size=15), 
      #  axis.title=element_text(size=15)) +
   theme(axis.title.y = element_text(hjust=0.5, vjust = -4)) + guides(fill=guide_legend(title="Archaeal Families"))
```

### FAPROTAX

```{r}
arc_asv.function.table <- read.table("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/FAPROTAX/functional_table.tsv", sep="\t", header=TRUE)
arc_rel.asv.fn.df <- data.frame(arc_asv.function.table)

arc_asv.melt <- melt(arc_rel.asv.fn.df, id.vars = "group")
arc_asv.melt <- subset(arc_asv.melt, value > 0)
arc_asv.melt$variable <- as.factor(arc_asv.melt$variable)
arc_asv.melt <- arc_asv.melt %>% dplyr::rename(Sample = variable)
arc_asv.melt <- arc_asv.melt %>% dplyr::right_join(arc_env_data)
arc_asv.melt$value <- as.numeric(arc_asv.melt$value)
```

#### across plots

```{r}
#bubble graph
ggplot(arc_asv.melt[which(arc_asv.melt$value > 0),], aes(x=Plot, y=group)) + geom_point(aes(colour= Depth, size = value)) + theme(axis.text.x = element_blank()) + ylab("") + theme_minimal() + xlab("")

#heat map
ggplot(arc_asv.melt[which(arc_asv.melt$value > 0),], aes(x=Plot, y=group, fill = value)) + geom_tile() + theme_minimal() + ylab("Function") + scale_color_gradientn(colors=met.brewer("VanGogh3"))
```

#### compositional bar graphs

```{r}
arc_biofunction_16S <- arc_asv.melt[!is.na(arc_asv.melt$group),] 
arc_biofunction_16S$group<-gsub("_"," ",as.character(arc_biofunction_16S$group))

# put small abund stuff tgt under others
arc_df.plot = arc_biofunction_16S |>
  mutate(group = if_else(value < 0.01, NA, group)) #making small abundance NA for plot

# put small abund stuff tgt under others
arc_df.plot <- arc_df.plot |>
  mutate(group = if_else(value < 0.05, "Other (< 5%)", group))

unique(arc_df.plot$group)

arc_df.plot$group <- factor(arc_df.plot$group)
lapply(arc_df.plot$Depth, levels) #check levels
arc_df.plot$group <- fct_relevel(arc_df.plot$group, "Other (< 5%)", after = Inf)
arc_df.plot$Depth = factor(arc_df.plot$Depth, levels = c("100-150", "50-100", "10-50", "0-10"))

#renaming groups
methanogenesis <- paste(c("methanogenesis by reduction of methyl compounds with H2", "methanogenesis", "hydrogenotrophic methanogenesis"), collapse = '|')
arc_df.plot <- arc_df.plot %>% mutate(group = gsub(methanogenesis, "Methanogenesis", group)) 
sulfur <- paste(c("dark oxidation of sulfur compounds", "dark sulfur oxidation", "respiration of sulfur compounds", "thiosulfate respiration"), collapse = '|')
arc_df.plot <- arc_df.plot %>% mutate(group = gsub(sulfur, "Sulfur respiration (anaerobic)", group)) 
nitrogen <- paste(c("nitrate reduction", "nitrate respiration", "nitrogen respiration"), collapse = '|')
arc_df.plot <- arc_df.plot %>% mutate(group = gsub(nitrogen, "Nitrogen respiration (anaerobic)", group)) 
arc_df.plot <- arc_df.plot %>% mutate(group = gsub("methylotrophy", "Methylotrophy", group)) %>% 
   mutate(group = gsub("chemoheterotrophy", "Chemoheterotrophy", group)) %>% 
  mutate(group = gsub("aerobic ammonia oxidation", "Aerobic ammonia oxidation", group)) %>% 
  mutate(group = gsub("nitrification", "Nitrification", group)) %>% 
  mutate(group = gsub("fermentation", "Fermentation", group))
arc_df.plot$group <- fct_relevel(arc_df.plot$group, "Other (< 5%)", after = Inf)
arc_df.plot_sum <- arc_df.plot %>%
  group_by(Transect, Plot, Depth, Sample, group) %>%  # Group by columns that should remain unchanged
  summarise(value = sum(value), .groups = "drop")  # Summarize the 'value' column by taking the sum

arc_func_colors <- c("#4E79A7FF", "#B07AA1FF","#F28E2BFF",  "#E15759FF",  "#F1CE63FF", "#499894FF", "#90D0E0FF",  "#a9845b",  "#D8D8E8FF")

#making the plot using ggplots
ggplot(arc_df.plot, aes(y = Depth, x = value, fill = group)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = arc_func_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  xlab("Relative Abundance (Function > 2%) \n") +
  ylab("Depth") +
  theme_minimal() + facet_grid(Plot~.) +
  ggtitle("Composition of Archaeal Functional Groups Along Degradation Transects in Brunei") + scale_x_continuous(labels = scales::percent) + # can change y axis to percentages (100%)
  theme_bw() + theme(axis.text.y = element_text(angle=45))

#PLOT only
arc_function_plot <- ggplot(arc_df.plot, aes(x = Plot, y = value, fill = group)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = arc_func_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Function > 1%) \n") +
  xlab("Plot") +
  #facet_grid(Transect~.) +
  theme(plot.title = element_blank()) +
  scale_y_continuous(labels = scales::percent) + # can change y axis to percentages (100%)
  theme_bw() + guides(fill=guide_legend(title="Biofunction")) + 
   theme(axis.text.x = element_text(margin = margin(t = 5))) +
  theme(axis.text = element_text(size=10), 
        axis.title=element_text(size=12)) + 
  theme(axis.title.y = element_text(hjust=0.5, vjust = -4)) +
   theme(legend.title = element_text(size = 12), 
       legend.text = element_text(size =20)) + theme(legend.position = "none")

tiff("Figures/Plot3E.tiff", width = 3, height = 4, units = 'in', res = 300)
arc_function_plot
dev.off()

#DEPTH only
ggplot(arc_df.plot, aes(y = Depth, x = value, fill = group)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = taxa_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  xlab("Relative Abundance (Function > 1.5%) \n") +
  ylab("Depth") +
  theme_minimal() + 
  #facet_grid(Transect~.) +
  ggtitle("Composition of Archaeal Functional Groups Along Degradation Transects in Brunei") + scale_x_continuous(labels = scales::percent) + # can change y axis to percentages (100%)
  theme_bw() #+ theme(axis.text.y = element_text(angle=45))
```


```{r}
#methane pathway
arc_methane_group <- arc_df.plot %>% filter(stringr::str_detect(group, "Meth"))
arc_methane_group$group <- factor(arc_methane_group$group)
arc_methane_group$Depth = factor(arc_methane_group$Depth, levels = c("0-10", "10-50", "50-100", "100-150"))

ggplot(arc_methane_group, aes(y = value, x = Plot, fill = group)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values=met.brewer("Hiroshige", 6)) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance") +
  xlab("Plot") +
  theme_minimal() + 
  ggtitle("Methane Dynamics Along Degradation Transects in Brunei") + scale_y_continuous(labels = scales::percent) +
  theme(axis.text.x = element_text(margin = margin(t = 10))) +
  theme(axis.text = element_text(size=12), 
        axis.title=element_text(size=15)) + 
   theme(legend.title = element_text(size = 15), 
       legend.text = element_text(size =12)) + guides(fill=guide_legend(title="Biofunction"))

#nitrogen pathway
arc_nitrogen_group <- arc_biofunction_16S %>% filter(stringr::str_detect(group, "nitr"))
arc_nitrogen_group$group <- factor(arc_nitrogen_group$group)
arc_nitrogen_group$Depth = factor(arc_nitrogen_group$Depth, levels = c("0-10", "10-50", "50-100", "100-150"))
unique(arc_nitrogen_group$group)
arc_nitrogen_group$group = factor(arc_nitrogen_group$group, levels = 
                                c("nitrification", "nitrate respiration", "nitrate reduction", 
                                  "nitrogen respiration"))

nitrogen_colors <- c("#134b73","#c2d6a4",  "#a9845b", "#1e3d14")

ggplot(arc_nitrogen_group, aes(y = value, x = Plot, fill = group)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values=nitrogen_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance") +
  xlab("Plot") +
  theme_minimal() +
  ggtitle("Nitrogen Dynamics Along Degradation Transects in Brunei") + scale_y_continuous(labels = scales::percent) +
  theme_bw() + theme(axis.text.x = element_text(margin = margin(t = 10))) +
  theme(axis.text = element_text(size=12), 
        axis.title=element_text(size=15)) + 
   theme(legend.title = element_text(size = 15), 
       legend.text = element_text(size =12)) + guides(fill=guide_legend(title="Biofunction"))
```

#### treemap

```{r}
arc_df.plot.df <- arc_df.plot %>%
  group_by(Plot, group) %>%
  summarize(Abundance=sum(value))

ggplot(arc_df.plot.df, aes(area = Abundance, label = group, fill=group, subgroup = group)) +
  geom_treemap() +
  geom_treemap_text(colour = "black", place = "topleft", reflow = T) +
  facet_wrap(~Plot) +
  geom_treemap_subgroup_border(colour="black", size = 2) + theme_classic() +
  ggtitle("Composition of Archaeal Functional Groups Across Degradation Transects in Brunei") +
   theme(plot.title = element_text(hjust = 0.5,size=10)) + scale_fill_manual(values = taxa_colors) +
   theme(axis.text = element_text(size=10), 
        axis.title=element_text(size=20))
```

### picrust2

#### pathways relative abund

```{r}
#import data (predicted pathways)
arc_pathways <- read_tsv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/picrust2-2.5.2/picrust2_out_pipeline/pathways_out/path_abun_unstrat.tsv") 
arc_pathways <- arc_pathways[, c(1:74)]
arc_pathways <- arc_pathways %>% pivot_longer(!pathway, names_to = "Sample", values_to = "abund")
arc_pathways$Sample <-gsub(pattern = "_S.*", replacement = "", x = arc_pathways$Sample)
arc_pathways <- arc_pathways %>% dplyr::left_join(arc_env_data)

#remove data w NA in group
arc_pathways$pathway<-gsub("_"," ",as.character(arc_pathways$pathway))
arc_picrust <- arc_pathways

#biofunction_16S_filtered <- subset(arc_picrust, abund>0.01) #see what people do for abund

arc_picrust.df <- arc_picrust %>% group_by(Sample) %>% mutate(relabund = abund/sum(abund))

# put small abund stuff tgt under others
arc_df.picrust.plot = arc_picrust.df |>
  mutate(pathway = if_else(relabund < 0.025, NA, pathway)) #making small abundance NA for plot

arc_df.picrust.plot$pathway <- factor(arc_df.picrust.plot$pathway)
unique(arc_df.picrust.plot$pathway)
arc_df.picrust.plot$pathway <- fct_relevel(arc_df.picrust.plot$pathway, "Other (< 2.5%)", after = Inf)
head(arc_df.picrust.plot)

#making the plot using ggplots
arc_df.picrust.plot <- arc_df.picrust.plot %>% subset(!is.na(pathway))

arc_metacyc_pathways <- read.csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/MetaCyc_Arc.csv") %>% dplyr::rename(pathway = Archaea)
arc_picrust_clean <- arc_df.picrust.plot %>% dplyr::left_join(arc_metacyc_pathways) 
arc_picrust_clean <- arc_picrust_clean %>% select(-1,) %>% dplyr::rename(Pathway = Re.label, Biofunction = Group)
head(arc_picrust_clean)
unique(arc_picrust$Plot)

#arc_picrust_clean$Pathway = factor(arc_picrust_clean$Pathway, levels = c(
  #"Amino acid biosynthesis", "L-isoleucine biosynthesis", "L-valine biosynthesis", "Aerobic respiration", 
  #"Isobutanol biosynthesis", "Gondoate biosynthesis (anaerobic)", "TCA cycle", "Fatty acid elongation", "GDP-mannose biosynthesis", #"cis-Vaccenate biosynthesis", "Pyrimidine nucleobases salvage I", "Romatic biogenic amine degradation", "Peptidoglycan maturation"
#))

#arc_pathway_colors <- c("#2d223c","#90719f","#dec5da","#92351e","#ecb27d", "#b9563f", "#fbe3c2","#0a3351","#72aeb6", "#134b73","#abc9c8","#2f70a1","#4692b0")

ggplot(data = arc_picrust_clean, aes(y = relabund, x = Plot, fill = Pathway)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = taxa_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Function > 2%) \n") +
  xlab("Plot") +
  theme_minimal() +
  ggtitle("Composition of Archaeal Functional Pathways Along Degradation Transects in Brunei") + scale_y_continuous(labels = scales::percent) + # can change y axis to percentages (100%)
  theme(axis.text.x = element_text(margin = margin(t = 5))) +
  theme(axis.text = element_text(size=12), 
        axis.title=element_text(size=15)) + 
  theme(axis.title.y = element_text(hjust=0.5, vjust = -4)) +
   theme(legend.title = element_text(size = 15), 
       legend.text = element_text(size =12))
```

#### ANOVAs for function

```{r}
head(arc_picrust_clean)
arc_function_bare_pathway <- arc_picrust_clean %>% dplyr:: select(Sample, Transect, Plot, Depth, relabund, Pathway) 
arc_function_bare_pathway <- arc_function_bare_pathway %>% group_by(Sample, Transect, Plot, Depth, Pathway) %>% mutate(totalrelabund = sum(relabund)) %>% select(-relabund) %>% unique() %>% pivot_wider(names_from = Pathway, values_from = totalrelabund)
head(arc_function_bare_pathway)
summary(arc_function_bare_pathway)
arc_function_bare_pathway[is.na(arc_function_bare_pathway)] <- 0

arc_function_bare_biofunction <- arc_picrust_clean %>% dplyr:: select(Sample, Transect, Plot, Depth, relabund, Biofunction) 
arc_function_bare_biofunction <- arc_function_bare_biofunction %>% group_by(Sample, Transect, Plot, Depth, Biofunction) %>% mutate(totalrelabund = sum(relabund)) %>% select(-relabund) %>% unique() %>% pivot_wider(names_from = Biofunction, values_from = totalrelabund)
arc_function_bare_biofunction[is.na(arc_function_bare_biofunction)] <- 0

head(arc_function_bare_biofunction)


#ANOVAs

hist((arc_function_bare_pathway$`Amino acid biosynthesis`))
##Amino acid biosynthesis (significant between 75/250 and 900 - higher in 900)
function_anova <- glmmTMB((`Amino acid biosynthesis`) ~ Plot+Depth + (1|Transect), data = arc_function_bare_pathway, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist(log(arc_function_bare_pathway$`L-isoleucine biosynthesis`))
##L-isoleucine biosynthesis (only diff between 250-900 - higher in 900)
function_anova <- lmer(log(`L-isoleucine biosynthesis`) ~ Plot+Depth + (1|Transect), data = arc_function_bare_pathway)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

hist(log(arc_function_bare_pathway$`Reverse TCA`))
##Reverse TCA
function_anova <- glmmTMB((`Reverse TCA`) ~ Plot+Depth + (1|Transect), data = arc_function_bare_pathway, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist(log(arc_function_bare_pathway$`Aerobic respiration`))
##Aerobic respiration
function_anova <- glmmTMB((`Aerobic respiration`) ~ Plot+Depth + (1|Transect), data = arc_function_bare_pathway, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist(log(arc_function_bare_pathway$`Fermentation`))
##Fermentation
function_anova <- glmmTMB((`Fermentation`) ~ Plot+Depth + (1|Transect), data = arc_function_bare_pathway, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist(log(arc_function_bare_pathway$`L-valine biosynthesis`))
##L-valine biosynthesis (significant between 75 and 900/1900 and 250 and 900 - lower in 75/250 and higher in 900/1900)
function_anova <- glmmTMB((`L-valine biosynthesis`) ~ Plot+Depth + (1|Transect), data = arc_function_bare_pathway, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

hist((arc_function_bare_pathway$`Amino acid biosynthesis`))
##Amino acid biosynthesis (only sig between 250 and 900 - higher in 900)
function_anova <- lmer((`Amino acid biosynthesis`) ~ Plot+Depth + (1|Transect), data = arc_function_bare_biofunction)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

hist((arc_function_bare_pathway$`Amino acid biosynthesis`))
##Energy production (only sig between 30 and 900 - higher in 900)
function_anova <- lmer((`Energy production`) ~ Plot+Depth + (1|Transect), data = arc_function_bare_biofunction)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")
```

#### manually running ggpicrust2

##### KEGG pathway

```{r, eval = FALSE}
#load metadata
arc_metadata <- as_tibble(arc_env_data)

#load KEGG pathway abundance
arc_kegg_abundance <- ko2kegg_abundance("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/picrust2-2.5.2/picrust2_out_pipeline/KO_metagenome_out/pred_metagenome_unstrat.tsv") 
arc_kegg_abundance <- arc_kegg_abundance[1:73]
names(arc_kegg_abundance) = gsub(pattern = "_S.*", replacement = "", x = names(arc_kegg_abundance))

#pathway differential abundance analysis (DAA) using DESeq2 (multiple factors)
arc_daa_results <- pathway_daa(abundance = arc_kegg_abundance, metadata = arc_metadata, group = "Plot", daa_method = "DESeq2", select = NULL, reference = "1900") 

#filter results
unique(arc_daa_results$method)
arc_daa_results <- arc_daa_results[arc_daa_results$method == "DESeq2", ]
arc_daa_sig_results <- arc_daa_results[arc_daa_results$p_adjust< 0.05, ]
arc_daa_sig_results <- arc_daa_sig_results %>% drop_na(feature)

#split for practical reasons
daa_sig_results1 <- arc_daa_sig_results[1:75,]
daa_sig_results2 <- arc_daa_sig_results[76:150,]
daa_sig_results3 <- arc_daa_sig_results[151:227,]

#annotate pathway results using KO to KEGG conversion
daa_annotated_sig_results1 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results1, ko_to_kegg = TRUE)
daa_annotated_sig_results2 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results2, ko_to_kegg = TRUE)
daa_annotated_sig_results3 <- pathway_annotation(pathway = "KO", daa_results_df = daa_sig_results3, ko_to_kegg = TRUE)

#bind full dataset
arc_daa_results_full <- rbind(daa_annotated_sig_results1, daa_annotated_sig_results2, 
                          daa_annotated_sig_results3)
arc_daa_results_full <- arc_daa_results_full[!is.na(arc_daa_results_full$pathway_name),]
#choose levels to compare
arc_daa_results_30 <- arc_daa_results_full[arc_daa_results_full$group1 == "30", ]
arc_daa_results_30 <- arc_daa_results_30[arc_daa_results_30$group2 == "1900", ]

#filter
arc_daa_results_30$p_adjust <- round(arc_daa_results_30$p_adjust,5)
arc_kegg <- arc_kegg_abundance
arc_kegg$avgabund <- rowMeans(arc_kegg, na.rm = TRUE)
arc_kegg <- arc_kegg %>% rownames_to_column() %>% select(rowname, avgabund) %>% dplyr::rename(feature = rowname)
arc_daa_results_30 <- arc_daa_results_30 %>% dplyr::left_join(arc_kegg)

arc_low_p_feature <- arc_daa_results_30[order(arc_daa_results_30$avgabund, decreasing = TRUE),]$feature[1:20]
```

##### pathway error bar plot

```{r, eval = FALSE}
ggpicrust2::pathway_errorbar(abundance = arc_kegg_abundance, daa_results_df = arc_daa_results_30, 
                             Group = arc_metadata$Plot, ko_to_kegg = TRUE, p_values_threshold = 0.05, 
                             order = "pathway_class", select = arc_low_p_feature, p_value_bar = TRUE, 
                             colors = NULL, x_lab = "pathway_name")
```

##### EC pathway error barplot

```{r, eval = FALSE}
arc_ec_abundance <- read_tsv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/picrust2-2.5.2/picrust2_out_pipeline/EC_metagenome_out/pred_metagenome_unstrat.tsv")
arc_ec_abundance <- arc_ec_abundance[1:74]
names(arc_ec_abundance) = gsub(pattern = "_S.*", replacement = "", x = names(arc_ec_abundance))

#DAA
arc_ec_daa_results <- pathway_daa(abundance = arc_ec_abundance %>% column_to_rownames("function"), metadata = arc_metadata, group = "Plot", daa_method = "DESeq2")

#annotate pathway results without KO to KEGG conversion
arc_ec_daa_annotated_results <- pathway_annotation(pathway = "EC", daa_results_df = arc_ec_daa_results, ko_to_kegg = FALSE)
#filter
arc_ec_daa_annotated_results <- arc_ec_daa_annotated_results[!is.na(arc_ec_daa_annotated_results$feature),]
arc_ec_daa_annotated_results <- arc_ec_daa_annotated_results[arc_ec_daa_annotated_results$group1 == "30", ]
arc_ec_daa_annotated_results <- arc_ec_daa_annotated_results[arc_ec_daa_annotated_results$group2 == "1900", ]
arc_ec_daa_annotated_results$p_adjust <- round(arc_ec_daa_annotated_results$p_adjust,5)
arc_low_p_feature_ec <- arc_ec_daa_annotated_results[order(arc_ec_daa_annotated_results$p_adjust), ]$feature[1:20]

#pathway error bar plot
pathway_errorbar(abundance = arc_ec_abundance %>% column_to_rownames("function"), daa_results_df = arc_ec_daa_annotated_results, Group = arc_metadata$Plot, ko_to_kegg = FALSE, p_values_threshold = 0.1, order = "group", select = arc_low_p_feature_ec, p_value_bar = TRUE, colors = NULL, x_lab = "description")
```

##### MetaCyc pathway analysis

```{r, eval = FALSE}
#load MetaCyc pathway abundance and metadata
arc_metacyc_abundance <- read_tsv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/picrust2-2.5.2/picrust2_out_pipeline/pathways_out/path_abun_unstrat.tsv")
arc_metacyc_abundance <- arc_metacyc_abundance[1:74]
names(arc_metacyc_abundance) = gsub(pattern = "_S.*", replacement = "", x = names(arc_metacyc_abundance))

#pathway DAA using DESeq2 method
arc_metacyc_daa_results <- pathway_daa(abundance = arc_metacyc_abundance %>% column_to_rownames("pathway"), metadata = arc_metadata, group = "Plot", daa_method = "DESeq2")

#annotate MetaCyc pathway results without KO to KEGG conversion
arc_metacyc_daa_annotated_results <- pathway_annotation(pathway = "MetaCyc", daa_results_df = arc_metacyc_daa_results, ko_to_kegg = FALSE)

arc_metacyc_daa_annotated_results <- arc_metacyc_daa_annotated_results[!is.na(arc_metacyc_daa_annotated_results$feature),]
arc_metacyc_daa_annotated_results <- arc_metacyc_daa_annotated_results[arc_metacyc_daa_annotated_results$group1 == "30", ]
arc_metacyc_daa_annotated_results <- arc_metacyc_daa_annotated_results[arc_metacyc_daa_annotated_results$group2 == "1900", ]
arc_metacyc_daa_annotated_results$p_adjust <- round(arc_metacyc_daa_annotated_results$p_adjust,5)
arc_low_p_feature_metacyc <- arc_metacyc_daa_annotated_results[order(arc_metacyc_daa_annotated_results$p_adjust), ]$feature[1:20]

arc_metacyc_abundance_subset <- arc_metacyc_abundance[1:20,]

#pathway heatmap
pathway_heatmap(abundance = arc_metacyc_abundance_subset %>% filter(pathway %in% arc_metacyc_abundance_subset$pathway) %>% column_to_rownames("pathway"), metadata = arc_metadata, group = "Plot")

#pathway error bar plot
pathway_errorbar(abundance = arc_metacyc_abundance %>% column_to_rownames("pathway"), daa_results_df = arc_metacyc_daa_annotated_results, Group = arc_metadata$Plot, ko_to_kegg = FALSE, p_values_threshold = 0.5, order = "p_values", select = arc_low_p_feature_metacyc, p_value_bar = TRUE, colors = NULL, x_lab = "description")
```

##### picrust2 PCA

```{r, eval = FALSE}
#pathway PCA plot
arc_abundance <- arc_metacyc_abundance %>% column_to_rownames("pathway") %>% filter(rowSums(. != 0) > 0)

#plot
pathway_pca(abundance = arc_abundance, 
            metadata = arc_metadata, group = "Plot")
```


### Major shifts

```{r}
#richness and shannon div
arc_DIV_merge <- arc_DIV %>% dplyr::select(Sample, Archaea_Observed, Shannon, Transect, Plot, Depth)

arc_pseq_phylum <- aggregate_rare(arc_pseq, level = "Phylum", detection = 1/100, prevalence = 2/100)
#create dataframe from phyloseq 
arc_df_phylum <- psmelt(arc_pseq_phylum)

#abundance of major phyla
arc_df_merge <- arc_df_phylum %>% dplyr::select(Sample, Phylum, Abundance, Transect, Plot, Depth)
unique(arc_df_merge$Phylum)
arc_phyla_list <- paste(c("Crenarchaeota", "Halobacterota", "Thermoplasmatota"), collapse = '|')
arc_df_merge <- arc_df_merge %>% filter(stringr::str_detect(Phylum, arc_phyla_list))
arc_df_merge <- arc_df_merge %>% group_by(Phylum) %>%
  pivot_wider(names_from = Phylum, values_from = Abundance)

#family level
arc_fam_merge <- arc_df_fam %>% dplyr::select(Sample, Family, Abundance, Transect, Plot, Depth)
unique(arc_fam_merge$Family)
arc_fam_list <- paste(c("Nitrosotaleaceae", "Methanomassiliicoccaceae", "Methanocellaceae", "Nitrososphaeraceae"), collapse = '|')
arc_fam_merge <- arc_fam_merge %>% filter(stringr::str_detect(Family, arc_fam_list))
arc_fam_merge <- arc_fam_merge %>% group_by(Family) %>%
  pivot_wider(names_from = Family, values_from = Abundance)

#abundance of major functions

#abundance of functional grps (faprotax)
arc_func_merge <- arc_df.plot %>% dplyr::select(Sample, group, value, Transect, Plot, Depth)
unique(arc_func_merge$group)
arc_func_list <- paste(c("Methanogenesis", "Methylotrophy", "Nitrification", "Chemoheterotrophy", "Aerobic ammonia oxidation"), collapse = '|')
arc_func_merge <- arc_func_merge %>% filter(stringr::str_detect(group, arc_func_list))
arc_func_merge <- unique(arc_func_merge)
arc_func_merge <- arc_func_merge %>% group_by(Sample, Transect, Plot, Depth, group) %>% mutate(totalrelabund = sum(value)) %>% select(-value) %>% unique() %>% pivot_wider(names_from = group, values_from = totalrelabund)
arc_func_merge <- arc_func_merge %>% dplyr::rename(methanogenesis_arc = Methanogenesis, 
                                       methylotrophy_arc = Methylotrophy, 
                                       chemoheterotrophy_arc = Chemoheterotrophy, 
                                       nitrification_arc = Nitrification, 
                                       aerobic_ammonia_oxidation_arc = `Aerobic ammonia oxidation`)

arc_major_merge <- arc_DIV_merge %>% dplyr::full_join(arc_df_merge) %>% dplyr::full_join(arc_fam_merge) %>% dplyr::full_join(arc_func_merge)
arc_major_merge <- arc_major_merge %>% dplyr::rename(Archaea_Shannon = Shannon) %>% 
  dplyr::select(-Sample)

#make datasets
#phyla
arc_df_merge_sum <- arc_df_merge %>% group_by(Plot) %>% summarise_if(is.numeric, mean, na.rm = TRUE)
arc_df_merge_sum <- arc_df_merge_sum %>% pivot_longer(!c(Plot), names_to = "Group", values_to = "Abundance")
arc_df_merge_sum$Plot <- as.numeric(arc_df_merge_sum$Plot)
arc_df_merge_sum$Group <- factor(arc_df_merge_sum$Group, levels = c("Crenarchaeota", "Halobacterota", "Thermoplasmatota"))
#arc_df_merge_sum <- arc_df_merge_sum %>% mutate(Plot = as.numeric(as.character(Plot)))

arc_df_merge_new <- arc_df_merge %>% dplyr::select(-Sample) %>% pivot_longer(!c(Plot, Transect, Depth), names_to = "Group", values_to = "Abundance")
#arc_df_merge_new <- arc_df_merge_new %>% mutate(Plot = as.numeric(as.character(Plot)))
arc_df_merge_new$Plot <- as.numeric(arc_df_merge_new$Plot)
arc_df_merge_new$Group <- factor(arc_df_merge_new$Group, levels = c("Crenarchaeota", "Halobacterota", "Thermoplasmatota"))

#family
arc_fam_merge_sum <- arc_fam_merge %>% group_by(Plot) %>% summarise_if(is.numeric, mean, na.rm = TRUE)
arc_fam_merge_sum <- arc_fam_merge_sum %>% pivot_longer(!c(Plot), names_to = "Group", values_to = "Abundance")
arc_fam_merge_sum$Plot <- as.numeric(arc_fam_merge_sum$Plot)
arc_fam_merge_sum$Group <- factor(arc_fam_merge_sum$Group, levels = c("Nitrosotaleaceae", "Methanomassiliicoccaceae", "Methanocellaceae", "Nitrososphaeraceae"))
#arc_fam_merge_sum <- arc_fam_merge_sum %>% mutate(Plot = as.numeric(as.character(Plot)))

arc_fam_merge_new <- arc_fam_merge %>% dplyr::select(-Sample) %>% pivot_longer(!c(Plot, Transect, Depth), names_to = "Group", values_to = "Abundance")
#arc_fam_merge_new <- arc_fam_merge_new %>% mutate(Plot = as.numeric(as.character(Plot)))
arc_fam_merge_new$Plot <- as.numeric(arc_fam_merge_new$Plot)
arc_fam_merge_new$Group <- factor(arc_fam_merge_new$Group, levels = c("Nitrosotaleaceae", "Methanomassiliicoccaceae", "Methanocellaceae", "Nitrososphaeraceae"))

arc_taxshift_plot <- ggplot() + theme_classic() + 
  geom_line(aes(x = Plot, y = Abundance, color = Group), data = arc_df_merge_sum) + 
  geom_point(aes(x = Plot, y = Abundance, color = Group), size = 6, data = arc_df_merge_sum) +
  geom_jitter(aes(x = Plot, y = Abundance, color = Group), 
             alpha = 0.5, width = 0.15, data = arc_df_merge_new) + 
  #geom_line(aes(x = Plot, y = Abundance, color = Group), linetype="dotted", data = arc_fam_merge_sum) + 
  #geom_point(aes(x = Plot, y = Abundance, color = Group), size = 2, data = arc_fam_merge_sum) +
  ylab("Relative Abundance") + scale_color_manual(values = c("#DD5129", "#FAB255", "#0F7BA2","#4c2f61", "#DD5129", "#0F7BA2","#FAB255"))

tiff("Figures/Plot3B.tiff", width = 6, height = 4, units = 'in', res = 300)
arc_taxshift_plot
dev.off()
```

### CCA
 
#### Prep data
 
```{r, eval = FALSE}
arc_community_cca <- arc_community[!(row.names(arc_community) %in% c("ArcTE16SBED","ArcTE16SCAC", "ArcTE16SECD", "ArcTE16SCCC","ArcTE16SDAD", "ArcTE16SDEC", "ArcTE16SAAC", "ArcTE16SDAA", "ArcTE16SDDA", "ArcTE16SDDB", "ArcTE16SEBA")),]

arc_env_cca <- arc_env_data %>% tibble::column_to_rownames("Sample") %>% dplyr::select(-Transect, -Plot, -Depth)
arc_env_cca <- arc_env_cca[!(row.names(arc_env_cca) %in% c("ArcTE16SBED","ArcTE16SCAC", "ArcTE16SECD", "ArcTE16SCCC","ArcTE16SDAD", "ArcTE16SDEC", "ArcTE16SAAC", "ArcTE16SDAA", "ArcTE16SDDA", "ArcTE16SDDB", "ArcTE16SEBA")),]
 
#Hellinger transformation to species matrix due to rare sp.
arc_community_trans <- decostand(arc_community_cca, "hellinger")
#scale and center soil parameters due to diff units
arc_env_trans <- decostand(arc_env_cca, method = "standardize")
#variables are now centered around a mean of 0
round(apply(arc_env_trans, 2, mean), 1)
#and scaled to have a standard deviation of 1
apply(arc_env_trans, 2, sd)
```
 
#### Run CCA
 
```{r, eval = FALSE}
#run CCA
arc_CCA <- cca(arc_community_trans~., arc_env_trans)
plot(arc_CCA) #first visualisation
summary(arc_CCA) #look at model
 
#stepwise model
arc_finalmodel<- ordistep(arc_CCA, scope=formula(arc_CCA))
vif<-vif.cca(arc_finalmodel)
vif #go to original dataset and remove redundant terms (VIF > 10)
arc_finalmodel
#18.8% inertia explained
plot(arc_finalmodel)
capture.output(vif, file = "vif.txt")
summary(arc_finalmodel)
screeplot(arc_finalmodel)
 
#significance tests
anova.cca(arc_finalmodel)
anova.cca(arc_finalmodel, by="terms")
anova.cca(arc_finalmodel, by="axis")
 
#colors for FT
arc_env_data$color <- arc_env_data$Plot %>% as.character()
arc_env_data$color[arc_env_data$color == "30"] <- "#74c8c3"
arc_env_data$color[arc_env_data$color == "75"] <- "#5a97c1" 
arc_env_data$color[arc_env_data$color == "250"] <- "#0a2e57"
arc_env_data$color[arc_env_data$color == "900"] <- "#d8d97a"
arc_env_data$color[arc_env_data$color == "1900"] <- "#669d62"
summary(arc_env_data)
 
#generate biplot for sites and soil var
plot(c(-2.5, 2), c(-1.5, 1.5), xlab="CCA1 (41.8%)", ylab="CCA2 (33.9%)", type="n")
abline(h = 0, v = 0, lty = 2, lwd =.5)
points(arc_finalmodel, dis="lc", pch=21, col="black", bg=arc_env_data$color, cex=1.5, scaling = "sites")
text(arc_finalmodel, dis="cn", cex = 1.5, scaling = "sites")
arc_cca_plot <- recordPlot()
 
#generate biplot of sites and species
plot(arc_finalmodel, scaling = "sites", type="n")
points(arc_finalmodel, "species", pch=1, col="forestgreen", cex=0.8, scaling = "sites")
text(arc_finalmodel, "species", col="black", cex=0.4, scaling = "sites")

#colors for depth
arc_env_data$color <- arc_env_data$Depth %>% as.character()
arc_env_data$color[arc_env_data$color == "0-10"] <- "#e7e5cc"
arc_env_data$color[arc_env_data$color == "10-50"] <- "#c2d6a4" 
arc_env_data$color[arc_env_data$color == "50-100"] <- "#669d62"
arc_env_data$color[arc_env_data$color == "100-150"] <- "#192813"
summary(arc_env_data)
 
#generate biplot for sites and soil var
plot(c(-2.5, 2), c(-1.5, 1.5), xlab="CCA1 (41.8%)", ylab="CCA2 (33.9%)", type="n")
abline(h = 0, v = 0, lty = 2, lwd =.5)
points(arc_finalmodel, dis="lc", pch=21, col="black", bg=arc_env_data$color, cex=1.5, scaling = "sites")
text(arc_finalmodel, dis="cn", cex = 1.5, scaling = "sites")
arc_cca_plot_depth <- recordPlot()
 
#generate biplot of sites and species
plot(arc_finalmodel, scaling = "sites", type="n")
points(arc_finalmodel, "species", pch=1, col="forestgreen", cex=0.8, scaling = "sites")
text(arc_finalmodel, "species", col="black", cex=0.4, scaling = "sites")
```

## Fungi
### PERMANOVA

```{r}
#log transform
fun_physeq.t <- microbiome::transform(fun_phyloseq_rare,transform="log10")
#calculating bray curtis distances
fun_bray.dist<-phyloseq::distance(fun_physeq.t,"bray") 
fun_samp.data<-sample_data(fun_phyloseq_rare)

#modeling
fun_physeq.model.plot <-adonis2(fun_bray.dist~fun_samp.data$Plot); fun_physeq.model.plot
fun_physeq.model.depth <-adonis2(fun_bray.dist~fun_samp.data$Depth); fun_physeq.model.depth
fun_physeq.model.trans <-adonis2(fun_bray.dist~fun_samp.data$Transect); fun_physeq.model.trans

#check for homogeneity
dis <- vegdist(fun_bray.dist)
#plot
dispersion_model <- betadisper(dis, fun_samp.data$Plot)
permutest(dispersion_model, pairwise = TRUE)
plot(dispersion_model)
#transect
dispersion_model <- betadisper(dis, fun_samp.data$Transect)
permutest(dispersion_model, pairwise = TRUE)
plot(dispersion_model)
#depth
dispersion_model <- betadisper(dis, fun_samp.data$Depth)
permutest(dispersion_model, pairwise = TRUE)
plot(dispersion_model)
```

### NMDS w/ envfit

```{r}
fun_community <- fun_phyloseq_rare.df %>% select(c(1:3),) %>% pivot_wider(names_from = "OTU", values_from = "Abundance") %>% column_to_rownames("Sample")
fun_community.matrix <- as.matrix(fun_community)

#nmds code
set.seed(123)
fun_nmds = metaMDS(fun_community.matrix, distance = "bray"); fun_nmds

fun_en = envfit(fun_nmds, fun_env_data, permutations = 999, na.rm = TRUE)

plot(fun_nmds)
plot(fun_en)

#extract NMDS scores (x and y coordinates) for sites from newer versions of vegan package
fun_data.scores = as.data.frame(scores(fun_nmds)$sites)
fun_data.scores <- tibble::rownames_to_column(fun_data.scores, "Sample")
fun_data.scores <- fun_data.scores %>% dplyr::inner_join(fun_env_data) %>% select(c(1:6),) %>% column_to_rownames("Sample")

fun_en_coord_cont = as.data.frame(scores(fun_en, "vectors")) * ordiArrowMul(fun_en)
fun_en_coord_cat = as.data.frame(scores(fun_en, "factors")) * ordiArrowMul(fun_en)

fun_nmds_plot <- ggplot(data = fun_data.scores, aes(x = NMDS1, y = NMDS2)) + 
  stat_ellipse(aes(fill = Plot), geom = "polygon", alpha = 0.25, show.legend = FALSE) +
  geom_point(data = fun_data.scores, aes(fill = Plot, shape = Depth), size = 5, alpha = 1, show.legend = TRUE) + 
  scale_shape_manual(values = c(22, 21, 24, 25)) +
  scale_fill_manual(values = plot_colors)  + 
  scale_color_manual(values=plot_colors)  +
    # geom_segment(aes(x = 0, y = 0, xend = NMDS1, yend = NMDS2), 
      # data = fun_en_coord_cont, alpha = 0.5) +
   #  geom_text_repel(data = fun_en_coord_cont, aes(x = NMDS1, y = NMDS2), 
      # fontface = "bold", label = row.names(fun_en_coord_cont), size = 8) + 
     theme(axis.title = element_text(size = 20), 
       panel.background = element_blank(), panel.border = element_rect(fill = NA), 
       axis.ticks = element_blank(), axis.text = element_blank(), legend.key = element_blank(), 
       legend.title = element_text(size = 25), 
       legend.text = element_text(size =20)) + 
  #guides(fill = guide_legend(title = "Group"), shape = guide_legend(title = "Group")) +
    labs(fill = "Plot") #+ theme(legend.position = "none")
fun_nmds_plot

fun_nmds_plot <- fun_nmds_plot + #theme(plot.background = element_rect(fill = "#C4E4B0")) +
  theme(axis.text = element_text(size=15), 
        axis.title=element_text(size=25)) +
  theme(panel.background = element_rect(fill = 'white', color = 'black'),
        panel.grid.major = element_line(size = 0.07, linetype = 'solid', colour = "black"))

tiff("Figures/Plot2F.tiff", width = 7, height = 4, units = 'in', res = 300)
fun_nmds_plot
dev.off()
```

### Richness

```{r}
fun_DIV <- estimate_richness(fun_phyloseq_rare); fun_DIV
summary(fun_DIV)

#join env data
fun_DIV <- rownames_to_column(fun_DIV) %>% dplyr::rename(Sample = rowname)

fun_plot <- fun_phyloseq_rare.df %>% select("Sample", "Transect", "Plot", "Depth") %>% unique()
rownames(fun_plot) <- NULL
fun_DIV <- fun_DIV %>% dplyr::right_join(fun_plot) %>% dplyr::rename(Fungi_Observed = Observed)
levels(fun_DIV$Depth)

#anova
fun_obvs_anova <- lmer((Fungi_Observed) ~ Plot*Depth + (1|Transect), data = fun_DIV)
check_model(fun_obvs_anova)
anova(fun_obvs_anova, type = 2)
emmeans(fun_obvs_anova, list(pairwise ~Plot*Depth), adjust = "tukey")
fun_obvs_emmeans <- emmeans(fun_obvs_anova, ~Plot*Depth)
fun_obvs_emmeans <- as.data.frame(fun_obvs_emmeans)

# PLOT x DEPTH
fun_obvs_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = fun_obvs_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = fun_obvs_emmeans) +
  geom_jitter(aes(x = Plot, y = Fungi_Observed, color = Depth), width = 0.25, alpha = 0.5, data = fun_DIV) + 
  labs(y = "Observed richness", x = "Plot") + scale_color_manual(values = depth_colors) + theme_classic() + theme(axis.text = element_text(size=15), axis.title=element_text(size=20)) + theme(legend.position = "none")

tiff("Figures/PlotS2C.tiff", width = 6, height = 4, units = 'in', res = 500)
fun_obvs_plot
dev.off()
```

### Shannon's diversity

```{r}
#anova
fun_shannon_anova <- lmer((Shannon) ~ Plot*Depth + (1|Transect), data = fun_DIV)
check_model(fun_shannon_anova)
anova(fun_shannon_anova, type = 2)
fun_shannon_emmeans <- emmeans(fun_shannon_anova, ~Depth)
fun_shannon_emmeans <- as.data.frame(fun_shannon_emmeans)

# DEPTH alone
fun_shannon_plot <- ggplot() + geom_point(aes(x = Depth, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = fun_shannon_emmeans) +
  geom_errorbar(aes(x = Depth, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = fun_shannon_emmeans) +
  geom_jitter(aes(x = Depth, y = Shannon, color = Depth), width = 0.25, alpha = 0.5, data = fun_DIV) + 
  labs(y = "Fungi Shannon diversity", x = "Depth") + scale_color_manual(values = depth_colors) + theme_classic() +
  theme(legend.position = "none")
```

### Taxonomic composition

#### microshades

```{r}
#agglomerate, normalize, and melt the phyloseq object
mdf_object<-merge_samples(fun_phyloseq_filt, "Plot")
fun_mdf_prep<-prep_mdf(mdf_object,subgroup_level = "Family")

#fun_mdf_prep$ID <- fun_mdf_prep$Sample
#fun_mdf_prep$ID <-gsub(pattern = "TEITS", replacement = "", x = fun_mdf_prep$ID)
#fun_mdf_prep <- fun_mdf_prep %>%
  #mutate(newID = paste0(substr(ID, nchar(ID), nchar(ID)), 
                                 #substr(ID, 1, nchar(ID) - 1)))

fun_mdf_prep$newID <- fun_mdf_prep$Plot

#create a color object for the specified data
tail(names(sort(table(fun_mdf_prep$Phylum))), 6)

fun_color_obj <- create_color_dfs(fun_mdf_prep, group_level = "Phylum", subgroup_level = "Family", selected_groups=c('Basidiomycota', 'Ascomycota'), cvd = FALSE)

#extract
fun_mdf <- fun_color_obj$mdf
fun_cdf <- fun_color_obj$cdf

fun_cdf <- color_reassign(fun_cdf, "Basidiomycota", "micro_blue", group_level = "Phylum")
fun_cdf <- color_reassign(fun_cdf, "Ascomycota", "micro_cvd_orange", group_level = "Phylum")

#reorder
sample_order <- fun_mdf_prep$newID
color_reorder <- reorder_samples_by(fun_mdf, fun_cdf, sample_variable = "newID")

#extract
fun_mdf <- color_reorder$mdf
fun_mdf$Family <-gsub(pattern = "Unknown", replacement = "Other", x = fun_mdf$Family)
fun_cdf <- color_reorder$cdf

fun_cdf[fun_cdf=="#eff3ff"] <- "#134b73"

#plot
fun_microshades <- plot_microshades(fun_mdf, fun_cdf, x = "newID") + scale_y_continuous(labels = scales::percent, expand = expansion(0)) + theme_classic() +
  theme(legend.position = "none") +
  theme(plot.margin = margin(6,10,6,6)) +
  xlab("Plot") + #facet_grid(~Plot, scales = "free_x") +
  theme(axis.text = element_text(size=12), axis.title=element_text(size=15)) 

tiff("Figures/PlotS6C1.tiff", width = 5, height = 4, units = 'in', res = 300)
fun_microshades
dev.off()

#legend
fun_micro_legend <- custom_legend(fun_mdf, fun_cdf, group_level = "Phylum",
  subgroup_level = "Family",legend_key_size = 1, legend_text_size = 15,
                                 legend_orientation = "horizontal")

tiff("Figures/PlotS6C2.tiff", width = 10, height = 4, units = 'in', res = 300)
fun_micro_legend
dev.off()

center_legend <- plot_grid(NULL, fun_micro_legend, nrow=1, rel_widths=c(0.04, 0.96))

#final microshades plot
plot_grid(fun_microshades, fun_micro_legend, nrow = 2, rel_heights = c(1, 0.3))
```

#### by phyla (filtered, not rarefied - just replace fun_phyloseq_filt)

```{r}
#create ASV-TAX table
taxa_names(fun_phyloseq_rare) <- paste0("Seq", seq(ntaxa(fun_phyloseq_rare)))
fun_ASV <- data.frame(otu_table(fun_phyloseq_rare))
fun_TAX <- data.frame(tax_table(fun_phyloseq_rare))
fun_TAX <- fun_TAX %>% filter(!str_detect(Family, "Aspergillaceae"))
fun_t_ASV <- t(fun_ASV)
FUN_ASV_TAX <- merge(fun_t_ASV, fun_TAX, by = "row.names")

## create the combined OTU table plus taxonomy string metacoder needs

#read it in using phyloseq object
tax_table(fun_phyloseq_rare)<-tax_table(fun_phyloseq_rare)[,1:7]
fun_x1<-parse_phyloseq(fun_phyloseq_rare) #this takes ages

#calculate relative abundance
fun_x1$data$abund_data <- calc_obs_props(fun_x1, "otu_table")

#calculate total abundance by forest type
fun_x1$data$Plot<-calc_taxon_abund(fun_x1, "abund_data",groups=fun_x1$data$sample_data$Plot) #this takes a while

### make taxonomy plots
# make sure we use functions from correct package
transform <- microbiome::transform
# merge rare taxa to speed up examples
fun_pseq <- microbiome::transform(fun_phyloseq_filt, "compositional")
fun_pseq_class <- aggregate_rare(fun_pseq, level = "Class", detection = 1/100, prevalence = 2/100)

#create dataframe from phyloseq 
FUN_df <- psmelt(fun_pseq_class)

### construct a stacked barplot showing relative abundance at 'Phylum' level
#expand color palettes to fit data
colourCount <- length(unique(FUN_df$Class))
getPalette <- colorRampPalette(brewer.pal(9, "Paired"))

#filter taxa by Abundance + merge "Unknown and Other" groups
FUN_df$Class[FUN_df$Abundance < 0.02] <- "Taxa < 2% Abundant"
FUN_df$Class[FUN_df$Class == "Unknown" | FUN_df$Class == "Other" | FUN_df$Class == "Taxa < 2% Abundant"] <- "Taxa < 2% Abundant"

#change relevel <2% to be at the bottom
#FUN_df$Class <- relevel(factor(FUN_df$Class), ref = "Taxa < 5% Abundant")
FUN_df$Class <- factor(FUN_df$Class)
lapply(FUN_df, levels) #check levels
FUN_df$Class <- fct_relevel(FUN_df$Class, "Taxa < 2% Abundant", after = Inf)

#plot the relative abundance by Class
fun_class_plot_facet <- ggplot(FUN_df, aes(fill=Class, y=Abundance, x=Depth)) + geom_bar(position="fill", stat="identity") + theme_minimal() + facet_grid(~Plot) + 
  ylab("Relative Abundance") + theme(axis.text.x = element_text(angle=45)) + scale_y_continuous(labels = scales::percent) + scale_fill_manual(values = getPalette(colourCount)) #+ scale_fill_manual(values = taxa_colors)

tiff("Figures/PlotS5C.tiff", width = 6, height = 4, units = 'in', res = 300)
fun_class_plot_facet
dev.off()

#for phylum
fun_pseq_phylum <- aggregate_rare(fun_pseq, level = "Phylum", detection = 1/100, prevalence = 2/100)
FUN_df_phylum <- psmelt(fun_pseq_phylum)
FUN_df_phylum$Phylum[FUN_df_phylum$Abundance < 0.02] <- "Taxa < 2% Abundant"
FUN_df_phylum$Phylum[FUN_df_phylum$Phylum == "Unknown" | FUN_df_phylum$Phylum == "Other" | FUN_df_phylum$Phylum == "Taxa < 2% Abundant"] <- "Taxa < 2% Abundant"
FUN_df_phylum$Phylum <- factor(FUN_df_phylum$Phylum)
```

#### across family, across plot

```{r, fig.fullwidth=TRUE, fig.height=6, fig.width=15}
#merge samples by category
fun_samples<-merge_samples(fun_phyloseq_rare, "Plot")
sample_data(fun_samples)$Plot = sample_names(fun_samples)
fun_samples@sam_data$Plot <- factor(fun_samples@sam_data$Plot, levels = c("30", "75", "250", "900", "1900"))

#aggregate samples by taxonomic rank (genus here)
fun_family <- fun_samples %>%
  tax_glom(taxrank = "Family") %>%                     # agglomerate at genus level
  transform_sample_counts(function(x) {x/sum(x)} ) %>% #transform to rel. abundance
  psmelt() %>%                                         #melt to long format
  filter(Abundance > 0.01) 

#make an other group
fun_family.plot = fun_family |>
  mutate(Family = if_else(Abundance < 0.03, "Others (< 3%)", Family))

fun_family.plot$Family <- as.factor(fun_family.plot$Family)
levels(fun_family.plot$Family)

fun_family.plot$Family = factor(fun_family.plot$Family, levels = c("Aspergillaceae", "Coniochaetaceae",  "Debaryomycetaceae", "Hypocreaceae", "Microascaceae", "Ophiocordycipitaceae", "Pyronemataceae", "Saccotheciaceae", "Sporidiobolaceae", "Tetragoniomycetaceae", "Trichocomaceae","Trimorphomycetaceae", "Others (< 3%)"))

fungen_colors <- c("#4E79A7FF", "#B07AA1FF", "#499894FF", "#FABFD2FF", "#59A14FFF", "#F1CE63FF", "#8CD17DFF", "#4c2f61", "#86BCB6FF","#805840FF", "#F28E2BFF", "#F8E898FF",  "#D3D3D3")

#making the plot using ggplots
fun_fam_plot <- ggplot(fun_family.plot, aes(x = Plot, y = Abundance, fill = Family)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = fungen_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Family > 1%) \n") +
  xlab("Plot") +
  ggtitle("Composition of Fungal Families") + scale_y_continuous(labels = scales::percent) + theme_bw() + #theme(axis.text = element_text(size=15), 
      #  axis.title=element_text(size=15)) +
  theme(legend.key.size = unit(0.5, 'cm')) +
   theme(axis.title.y = element_text(hjust=0.5, vjust = -4)) + guides(fill=guide_legend(title="Fungal Families"))
```

#### treemap, genera

```{r, eval=FALSE}
# grouping for abundances
fun_family.plot = fun_family |>
  mutate(Family = if_else(Abundance < 0.01, "Others", Family))

fun_family.plot <- fun_family.plot %>%
  group_by(Plot, Family) %>%
  summarize(Abundance=sum(Abundance))

ggplot(fun_family.plot, aes(area = Abundance, label = Family, fill=Family, subgroup = Family)) +
  geom_treemap() +
  geom_treemap_text(colour = "black", place = "topleft", reflow = T) +
  facet_wrap(~Plot) +
  geom_treemap_subgroup_border(colour="black", size = 2) + theme_classic() +
  ggtitle("Composition of Fungal Families Across Degradation Transects in Brunei") +
   theme(plot.title = element_text(hjust = 0.5,size=20)) + scale_fill_manual(values = taxa_colors) + 
  theme(axis.text = element_text(size=15), 
        axis.title=element_text(size=25))
```

### Functional guilds

```{r}
fun_function_colors <- c("#8CD17DFF",  "#FF9D9AFF", "#2f70a1","#4692b0", "#0a3351", "#A0CBE8FF", "#9d9dc7", "#edd7d9", "#5a5a83","#c9c9dd",  "#92351e", "#fbe3c2", "#b9563f", "#e69c6b", "#f2c88f", "#d37750", "#ecb27d")

fun_phyloseq_rare.df[fun_phyloseq_rare.df == ''] <- NA
fun_df.clean <- subset(fun_phyloseq_rare.df, primary_lifestyle != "NA")
fun_functions <- read.csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_ITS/Post_tax/Fungal_functions.csv") %>% dplyr::rename(primary_lifestyle = Function)
fun_df.clean <- fun_df.clean %>% dplyr::left_join(fun_functions)

#group
nectar_pollen_sap <- paste(c("Nectar/tap saprotroph", "Pollen saprotroph"), collapse = '|')
fun_df.clean <- fun_df.clean %>% mutate(Rename = gsub(nectar_pollen_sap, "Nectar/tap/pollen saprotroph", Rename)) 
fun_df.clean <- fun_df.clean %>%
  group_by(Transect, Plot, Depth, OTU, Sample, Phylum, Class, Order, Family, Genus, primary_lifestyle, Rename) %>%  # Group by columns that should remain unchanged
  summarise(Abundance = sum(Abundance), .groups = "drop")  # Summarize the 'value' column by taking the sum
unique(fun_df.clean$Rename)

#relevel
fun_df.clean$Rename = factor(fun_df.clean$Rename, 
                             levels = c("Lichenized", "Ectomycorrhizal", "Epiphyte",
                                        "Animal endosymbiont", "Foliar endophyte", "Sooty mold", 
                                        "Animal parasite", "Plant pathogen", "Mycoparasite", "Lichen parasite",
                                        "Soil saprotroph", "Litter saprotroph", "Wood saprotroph", "Dung saprotroph",
                                        "Nectar/tap/pollen saprotroph", "Pollen saprotroph", 
                                        "Unspecified saprotroph"))

fun_function_plot <- ggplot(fun_df.clean, aes(x = Depth, y = Abundance, fill = Rename)) + 
  geom_bar(position="fill",stat = "identity") +
  scale_fill_manual(values = fun_function_colors) +
  guides(fill = guide_legend(reverse = FALSE, keywidth = 1, keyheight = 1)) +
  ylab("Relative Abundance (Genera > 1%) \n") +
  xlab("Depth") + 
  facet_wrap(~Plot) + 
  theme(plot.title = element_blank()) +
  scale_y_continuous(labels = scales::percent) + # can change y axis to percentages (100%)
  theme_bw() + guides(fill=guide_legend(title="Biofunction")) + 
   theme(axis.text.x = element_text(margin = margin(t = 5))) +
  theme(axis.text = element_text(size=10), 
        axis.title=element_text(size=12)) + 
  theme(axis.title.y = element_text(hjust=0.5, vjust = -4)) +
   theme(legend.title = element_text(size = 12), 
       legend.text = element_text(size =20))


tiff("Figures/PlotS8.tiff", width = 12, height = 6, units = 'in', res = 300)
fun_function_plot
dev.off()

# grouping for abundances
fun_df.clean.plot = fun_df.clean |>
  mutate(Rename = if_else(Abundance < 0.0001, "Others", Rename))

fun_df.clean.plot.df <- fun_df.clean.plot %>%
  group_by(Plot, Rename) %>%
  summarize(Abundance=sum(Abundance))

ggplot(fun_df.clean.plot.df, aes(area = Abundance, label = Rename, fill=Rename, subgroup = Rename)) +
  geom_treemap() +
  geom_treemap_text(colour = "black", place = "topleft", reflow = T) +
  facet_wrap(~Plot) +
  geom_treemap_subgroup_border(colour="black", size = 2) + theme_classic() +
  ggtitle("Composition of Fungal Genera Across Degradation Transects in Brunei") +
   theme(plot.title = element_text(hjust = 0.5,size=20)) + scale_fill_manual(values = taxa_colors) + 
  theme(axis.text = element_text(size=15), 
        axis.title=element_text(size=25))
```

#### ANOVAs for function

```{r}
head(fun_df.clean)

fun_function_bare <- fun_df.clean %>% dplyr::select(Sample, Transect, Plot, Depth, Rename, Abundance)
fun_function_bare <- fun_function_bare %>% group_by(Sample, Transect, Plot, Depth, Rename) %>% mutate(totalrelabund = sum(Abundance)) %>% select(-Abundance) %>% unique()
fun_function_bare <- fun_function_bare %>% group_by(Sample, Transect, Plot, Depth) %>% mutate(relabund = totalrelabund/sum(totalrelabund)) %>% dplyr::select(-totalrelabund) %>% pivot_wider(names_from = Rename, values_from = relabund)

head(fun_function_bare)


#ANOVAs

hist(fun_function_bare$`Unspecified saprotroph`)
##Unspecified saprotroph (30,75,250 are sig higher than 900/1900)
function_anova <- lmer((`Unspecified saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

hist(log(fun_function_bare$`Soil saprotroph`))
##Soil saprotroph
function_anova <- lmer((`Soil saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)

function_anova <- glmmTMB((`Soil saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

##Animal parasite (kinda higher for most pristine)
hist((fun_function_bare$`Animal parasite`))
function_anova <- lmer((`Animal parasite`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

function_anova <- glmmTMB((`Animal parasite`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

##Wood saprotroph (higher for pristine)
hist((fun_function_bare$`Wood saprotroph`))
function_anova <- lmer((`Wood saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

function_anova <- glmmTMB((`Wood saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

##Nectar/tap/pollen saprotroph
hist((fun_function_bare$`Nectar/tap/pollen saprotroph`))
function_anova <- lmer((`Nectar/tap/pollen saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)

function_anova <- glmmTMB((`Nectar/tap/pollen saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

##Plant pathogen
function_anova <- lmer((`Plant pathogen`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)

function_anova <- glmmTMB((`Plant pathogen`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

##Litter saprotroph (higher for 900 plot specifically against degraded)
function_anova <- lmer((`Litter saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

function_anova <- glmmTMB((`Litter saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

##Mycoparasite (higher specifically for most pristine)
function_anova <- lmer((`Mycoparasite`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

function_anova <- glmmTMB((`Mycoparasite`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

##Sooty mold
hist(fun_function_bare$`Sooty mold`)
function_anova <- lmer((`Sooty mold`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)

function_anova <- glmmTMB((`Sooty mold`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
emmeans(function_anova, list(pairwise ~Depth), adjust = "tukey")

##Foliar endophyte
hist(fun_function_bare$`Foliar endophyte`)
function_anova <- glmmTMB((`Foliar endophyte`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
emmeans(function_anova, list(pairwise ~Depth), adjust = "tukey")

##Ectomycorrhizal
hist(fun_function_bare$`Ectomycorrhizal`)
function_anova <- lmer((`Ectomycorrhizal`) ~ Plot+Depth + (1|Transect), data = fun_function_bare)
check_model(function_anova)
anova(function_anova, type = 2)

function_anova <- glmmTMB((`Ectomycorrhizal`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
emmeans(function_anova, list(pairwise ~Plot*Depth), adjust = "tukey")

##Lichen parasite
hist(fun_function_bare$`Lichen parasite`)
function_anova <- glmmTMB((`Lichen parasite`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

##Lichenized (specifically higher in 75)
hist(fun_function_bare$`Lichenized`)
function_anova <- glmmTMB((`Lichenized`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

##Animal endosymbiont
hist(fun_function_bare$`Animal endosymbiont`)
function_anova <- glmmTMB((`Animal endosymbiont`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")

##Dung saprotroph
hist(fun_function_bare$`Dung saprotroph`)
function_anova <- glmmTMB((`Dung saprotroph`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
emmeans(function_anova, list(pairwise ~Plot), adjust = "tukey")

##Epiphyte
hist(fun_function_bare$`Epiphyte`)
function_anova <- glmmTMB((`Epiphyte`) ~ Plot+Depth + (1|Transect), data = fun_function_bare, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = function_anova))
drop1(function_anova, test = "Chisq")
```

### Rank abundance distributions

```{r, eval= FALSE}
fun_named.ASV <- t(otu_table(fun_phyloseq_rare))
fun_taxonomy.table <- data.frame(tax_table(fun_phyloseq_rare))
fun_samp.data<-data.frame(sample_data(fun_phyloseq_rare))

fun_ASV.names <- fun_taxonomy.table$Species
fun_ASV.numbers <- paste("ASV",taxa_names(fun_phyloseq_rare),sep="_")

fun_RankAbun1 <- rankabundance(fun_named.ASV) #this can take a while
rankabunplot(fun_RankAbun1, scale='logabun', addit=FALSE, specnames = c(1,2,3))
fun_allrankabund_plot <- recordPlot()
fun_rankabund <- rankabuncomp(fun_named.ASV, y=fun_samp.data,factor='Plot', scale='logabun', legend=FALSE,las=1) #this also takes a while
fun_plotrankabund_plot <- recordPlot()
fun_plotrankabund_plot

#legend("topright",pch=21,legend=c("0","1","2","4","8","16","24","36"),col=rainbow(8),bty="n",ncol=2,cex=1)
```

### Major shifts

```{r}
#richness and shannon div
fun_DIV_merge <- fun_DIV %>% dplyr::select(Sample, Fungi_Observed, Shannon, Transect, Plot, Depth)

#create dataframe from phyloseq 
fun_df_class <- psmelt(fun_pseq_class)

#abundance of major class
fun_df_merge <- fun_df_class %>% dplyr::select(Sample, Class, Abundance, Transect, Plot, Depth)
unique(fun_df_merge$Class)
fun_class_list <- paste(c("Eurotiomycetes", "Saccharomycetes", "Sordariomycetes", "Agaricomycetes", "Pezizomycetes"), collapse = '|')
fun_df_merge <- fun_df_merge %>% filter(stringr::str_detect(Class, fun_class_list))
fun_df_merge <- fun_df_merge %>% group_by(Class) %>%
  pivot_wider(names_from = Class, values_from = Abundance)

#abundance of major phyla
fun_df_phylum <- psmelt(fun_pseq_phylum)
fun_df_merge_phylum <- fun_df_phylum %>% dplyr::select(Sample, Phylum, Abundance, Transect, Plot, Depth)
unique(fun_df_merge_phylum$Phylum)
fun_phylum_list <- paste(c("Ascomycota", "Basidiomycota"), collapse = '|')
fun_df_merge_phylum <- fun_df_merge_phylum %>% filter(stringr::str_detect(Phylum, fun_phylum_list))
fun_df_merge_phylum <- fun_df_merge_phylum %>% group_by(Phylum) %>%
  pivot_wider(names_from = Phylum, values_from = Abundance)

#family
fun_pseq_fam <- aggregate_rare(fun_pseq, level = "Family", detection = 1/100, prevalence = 2/100)

#create dataframe from phyloseq 
fun_fam_df <- psmelt(fun_pseq_fam)

fun_fam_merge <- fun_fam_df %>% dplyr::select(Sample, Family, Abundance, Transect, Plot, Depth)
fam_list <- list(unique(fun_fam_merge$Family))
fun_fam_list <- paste(c("Trichocomaceae", "Hypocreaceae", "Microascaceae", "Trimorphomycetaceae", "Omphalotaceae", "Ophiocordycipitaceae", "Coniochaetaceae"), collapse = '|')
fun_fam_merge <- fun_fam_merge %>% filter(stringr::str_detect(Family, fun_fam_list))
fun_fam_merge <- fun_fam_merge %>% group_by(Family) %>%
  pivot_wider(names_from = Family, values_from = Abundance)

#function
fun_func_merge <- fun_function_bare %>% select(-`Animal parasite`, -`Plant pathogen`, -Mycoparasite, -`Sooty mold`, -`Foliar endophyte`, -`Lichen parasite`, -Lichenized, -`Animal endosymbiont`, -Epiphyte)
#add saprotrophs tgt
fun_func_merge_all <- fun_func_merge %>% rowwise() %>% 
  mutate(saprotroph_fun_total = sum(c_across(contains("saprotroph")), na.rm = TRUE)) %>% ungroup()
fun_func_merge <- fun_func_merge_all %>% dplyr::select(Transect, Plot, Depth, saprotroph_fun_total, Ectomycorrhizal)

fun_major_merge <- fun_DIV_merge %>% dplyr::full_join(fun_df_merge) %>% dplyr::full_join(fun_fam_merge) %>% dplyr::full_join(fun_func_merge) %>% dplyr::full_join(fun_df_merge_phylum)
fun_major_merge <- fun_major_merge %>% dplyr::rename(Fungi_Shannon = Shannon) %>% 
  dplyr::select(-Sample)

#make new datasets
fun_df_merge_sum <- fun_df_merge %>% group_by(Plot) %>% summarise_if(is.numeric, mean, na.rm = TRUE)
fun_df_merge_sum <- fun_df_merge_sum %>% pivot_longer(!c(Plot), names_to = "Group", values_to = "Abundance")
fun_df_merge_sum$Plot <- as.numeric(fun_df_merge_sum$Plot)

fun_df_merge_new <- fun_df_merge %>% dplyr::select(-Sample) %>% pivot_longer(!c(Plot, Transect, Depth), names_to = "Group", values_to = "Abundance")
fun_df_merge_new$Plot <- as.numeric(fun_df_merge_new$Plot)
fun_df_merge_new$Group <- factor(fun_df_merge_new$Group, levels = c("Eurotiomycetes", "Sordariomycetes", "Agaricomycetes","Saccharomycetes", "Pezizomycetes"))

#family
fun_fam_merge_sum <- fun_fam_merge %>% group_by(Plot) %>% summarise_if(is.numeric, mean, na.rm = TRUE)
fun_fam_merge_sum <- fun_fam_merge_sum %>% pivot_longer(!c(Plot), names_to = "Group", values_to = "Abundance")
fun_fam_merge_sum$Plot <- as.numeric(fun_fam_merge_sum$Plot)
fun_fam_merge_sum$Group <- factor(fun_fam_merge_sum$Group, levels = c("Trichocomaceae", "Hypocreaceae", "Microascaceae", "Omphalotaceae", "Trimorphomycetaceae", "Ophiocordycipitaceae", "Coniochaetaceae"))

fun_taxshift_plot <- ggplot() + theme_classic() + 
  geom_line(aes(x = Plot, y = Abundance, color = Group), data = fun_df_merge_sum) + 
  geom_point(aes(x = Plot, y = Abundance, color = Group), size = 6, data = fun_df_merge_sum) +
  geom_jitter(aes(x = Plot, y = Abundance, color = Group), 
             alpha = 0.5, width = 0.15, data = fun_df_merge_new) + 
 # geom_line(aes(x = Plot, y = Abundance, color = Group), linetype="dotted", data = fun_fam_merge_sum) + 
 # geom_point(aes(x = Plot, y = Abundance, color = Group), size = 2, data = fun_fam_merge_sum) +
  ylab("Relative Abundance") + scale_color_manual(values = c("#DD5129", "#FAB255", "#0F7BA2","#43B284", "#4c2f61"))

  
tiff("Figures/Plot3C.tiff", width = 6, height = 4, units = 'in', res = 300)
fun_taxshift_plot
dev.off()
```

### CCA
 
#### Prep data
 
```{r, eval = FALSE}
fun_community_cca <- fun_community[!(row.names(fun_community) %in% c("TEITSBED","TEITSCAC", "TEITSECD", "TEITSCCC","TEITSDAD", "TEITSDEC", "TEITSAAC", "TEITSDAA", "TEITSDDA", "TEITSDDB", "TEITSEBA")),]

fun_env_cca <- fun_env_data %>% tibble::column_to_rownames("Sample") %>% dplyr::select(-Transect, -Plot, -Depth)
fun_env_cca <- fun_env_cca[!(row.names(fun_env_cca) %in% c("TEITSBED","TEITSCAC", "TEITSECD", "TEITSCCC","TEITSDAD", "TEITSDEC", "TEITSAAC", "TEITSDAA", "TEITSDDA", "TEITSDDB", "TEITSEBA")),]
 
#Hellinger transformation to species matrix due to rare sp.
fun_community_trans <- decostand(fun_community_cca, "hellinger")
#scale and center soil parameters due to diff units
fun_env_trans <- decostand(fun_env_cca, method = "standardize")
#variables are now centered around a mean of 0
round(apply(fun_env_trans, 2, mean), 1)
#and scaled to have a standard deviation of 1
apply(fun_env_trans, 2, sd)
```
 
#### Run CCA
 
```{r, eval = FALSE}
#run CCA
fun_CCA <- cca(fun_community_trans~., fun_env_trans)
plot(fun_CCA) #first visualisation
summary(fun_CCA) #look at model
 
#stepwise model
fun_finalmodel<- ordistep(fun_CCA, scope=formula(fun_CCA))
vif<-vif.cca(fun_finalmodel)
vif #go to original dataset and remove redundant terms (VIF > 10)
fun_finalmodel
#16.3% inertia explained
plot(fun_finalmodel)
capture.output(vif, file = "vif.txt")
summary(fun_finalmodel)
screeplot(fun_finalmodel)
 
#significance tests
anova.cca(fun_finalmodel)
anova.cca(fun_finalmodel, by="terms")
anova.cca(fun_finalmodel, by="axis")
 
#colors for FT
fun_env_data$color <- fun_env_data$Plot %>% as.character()
fun_env_data$color[fun_env_data$color == "30"] <- "#74c8c3"
fun_env_data$color[fun_env_data$color == "75"] <- "#5a97c1" 
fun_env_data$color[fun_env_data$color == "250"] <- "#0a2e57"
fun_env_data$color[fun_env_data$color == "900"] <- "#d8d97a"
fun_env_data$color[fun_env_data$color == "1900"] <- "#669d62"
summary(fun_env_data)
 
#generate biplot for sites and soil var
plot(c(-2.5, 2), c(-1.5, 1.5), xlab="CCA1 (47.7%)", ylab="CCA2 (45.3%)", type="n")
abline(h = 0, v = 0, lty = 2, lwd =.5)
points(fun_finalmodel, dis="lc", pch=21, col="black", bg=fun_env_data$color, cex=1.5, scaling = "sites")
text(fun_finalmodel, dis="cn", cex = 1.5, scaling = "sites")
fun_cca_plot <- recordPlot()
 
#generate biplot of sites and species
plot(fun_finalmodel, scaling = "sites", type="n")
points(fun_finalmodel, "species", pch=1, col="forestgreen", cex=0.8, scaling = "sites")
text(fun_finalmodel, "species", col="black", cex=0.4, scaling = "sites")

#colors for depth
fun_env_data$color <- fun_env_data$Depth %>% as.character()
fun_env_data$color[fun_env_data$color == "0-10"] <- "#e7e5cc"
fun_env_data$color[fun_env_data$color == "10-50"] <- "#c2d6a4" 
fun_env_data$color[fun_env_data$color == "50-100"] <- "#669d62"
fun_env_data$color[fun_env_data$color == "100-150"] <- "#192813"
summary(fun_env_data)

#generate biplot for sites and soil var
plot(c(-2.5, 2), c(-1.5, 1.5), xlab="CCA1 (47.7%)", ylab="CCA2 (45.3%)", type="n")
abline(h = 0, v = 0, lty = 2, lwd =.5)
points(fun_finalmodel, dis="lc", pch=21, col="black", bg=fun_env_data$color, cex=1.5, scaling = "sites")
text(fun_finalmodel, dis="cn", cex = 1.5, scaling = "sites")
fun_cca_plot_depth <- recordPlot()
 
#generate biplot of sites and species
plot(fun_finalmodel, scaling = "sites", type="n")
points(fun_finalmodel, "species", pch=1, col="forestgreen", cex=0.8, scaling = "sites")
text(fun_finalmodel, "species", col="black", cex=0.4, scaling = "sites")
```

## Combined

### Comparing richness ratios

```{r}
bact_DIV_obs <- bact_DIV %>% dplyr::select(Transect, Plot, Depth, Bacteria_Observed)
arc_DIV_obs <- arc_DIV %>% dplyr::select(Transect, Plot, Depth, Archaea_Observed)
fun_DIV_obs <- fun_DIV %>% dplyr::select(Transect, Plot, Depth, Fungi_Observed)

all_DIV <- bact_DIV_obs %>% dplyr::full_join(arc_DIV_obs) %>% dplyr::full_join(fun_DIV_obs)
all_DIV$Bact_Arc <- all_DIV$Bacteria_Observed / all_DIV$Archaea_Observed
all_DIV$Bact_Fun <- all_DIV$Bacteria_Observed / all_DIV$Fungi_Observed
all_DIV$Arc_Fun <- all_DIV$Archaea_Observed / all_DIV$Fungi_Observed

#anova
bact_fun_obs <- lmer(log(Bact_Fun) ~ Plot*Depth + (1|Transect), data = all_DIV)
check_model(bact_fun_obs)
anova(bact_fun_obs, type = 2)
bact_fun_emmeans <- emmeans(bact_fun_obs, ~Depth)
bact_fun_emmeans <- as.data.frame(bact_fun_emmeans)
emmeans(bact_fun_obs, list(pairwise ~Depth), adjust = "tukey")

#back transform
bact_fun_emmeans$emmean <- exp(bact_fun_emmeans$emmean)
bact_fun_emmeans$lower.CL <- exp(bact_fun_emmeans$lower.CL)
bact_fun_emmeans$upper.CL <- exp(bact_fun_emmeans$upper.CL)

#plot emmeans
bact_fun_plot <- ggplot() + geom_point(aes(x = Depth, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = bact_fun_emmeans) +
  geom_errorbar(aes(x = Depth, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = bact_fun_emmeans) +
  geom_jitter(aes(x = Depth, y = Bact_Fun, color = Depth), width = 0.25, alpha = 0.5, data = all_DIV) + 
  labs(y = "Bacteria:Fungi", x = "Depth") + scale_color_manual(values = depth_colors) + theme_classic() + theme(legend.position = "none")

tiff("Figures/PlotS4A.tiff", width = 4, height = 3, units = 'in', res = 300)
bact_fun_plot
dev.off()

#anova
bact_arc_obs <- lmer(log(Bact_Arc) ~ Plot*Depth + (1|Transect), data = all_DIV)
check_model(bact_arc_obs)
anova(bact_arc_obs, type = 2)
bact_arc_emmeans <- emmeans(bact_arc_obs, ~Plot*Depth)
bact_arc_emmeans <- as.data.frame(bact_arc_emmeans)
bact_arc_emmeans_plot <- emmeans(bact_arc_obs, ~Plot)
bact_arc_emmeans_plot <- as.data.frame(bact_arc_emmeans_plot)
emmeans(bact_arc_obs, list(pairwise ~Plot), adjust = "tukey")
emmeans(bact_arc_obs, list(pairwise ~Depth), adjust = "tukey")

#back transform
bact_arc_emmeans$emmean <- exp(bact_arc_emmeans$emmean)
bact_arc_emmeans$lower.CL <- exp(bact_arc_emmeans$lower.CL)
bact_arc_emmeans$upper.CL <- exp(bact_arc_emmeans$upper.CL)

bact_arc_emmeans_plot$emmean <- exp(bact_arc_emmeans_plot$emmean)
bact_arc_emmeans_plot$lower.CL <- exp(bact_arc_emmeans_plot$lower.CL)
bact_arc_emmeans_plot$upper.CL <- exp(bact_arc_emmeans_plot$upper.CL)

#PLOT x DEPTH
Bact_Arc_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = bact_arc_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = bact_arc_emmeans) +
  geom_jitter(aes(x = Plot, y = Bact_Arc, color = Depth), width = 0.25, alpha = 0.5, data = all_DIV) + 
  labs(y = "Bacteria:Archaea", x = "Plot") + scale_color_manual(values = depth_colors) + theme_classic() + theme(legend.position = "none")

#PLOT ONLY
bact_arc_emmeans <- emmeans(bact_arc_obs, ~Plot)
bact_arc_emmeans <- as.data.frame(bact_arc_emmeans)
bact_arc_emmeans_plot <- emmeans(bact_arc_obs, ~Plot)
bact_arc_emmeans_plot <- as.data.frame(bact_arc_emmeans_plot)

#back transform
bact_arc_emmeans$emmean <- exp(bact_arc_emmeans$emmean)
bact_arc_emmeans$lower.CL <- exp(bact_arc_emmeans$lower.CL)
bact_arc_emmeans$upper.CL <- exp(bact_arc_emmeans$upper.CL)

#PLOT ONLY
Bact_Arc_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Plot), 
                                       position = position_dodge(width = 0.5), size = 5, data = bact_arc_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Plot), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = bact_arc_emmeans) +
  geom_jitter(aes(x = Plot, y = Bact_Arc, color = Plot), width = 0.25, alpha = 0.5, data = all_DIV) + 
  labs(y = "Bacteria:Archaea", x = "Plot") + scale_color_manual(values = plot_colors) + theme_classic() + theme(legend.position = "none")

tiff("Figures/PlotS4B.tiff", width = 4, height = 3, units = 'in', res = 300)
Bact_Arc_plot
dev.off()


#anova
bact_arc_emmeans <- emmeans(bact_arc_obs, ~Depth)
bact_arc_emmeans <- as.data.frame(bact_arc_emmeans)
bact_arc_emmeans_plot <- emmeans(bact_arc_obs, ~Depth)
bact_arc_emmeans_plot <- as.data.frame(bact_arc_emmeans_plot)

#back transform
bact_arc_emmeans$emmean <- exp(bact_arc_emmeans$emmean)
bact_arc_emmeans$lower.CL <- exp(bact_arc_emmeans$lower.CL)
bact_arc_emmeans$upper.CL <- exp(bact_arc_emmeans$upper.CL)

#DEPTH ONLY
Bact_Arc_depth <- ggplot() + geom_point(aes(x = Depth, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = bact_arc_emmeans) +
  geom_errorbar(aes(x = Depth, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = bact_arc_emmeans) +
  geom_jitter(aes(x = Depth, y = Bact_Arc, color = Depth), width = 0.25, alpha = 0.5, data = all_DIV) + 
  labs(y = "Bacteria:Archaea", x = "Depth") + scale_color_manual(values = depth_colors) + theme_classic() + theme(legend.position = "none")

tiff("Figures/PlotS4C.tiff", width = 4, height = 3, units = 'in', res = 300)
Bact_Arc_depth
dev.off()

#anova
arc_fun_obs <- lmer(log(Arc_Fun) ~ Plot + (1|Transect), data = all_DIV)
check_model(arc_fun_obs)
anova(arc_fun_obs, type = 2)
arc_fun_emmeans <- emmeans(arc_fun_obs, ~Plot)
arc_fun_emmeans <- as.data.frame(arc_fun_emmeans)
emmeans(arc_fun_obs, list(pairwise ~Plot), adjust = "tukey")

#back transform
arc_fun_emmeans$emmean <- exp(arc_fun_emmeans$emmean)
arc_fun_emmeans$lower.CL <- exp(arc_fun_emmeans$lower.CL)
arc_fun_emmeans$upper.CL <- exp(arc_fun_emmeans$upper.CL)

#plot emmeans
arc_fun_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Plot), 
                                       position = position_dodge(width = 0.5), size = 5, data = arc_fun_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Plot), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = arc_fun_emmeans) +
  geom_jitter(aes(x = Plot, y = Arc_Fun, color = Plot), width = 0.25, alpha = 0.5, data = all_DIV) + 
  labs(y = "Archaea:Fungi", x = "Plot") + scale_color_manual(values = plot_colors) + theme_classic() + theme(legend.position = "none")

tiff("Figures/PlotS4D.tiff", width = 4, height = 3, units = 'in', res = 300)
arc_fun_plot
dev.off()

ratios_plot <- (Bact_Arc_plot | Bact_Arc_depth)  / (arc_fun_plot | bact_fun_plot)
ratios_plot + plot_annotation(tag_levels = "A")
```

### Figure 2

```{r}
library(patchwork)
richness_panel <- bact_obvs_plot | arc_obvs_plot | fun_obvs_plot
shannon_panel <- bact_shannon_plot / arc_shannon_plot / fun_shannon_plot
nmds_panel <- bact_nmds_plot | arc_nmds_plot | fun_nmds_plot
taxshift_panel <- bact_taxshift_plot / arc_taxshift_plot / fun_taxshift_plot
function_panel <- bact_function_plot / arc_function_plot / fun_function_plot
family_panel <- bact_fam_plot / arc_fam_plot / fun_fam_plot
family_panel  + plot_annotation(tag_levels = "A")
#cca_panel <- ggarrange(plotlist = list(bact_cca_plot, arc_cca_plot, fun_cca_plot), ncol = 1, widths = 0.5, heights = 20)
#cca_panel_depth <- ggarrange(plotlist = list(bact_cca_plot_depth, arc_cca_plot_depth, fun_cca_plot_depth), ncol = 1, widths = 0.5, heights = 20)
#rankabund_panel <- ggarrange(plotlist = list(bact_plotrankabund_plot, arc_plotrankabund_plot, fun_plotrankabund_plot), ncol = 1, widths = 0.5, heights = 20)
```

### Combined table of rich/div, top phyla, top functions

```{r}
summarised_microbes <- bact_major_merge %>% dplyr::full_join(arc_major_merge) %>% dplyr::full_join(fun_major_merge) %>% rownames_to_column()
summarised_microbes <- summarised_microbes %>% dplyr::select(rowname, Transect, Plot, Depth, Bacteria_Observed, Archaea_Observed, Fungi_Observed, Bacteria_Shannon, Archaea_Shannon, Fungi_Shannon, Firmicutes, Actinobacteriota, Proteobacteria, Acidobacteriota, Crenarchaeota, Thermoplasmatota, Halobacterota, Agaricomycetes, Ascomycota, Basidiomycota, Eurotiomycetes, Pezizomycetes, Saccharomycetes, Sordariomycetes, everything())
write_csv(summarised_microbes, "/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/summarisedmicrobes.csv")
```

### Exploring specific functions

```{r}
#functional database (summarised)
function_data <- summarised_microbes %>% dplyr::select(Transect, Plot, Depth, cellulolysis_bact, fermentation_bact, aerobic_chemoheterotrophy_bact, chemoheterotrophy_bact, methanogenesis_arc, methylotrophy_arc, chemoheterotrophy_arc, aerobic_ammonia_oxidation_arc, nitrification_arc, saprotroph_fun_total, Ectomycorrhizal)
function_data$methanogen_methanotroph <- function_data$methanogenesis_arc / function_data$methylotrophy_arc
function_data[function_data == "NaN"] <- NA
function_data[function_data == "Inf"] <- NA
```

#### Nitrogen

```{r}
bact_nitrogen_group_wide <- bact_nitrogen_group %>% dplyr::select(Transect, Plot, Depth, Sample, group, value)
bact_nitrogen_group_wide$value <- as.numeric(bact_nitrogen_group_wide$value)

bact_nitrogen_group_wide <- bact_nitrogen_group_wide %>% group_by(Plot, Sample) %>% pivot_wider(names_from = group, values_from = value)
bact_nitrogen_group_wide <- bact_nitrogen_group_wide %>% dplyr::rename(Reduction = 'nitrate reduction', Fixation = 'nitrogen fixation')
bact_nitrogen_group_wide[is.na(bact_nitrogen_group_wide)] <- 0


#separate anovas for all
#reduction (higher for degraded)
reduction_anova <- lmer((Reduction) ~ Plot+Depth + (1|Transect), data = bact_nitrogen_group_wide)
check_model(reduction_anova)
anova(reduction_anova, type = 2)
emmeans(reduction_anova, list(pairwise ~Plot), adjust = "tukey")
reduction_emmeans <- emmeans(reduction_anova, ~Plot)
reduction_emmeans <- as.data.frame(reduction_emmeans)
reduction_emmeans$Group <- "Reduction"

hist(log(bact_nitrogen_group_wide$Reduction))
reduction_anova <- glmmTMB((Reduction) ~ Plot+Depth + (1|Transect), data = bact_nitrogen_group_wide, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = reduction_anova))
drop1(reduction_anova, test = "Chisq")
emmeans(reduction_anova, list(pairwise ~Plot), adjust = "tukey")
reduction_emmeans <- emmeans(reduction_anova, ~Plot, type = "response")
reduction_emmeans <- as.data.frame(reduction_emmeans)
reduction_emmeans$Group <- "Reduction"

#denitrification
denitrification_anova <- lmer((Denitrification) ~ Plot + (1|Transect), data = bact_nitrogen_group_wide)
check_model(denitrification_anova)
anova(denitrification_anova, type = 2)
emmeans(denitrification_anova, list(pairwise ~Plot), adjust = "tukey")
denitrification_emmeans <- emmeans(denitrification_anova, ~Plot, type = "response")
denitrification_emmeans <- as.data.frame(denitrification_emmeans)
denitrification_emmeans$Group <- "Denitrification"

hist(log(bact_nitrogen_group_wide$Denitrification))
denitrification_anova <- glmmTMB((Denitrification) ~ Plot+Depth + (1|Transect), data = bact_nitrogen_group_wide, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = denitrification_anova))
drop1(denitrification_anova, test = "Chisq")
emmeans(denitrification_anova, list(pairwise ~Plot), adjust = "tukey")
denitrification_emmeans <- emmeans(denitrification_anova, ~Plot, type = "response")
denitrification_emmeans <- as.data.frame(denitrification_emmeans)
denitrification_emmeans$Group <- "Denitrification"

#Respiration
respiration_anova <- lmer((Respiration) ~ Plot + (1|Transect), data = bact_nitrogen_group_wide)
check_model(respiration_anova)
anova(respiration_anova, type = 2)
emmeans(respiration_anova, list(pairwise ~Plot), adjust = "tukey")
respiration_emmeans <- emmeans(respiration_anova, ~Plot, type = "response")
respiration_emmeans <- as.data.frame(respiration_emmeans)
respiration_emmeans$Group <- "Respiration"

hist(log(bact_nitrogen_group_wide$Respiration))
respiration_anova <- glmmTMB((Respiration) ~ Plot+Depth + (1|Transect), data = bact_nitrogen_group_wide, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = respiration_anova))
drop1(respiration_anova, test = "Chisq")
emmeans(respiration_anova, list(pairwise ~Plot), adjust = "tukey")
respiration_emmeans <- emmeans(respiration_anova, ~Plot, type = "response")
respiration_emmeans <- as.data.frame(respiration_emmeans)
respiration_emmeans$Group <- "Respiration"

#fixation
fixation_anova <- lmer((Fixation) ~ Plot+Depth + (1|Transect), data = bact_nitrogen_group_wide)
check_model(fixation_anova)
anova(fixation_anova, type = 2)
emmeans(fixation_anova, list(pairwise ~Plot), adjust = "tukey")
fixation_emmeans <- emmeans(fixation_anova, ~Plot, type = "response")
fixation_emmeans <- as.data.frame(fixation_emmeans)
fixation_emmeans$Group <- "Fixation"

hist(log(bact_nitrogen_group_wide$Fixation))
fixation_anova <- glmmTMB((Fixation) ~ Plot+Depth + (1|Transect), data = bact_nitrogen_group_wide, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = fixation_anova))
drop1(fixation_anova, test = "Chisq")
emmeans(fixation_anova, list(pairwise ~Plot), adjust = "tukey")
fixation_emmeans <- emmeans(fixation_anova, ~Plot, type = "response")
fixation_emmeans <- as.data.frame(fixation_emmeans)
fixation_emmeans$Group <- "Fixation"

#figure
#combine emmeans datasets
nitrogen_emmeans <- rbind(reduction_emmeans, denitrification_emmeans, respiration_emmeans, fixation_emmeans)
nitrogen_emmeans$Group <- factor(nitrogen_emmeans$Group, levels = c("Reduction", "Denitrification", "Respiration", "Fixation"))
names(nitrogen_emmeans)

nitrogen_emmeans_plot <- ggplot(nitrogen_emmeans, aes(fill = fct_rev(Plot), x = Group)) +
  geom_bar(stat = "identity", position = "dodge", color = "black", aes(x = Group, y = response)) + 
  geom_errorbar(aes(ymin = response + SE, ymax = response), 
                position = position_dodge(width = 0.9), width = 0.2, colour = "black") +
  theme_classic() +
  scale_fill_manual(values = rev(plot_colors)) +
    xlab("Functional Group") + ylab("Relative Abundance") + 
  theme(axis.text = element_text(size = 15, color = "black")) +
  theme(axis.title.x = element_text(size = 15), legend.title = element_text(size =15),
        legend.text = element_text(size =15), axis.title.y = element_blank()) + coord_flip() +
  guides(fill = guide_legend(reverse = TRUE, title = "Plot")) + theme(legend.position = "none")

tiff("Figures/Plot3G.tiff", width = 4, height = 4, units = 'in', res = 300)
nitrogen_emmeans_plot
dev.off()
```

#### Methane

```{r}
#anova
hist(log(function_data$methanogen_methanotroph))
meth_anova <- lmer((methanogen_methanotroph) ~ Plot+Depth + (1|Transect), data = function_data)
check_model(meth_anova)
anova(meth_anova, type = 2)
emmeans(meth_anova, list(pairwise ~Plot), adjust = "tukey")
meth_emmeans <- emmeans(meth_anova, ~Plot)
meth_emmeans <- as.data.frame(meth_emmeans)

# PLOT
meth_plot <- ggplot() + geom_point(aes(x = Plot, y = response, color = Plot), 
                                       position = position_dodge(width = 0.5), size = 5, data = meth_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Plot), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = meth_emmeans) +
  geom_jitter(aes(x = Plot, y = methanogen_methanotroph, color = Plot), width = 0.25, alpha = 0.5, data = function_data) + 
  labs(y = "Methanogen:Methanotroph", x = "Plot") + scale_color_manual(values = plot_colors) + theme_classic() +
  theme(legend.position = "none")

#methanogen anova
meth_anova <- lmer((methanogenesis_arc) ~ Plot+Depth + (1|Transect), data = function_data)
check_model(meth_anova)
anova(meth_anova, type = 2)
emmeans(meth_anova, list(pairwise ~Plot), adjust = "tukey")
meth_emmeans <- emmeans(meth_anova, ~Plot)
meth_emmeans <- as.data.frame(meth_emmeans)

# PLOT
methanogen_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Plot), 
                                       position = position_dodge(width = 0.5), size = 5, data = meth_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Plot), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = meth_emmeans) +
  geom_jitter(aes(x = Plot, y = methanogenesis_arc, color = Plot), width = 0.25, alpha = 0.5, data = function_data) + 
  labs(y = "Methanogen Rel. Abund.", x = "Plot") + scale_color_manual(values = plot_colors) + theme_classic() + 
  theme(legend.position = "none")

#methanotroph anova
hist((function_data$methylotrophy_arc))
meth_anova <- lmer((methylotrophy_arc) ~ Plot+Depth + (1|Transect), data = function_data)
check_model(meth_anova)
anova(meth_anova, type = 2)
emmeans(meth_anova, list(pairwise ~Plot), adjust = "tukey")
meth_emmeans <- emmeans(meth_anova, ~Plot)
meth_emmeans <- as.data.frame(meth_emmeans)


# PLOT
methanotroph_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Plot), 
                                       position = position_dodge(width = 0.5), size = 5, data = meth_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Plot), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = meth_emmeans) +
  geom_jitter(aes(x = Plot, y = methylotrophy_arc, color = Plot), width = 0.25, alpha = 0.5, data = function_data) + 
  labs(y = "Methanotroph Rel. Abund.", x = "Plot") + scale_color_manual(values = plot_colors) + theme_classic()

meth_panel <- meth_plot | methanogen_plot | methanotroph_plot

#### REPRESENT W HORIZONTAL BAR GRAPHS

arc_meth_wide <- function_data %>% dplyr::select(Transect, Plot, Depth, methanogenesis_arc, methylotrophy_arc) %>% dplyr::rename(Methanogenesis = methanogenesis_arc, Methylotrophy = methylotrophy_arc)

#separate anovas for all
#methanogenesis
methanogen_anova <- lmer((Methanogenesis) ~ Plot+Depth + (1|Transect), data = arc_meth_wide)
check_model(methanogen_anova)
anova(methanogen_anova, type = 2)
emmeans(methanogen_anova, list(pairwise ~Plot), adjust = "tukey")
methanogen_emmeans <- emmeans(methanogen_anova, ~Plot)
methanogen_emmeans <- as.data.frame(methanogen_emmeans)
methanogen_emmeans$Group <- "Methanogenesis"

#methanotrophy
methanotroph_anova <- lmer((Methylotrophy) ~ Plot+Depth + (1|Transect), data = arc_meth_wide)
check_model(methanotroph_anova)
anova(methanotroph_anova, type = 2)
emmeans(methanotroph_anova, list(pairwise ~Plot), adjust = "tukey")
methanotroph_emmeans <- emmeans(methanotroph_anova, ~Plot)
methanotroph_emmeans <- as.data.frame(methanotroph_emmeans)
methanotroph_emmeans$Group <- "Methylotrophy"

#figure
#combine emmeans datasets
methane_emmeans <- rbind(methanogen_emmeans, methanotroph_emmeans)
methane_emmeans$Group <- factor(methane_emmeans$Group, levels = c("Methanogenesis", "Methylotrophy"))

meth_emmeans_plot <- ggplot(methane_emmeans, aes(fill = fct_rev(Plot), x = Group)) +
  geom_bar(stat = "identity", position = "dodge", color = "black", aes(x = Group, y = emmean)) + 
  geom_errorbar(aes(ymin = upper.CL, ymax = emmean), 
                position = position_dodge(width = 0.9), width = 0.2, colour = "black") +
  theme_classic() +
  scale_fill_manual(values = rev(plot_colors)) +
    xlab("") + ylab("Relative Abundance") + 
  theme(axis.text = element_text(size = 15, color = "black")) +
  theme(axis.title.x = element_text(size = 15), legend.title = element_text(size =15),
        legend.text = element_text(size =15), axis.title.y = element_blank()) +
  coord_flip() +
  guides(fill = guide_legend(reverse = TRUE, title = "Plot")) + theme(legend.position = "none")

tiff("Figures/Plot3H.tiff", width = 4, height = 4, units = 'in', res = 300)
meth_emmeans_plot
dev.off()
```

```{r, eval = FALSE}
#mean and sd instead
meth_sum <- function_data %>% dplyr::select(Transect, Plot, Depth, methanogenesis_arc, methylotrophy_arc) %>% pivot_longer(!c(Transect, Plot, Depth), names_to = "Group", values_to = "Abundance")

meth_sum <- meth_sum %>% group_by(Plot, Group) %>% summarise(n = n(), mean_Abundance = mean(Abundance, na.rm = TRUE), sd_Abundance = sd(Abundance, na.rm = TRUE)) %>%
  mutate(se = sd_Abundance/sqrt(n))
meth_sum$Group <- factor(meth_sum$Group)
meth_sum$Plot <- factor(meth_sum$Plot, levels = c("30", "75", "250", "900", "1900"))


methane_function_plot <- ggplot(meth_sum, aes(x = fct_rev(Group), weight = mean_Abundance, ymin = mean_Abundance-se, ymax = mean_Abundance+se, fill = fct_rev(Plot))) +
  geom_bar(stat = "identity", position = "dodge", aes(y = mean_Abundance)) + 
  geom_errorbar(aes(ymin = mean_Abundance-se, ymax = mean_Abundance+se), position = position_dodge(width = 0.9), width = 0.2, colour = "black") +
  theme_classic() +
  scale_fill_manual(values = rev(plot_colors)) +
    xlab("Functional Group") + ylab("Relative Abundance") + 
  theme(axis.text = element_text(size = 10, color = "black")) +
  theme(axis.title.y = element_text(size = 10, vjust = 2)) +
  theme(axis.title.x = element_text(size = 10)) + coord_flip() +
  guides(fill = guide_legend(reverse = TRUE, title = "Plot"))
```

#### Saprotrophs

```{r}
sapro_func_all <- fun_func_merge_all %>% dplyr::select(-Ectomycorrhizal)

#### REPRESENT W HORIZONTAL BAR GRAPHS
sapro_func_all <- sapro_func_all %>% dplyr::rename(Dung = `Dung saprotroph`, Litter = `Litter saprotroph`, `Nectar/Tap/Pollen` = `Nectar/tap/pollen saprotroph`, Total = saprotroph_fun_total, Soil = `Soil saprotroph`, Unspecified = `Unspecified saprotroph`, Wood = `Wood saprotroph`)
sapro_func_all_top <- sapro_func_all %>% filter(Depth == "10-50")

##Unspecified saprotroph (30,75,250 are sig higher than 900/1900)
hist(sapro_func_all$Unspecified)
unspecified_anova <- lmer((Unspecified) ~ Plot+Depth + (1|Transect), data = sapro_func_all)
check_model(unspecified_anova)
anova(unspecified_anova, type = 2)
emmeans(unspecified_anova, list(pairwise ~Plot), adjust = "tukey")
unspecified_emmeans <- emmeans(unspecified_anova, ~Plot, type = "response")
unspecified_emmeans <- as.data.frame(unspecified_emmeans)
unspecified_emmeans$Group <- "Unspecified"

##Soil saprotroph
hist((sapro_func_all$Soil))
soil_anova <- glmmTMB((Soil) ~ Plot+Depth + (1|Transect), data = sapro_func_all, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = soil_anova))
drop1(soil_anova, test = "Chisq")
emmeans(soil_anova, list(pairwise ~Plot), adjust = "tukey")
soil_emmeans <- emmeans(soil_anova, ~Plot, type = "response")
soil_emmeans <- as.data.frame(soil_emmeans)
soil_emmeans$Group <- "Soil"

##Wood saprotroph (higher for pristine)
hist((sapro_func_all$Wood))
wood_anova <- glmmTMB((Wood) ~ Plot+Depth + (1|Transect), data = sapro_func_all, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = wood_anova))
drop1(wood_anova, test = "Chisq")
emmeans(wood_anova, list(pairwise ~Plot), adjust = "tukey")
wood_emmeans <- emmeans(wood_anova, ~Plot, type = "response")
wood_emmeans <- as.data.frame(wood_emmeans)
wood_emmeans$Group <- "Wood"

##Nectar/tap/pollen saprotroph
hist((sapro_func_all$`Nectar/Tap/Pollen`))
nectar_anova <- glmmTMB((`Nectar/Tap/Pollen`) ~ Plot+Depth + (1|Transect), data = sapro_func_all, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = nectar_anova))
drop1(nectar_anova, test = "Chisq")
emmeans(nectar_anova, list(pairwise ~Plot), adjust = "tukey")
nectar_emmeans <- emmeans(nectar_anova, ~Plot, type = "response")
nectar_emmeans <- as.data.frame(nectar_emmeans)
nectar_emmeans$Group <- "Nectar/Tap/Pollen"

##Litter saprotroph (higher for 900 plot specifically against degraded)
hist((sapro_func_all$Litter))
litter_anova <- glmmTMB((Litter) ~ Plot+Depth + (1|Transect), data = sapro_func_all, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = litter_anova))
drop1(litter_anova, test = "Chisq")
emmeans(litter_anova, list(pairwise ~Plot), adjust = "tukey")
litter_emmeans <- emmeans(litter_anova, ~Plot, type = "response")
litter_emmeans <- as.data.frame(litter_emmeans)
litter_emmeans$Group <- "Litter"

##Dung saprotroph
hist(sapro_func_all$Dung)
dung_anova <- glmmTMB((Dung) ~ Plot+Depth + (1|Transect), data = sapro_func_all, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = dung_anova))
drop1(dung_anova, test = "Chisq")
emmeans(dung_anova, list(pairwise ~Plot), adjust = "tukey")
dung_emmeans <- emmeans(dung_anova, ~Plot, type = "response")
dung_emmeans <- as.data.frame(dung_emmeans)
dung_emmeans$Group <- "Dung"

#total
hist(sapro_func_all$Total)
total_anova <- lmer((Total) ~ Plot+Depth + (1|Transect), data = sapro_func_all)
check_model(total_anova)
anova(total_anova, type = 2)
emmeans(total_anova, list(pairwise ~Plot), adjust = "tukey")
total_emmeans <- emmeans(total_anova, ~Plot, type = "response")
total_emmeans <- as.data.frame(total_emmeans)
total_emmeans$Group <- "Total"

#figure
#combine emmeans datasets
sapro_emmeans <- rbind(dung_emmeans, soil_emmeans, litter_emmeans, nectar_emmeans, unspecified_emmeans, wood_emmeans, total_emmeans)
sapro_emmeans$Group <- factor(sapro_emmeans$Group, levels = c("Total", "Unspecified", "Wood", "Soil", "Litter", "Nectar/Tap/Pollen", "Dung"))
#sapro_colors <- c("#4E79A7FF", "#9898A8FF", "#9D7660FF","#000000FF", "#59A14FFF", "#E0A858FF", "#3898B0FF", "#F8E898FF")

sapro_emmeans_plot <- ggplot(sapro_emmeans, aes(fill = fct_rev(Plot), x = Group)) +
  geom_bar(stat = "identity", position = "dodge", color = "black", aes(x = Group, y = response)) + 
  geom_errorbar(aes(ymin = upper.CL, ymax = response), 
                position = position_dodge(width = 0.9), width = 0.2, colour = "black") +
  theme_classic() +
  scale_fill_manual(values = rev(plot_colors)) +
    xlab("") + ylab("Relative Abundance") + 
  theme(axis.text = element_text(size = 15, color = "black")) +
  theme(axis.title.x = element_text(size = 15), legend.title = element_text(size =15),
        legend.text = element_text(size =15), axis.title.y = element_blank()) +
  coord_flip() +
  guides(fill = guide_legend(reverse = TRUE, title = "Plot"))  + theme(legend.position = "none")

#for top layer only

#dung
dung_anova <- lmer((Dung) ~ Plot + (1|Transect), data = sapro_func_all_top)
check_model(dung_anova)
anova(dung_anova, type = 2)
emmeans(dung_anova, list(pairwise ~Plot), adjust = "tukey")
dung_emmeans <- emmeans(dung_anova, ~Plot)
dung_emmeans <- as.data.frame(dung_emmeans)
dung_emmeans$Group <- "Dung"

#soil
soil_anova <- lmer((Soil) ~ Plot + (1|Transect), data = sapro_func_all_top)
check_model(soil_anova)
anova(soil_anova, type = 2)
emmeans(soil_anova, list(pairwise ~Plot), adjust = "tukey")
soil_emmeans <- emmeans(soil_anova, ~Plot)
soil_emmeans <- as.data.frame(soil_emmeans)
soil_emmeans$Group <- "Soil"

#litter
litter_anova <- lmer((Litter) ~ Plot + (1|Transect), data = sapro_func_all_top)
check_model(litter_anova)
anova(litter_anova, type = 2)
emmeans(litter_anova, list(pairwise ~Plot), adjust = "tukey")
litter_emmeans <- emmeans(litter_anova, ~Plot)
litter_emmeans <- as.data.frame(litter_emmeans)
litter_emmeans$Group <- "Litter"

#nectar
nectar_anova <- lmer((`Nectar/Tap/Pollen`) ~ Plot + (1|Transect), data = sapro_func_all_top)
check_model(nectar_anova)
anova(nectar_anova, type = 2)
emmeans(nectar_anova, list(pairwise ~Plot), adjust = "tukey")
nectar_emmeans <- emmeans(nectar_anova, ~Plot)
nectar_emmeans <- as.data.frame(nectar_emmeans)
nectar_emmeans$Group <- "Nectar/Tap/Pollen"

#unspecified
unspecified_anova <- lmer((Unspecified) ~ Plot + (1|Transect), data = sapro_func_all_top)
check_model(unspecified_anova)
anova(unspecified_anova, type = 2)
emmeans(unspecified_anova, list(pairwise ~Plot), adjust = "tukey")
unspecified_emmeans <- emmeans(unspecified_anova, ~Plot)
unspecified_emmeans <- as.data.frame(unspecified_emmeans)
unspecified_emmeans$Group <- "Unspecified"

#wood
wood_anova <- lmer((Wood) ~ Plot + (1|Transect), data = sapro_func_all_top)
check_model(wood_anova)
anova(wood_anova, type = 2)
emmeans(wood_anova, list(pairwise ~Plot), adjust = "tukey")
wood_emmeans <- emmeans(wood_anova, ~Plot)
wood_emmeans <- as.data.frame(wood_emmeans)
wood_emmeans$Group <- "Wood"

#total
total_anova <- lmer((Total) ~ Plot + (1|Transect), data = sapro_func_all_top)
check_model(total_anova)
anova(total_anova, type = 2)
emmeans(total_anova, list(pairwise ~Plot), adjust = "tukey")
total_emmeans <- emmeans(total_anova, ~Plot)
total_emmeans <- as.data.frame(total_emmeans)
total_emmeans$Group <- "Total"

#figure
#combine emmeans datasets
sapro_emmeans_top <- rbind(dung_emmeans, soil_emmeans, litter_emmeans, nectar_emmeans, unspecified_emmeans, wood_emmeans, total_emmeans)
sapro_emmeans_top$Group <- factor(sapro_emmeans_top$Group, levels = c("Total", "Unspecified", "Wood", "Soil", "Litter", "Nectar/Tap/Pollen", "Dung"))
#sapro_colors <- c("#4E79A7FF", "#9898A8FF", "#9D7660FF","#000000FF", "#59A14FFF", "#E0A858FF", "#3898B0FF", "#F8E898FF")

sapro_emmeans_top_plot <- ggplot(sapro_emmeans_top, aes(fill = fct_rev(Plot), x = Group)) +
  geom_bar(stat = "identity", position = "dodge", color = "black", aes(x = Group, y = emmean)) + 
  geom_errorbar(aes(ymin = upper.CL, ymax = emmean), 
                position = position_dodge(width = 0.9), width = 0.2, colour = "black") +
  theme_classic() +
  scale_fill_manual(values = rev(plot_colors)) +
    xlab("Functional Group") + ylab("Relative Abundance") + 
  theme(axis.text = element_text(size = 10, color = "black")) +
  theme(axis.title.y = element_text(size = 10, vjust = 2)) +
  theme(axis.title.x = element_text(size = 10)) + coord_flip() +
  guides(fill = guide_legend(reverse = TRUE, title = "Plot"))

sapro_panel <- sapro_emmeans_plot | sapro_emmeans_top_plot

#figure 3
specific_function_panel <- nitrogen_emmeans_plot | meth_emmeans_plot | sapro_emmeans_plot


tiff("Figures/Plot3I.tiff", width = 4, height = 4, units = 'in', res = 300)
sapro_emmeans_plot
dev.off()
```

```{r, eval = FALSE}
#mean and sd instead
sapro_sum <- sapro_sum %>% group_by(Plot, Group) %>% summarise(n = n(), mean_Abundance = mean(Abundance, na.rm = TRUE), sd_Abundance = sd(Abundance, na.rm = TRUE)) %>%
  mutate(se = sd_Abundance/sqrt(n))
sapro_sum$Group <- factor(sapro_sum$Group)
sapro_sum$Plot <- factor(sapro_sum$Plot, levels = c("30", "75", "250", "900", "1900"))
sapro_sum$Group <- factor(sapro_sum$Group, levels = c("Litter", "Dung", "Nectar/Tap", "Pollen", "Wood", "Soil", "Unspecified", "Total"))

sapro_func_plot <- ggplot(sapro_sum, aes(x = fct_rev(Group), weight = mean_Abundance, ymin = mean_Abundance-se, ymax = mean_Abundance+se, fill = fct_rev(Plot))) +
  geom_bar(stat = "identity", position = "dodge", aes(y = mean_Abundance)) + 
  geom_errorbar(aes(ymin = mean_Abundance-se, ymax = mean_Abundance+se), position = position_dodge(width = 0.9), width = 0.2, colour = "black") +
  theme_classic() +
  scale_fill_manual(values = rev(plot_colors)) +
    xlab("Functional Group") + ylab("Relative Abundance") + 
  theme(axis.text = element_text(size = 10, color = "black")) +
  theme(axis.title.y = element_text(size = 10, vjust = 2)) +
  theme(axis.title.x = element_text(size = 10)) + coord_flip() +
  guides(fill = guide_legend(reverse = TRUE, title = "Plot"))
```

#### fungi phyla

```{r}
fun_ecm <- fun_major_merge %>% dplyr::select(-Fungi_Observed, -Fungi_Shannon, -saprotroph_fun_total) %>% dplyr::rename(EcM = Ectomycorrhizal)

#ascomycota
hist(fun_ecm$Ascomycota)
asco_anova <- lmer((Ascomycota) ~ Plot+Depth + (1|Transect), data = fun_ecm)
check_model(asco_anova)
anova(asco_anova, type = 2)
asco_emmeans <- emmeans(asco_anova, ~Depth)
asco_emmeans <- as.data.frame(asco_emmeans)
asco_emmeans$Phylum <- "Ascomycota"

#basidiomycota
hist(fun_ecm$Basidiomycota)
basidio_anova <- glmmTMB((Basidiomycota) ~ Plot+Depth + (1|Transect), data = fun_ecm, family = beta_family(link = "logit"), ziformula = ~1)
plot(simulateResiduals(fittedModel = basidio_anova))
drop1(basidio_anova, test = "Chisq")
emmeans(basidio_anova, list(pairwise ~Depth), adjust = "tukey")
basidio_emmeans <- emmeans(basidio_anova, ~Depth, type = "response")
basidio_emmeans <- as.data.frame(basidio_emmeans)
basidio_emmeans$Group <- "Basidiomycota"

fun_phyla_emmeans <- rbind(asco_emmeans, basidio_emmeans)
fun_phyla_emmeans$Phylum <- factor(fun_phyla_emmeans$Phylum, levels = c("Ascomycota", "Basidiomycota"))
fun_phyla_emmeans$Depth <- as.numeric(fun_phyla_emmeans$Depth)

fun_ecm <- fun_ecm %>% pivot_longer(!c(Transect, Plot, Depth), names_to = "Group", values_to = "Abundance")
fun_ecm$Depth <- as.numeric(fun_ecm$Depth)

fun_ecm_sum <- fun_ecm %>% group_by(Depth, Group) %>% summarise(n = n(), mean_Abundance = mean(Abundance, na.rm = TRUE), 
                                                 sd_Abundance = sd(Abundance, na.rm = TRUE)) %>%
  mutate(se = sd_Abundance/sqrt(n))
fun_ecm_sum$Group <- factor(fun_ecm_sum$Group, levels = c("Ascomycota", "Basidiomycota"))
fun_ecm_sum$Depth <- factor(fun_ecm_sum$Depth, levels = c("1", "2", "3", "4", "5"))


ggplot(fun_ecm_sum, aes(x = fct_rev(Group), weight = mean_Abundance, ymin = mean_Abundance-se, ymax = mean_Abundance+se, fill = fct_rev(Depth))) +
  geom_bar(stat = "identity", position = "dodge", aes(y = mean_Abundance)) + 
  geom_errorbar(aes(ymin = mean_Abundance-se, ymax = mean_Abundance+se), position = position_dodge(width = 0.9), width = 0.2, colour = "black") +
  theme_classic() +
  scale_fill_manual(values = rev(plot_colors)) +
    xlab("Phylum") + ylab("Relative Abundance") + 
  theme(axis.text = element_text(size = 10, color = "black")) +
  theme(axis.title.y = element_text(size = 10, vjust = 2)) +
  theme(axis.title.x = element_text(size = 10)) + coord_flip() +
  guides(fill = guide_legend(reverse = TRUE, title = "Depth"))

fun_phyla_plot <- ggplot(fun_phyla_emmeans, aes(x = Depth, y = emmean, color = Phylum)) + theme_classic() + geom_line() + geom_point(size = 5) +
  geom_jitter(aes(x = Depth, y = Abundance, color = Group), 
            alpha = 0.5, width = 0.15, data = fun_ecm) + 
  ylab("Relative Abundance") + scale_color_manual(values = taxa_colors)
```

#### EcM

```{r}
#EcM anova
ecm_data <- function_data %>% select(Transect, Plot, Depth, Ectomycorrhizal) %>% drop_na()

ecm_anova <- lmer((Ectomycorrhizal) ~ Plot+Depth + (1|Transect), data = ecm_data)
check_model(ecm_anova)
anova(ecm_anova, type = 2)
emmeans(ecm_anova, list(pairwise ~Plot*Depth), adjust = "tukey")
ecm_emmeans <- emmeans(ecm_anova, ~Plot*Depth)
ecm_emmeans <- as.data.frame(ecm_emmeans)

## PLOT x DEPTH
ECM_plot <- ggplot() + geom_point(aes(x = Plot, y = emmean, color = Depth), 
                                       position = position_dodge(width = 0.5), size = 5, data = ecm_emmeans) +
  geom_errorbar(aes(x = Plot, ymin = lower.CL, ymax = upper.CL, color = Depth), width = 0.2, size = 1, 
                position = position_dodge(width = 0.5), data = ecm_emmeans) +
  geom_jitter(aes(x = Plot, y = Ectomycorrhizal, color = Depth), width = 0.25, alpha = 0.5, data = ecm_data) + 
  labs(y = "EcM Rel. Abund.", x = "Plot") + scale_color_manual(values = depth_colors) + theme_classic()
```

### Rarefaction curve - bacteria

```{r, eval = FALSE}
bact_ASV_num <- bact_ASV[,sapply(bact_ASV, is.numeric)]
str(bact_ASV_num)
names(bact_ASV_num) = gsub(pattern = "TE16S", replacement = "", x = names(bact_ASV_num))

#split datasets into depth specific
depth1 <- bact_ASV_num %>% dplyr::select(ends_with("A"))
depth2 <- bact_ASV_num %>% dplyr::select(ends_with("B"))
depth3 <- bact_ASV_num %>% dplyr::select(ends_with("C"))
depth4 <- bact_ASV_num %>% dplyr::select(ends_with("D"))
#convert to matrix
depth1_mat <- as.matrix(depth1)
depth2_mat <- as.matrix(depth2)
depth3_mat <- as.matrix(depth3)
depth4_mat <- as.matrix(depth4)
#combine to form final matrix
bact_depths_mat <- list(A = depth1_mat, B = depth2_mat, C = depth3_mat, D = depth4_mat)
bact_depths_mat <- na.omit(bact_depths_mat)

#run curve
bact_abund <- iNEXT(bact_depths_mat, datatype = "abundance") #takes a while
bact_rarefaction_depth <- ggiNEXT(bact_abund, se = TRUE) + theme_classic2() + scale_color_manual(values = depth_colors) + theme(legend.position = "none")

tiff("Figures/PlotS7A.tiff", width = 4, height = 4, units = 'in', res = 300)
bact_rarefaction_depth
dev.off()

#split datasets into plot specific
bact_ASV_num_plot <- bact_ASV_num
#reorder naming so last letter is for plot
names(bact_ASV_num_plot) <- sapply(names(bact_ASV_num_plot), function(x) paste0(substr(x, nchar(x), nchar(x)), 
                                 substr(x, 1, nchar(x) - 1)))
plot30 <- bact_ASV_num_plot %>% dplyr::select(ends_with("A"))
plot75 <- bact_ASV_num_plot %>% dplyr::select(ends_with("B"))
plot250 <- bact_ASV_num_plot %>% dplyr::select(ends_with("C"))
plot900 <- bact_ASV_num_plot %>% dplyr::select(ends_with("D"))
plot1900 <- bact_ASV_num_plot %>% dplyr::select(ends_with("E"))

#convert to matrix
plot30_mat <- as.matrix(plot30)
plot75_mat <- as.matrix(plot75)
plot250_mat <- as.matrix(plot250)
plot900_mat <- as.matrix(plot900)
plot1900_mat <- as.matrix(plot1900)
#combine to form final matrix
bact_plots_mat <- list(A = plot30_mat, B = plot75_mat, C = plot250_mat, D = plot900_mat, E = plot1900_mat)
bact_plots_mat <- na.omit(bact_plots_mat)

#run curve
bact_abund_plot <- iNEXT(bact_plots_mat, datatype = "abundance") #takes a while
bact_rarefaction_plot <- ggiNEXT(bact_abund_plot, se = TRUE) + theme_classic2() + scale_color_manual(values = plot_colors) + theme(legend.position = "none")

tiff("Figures/PlotS7D.tiff", width = 4, height = 4, units = 'in', res = 300)
bact_rarefaction_plot
dev.off()
```
I 
### Rarefaction curve - fungi

```{r, eval = FALSE}
fun_ASV_num <- fun_ASV[,sapply(fun_ASV, is.numeric)]
str(fun_ASV_num)
names(fun_ASV_num) = gsub(pattern = "TEITS", replacement = "", x = names(fun_ASV_num))

#split datasets into depth specific
depth1 <- fun_ASV_num %>% dplyr::select(ends_with("A"))
depth2 <- fun_ASV_num %>% dplyr::select(ends_with("B"))
depth3 <- fun_ASV_num %>% dplyr::select(ends_with("C"))
depth4 <- fun_ASV_num %>% dplyr::select(ends_with("D"))
#convert to matrix
depth1_mat <- as.matrix(depth1)
depth2_mat <- as.matrix(depth2)
depth3_mat <- as.matrix(depth3)
depth4_mat <- as.matrix(depth4)
#combine to form final matrix
fun_depths_mat <- list(A = depth1_mat, B = depth2_mat, C = depth3_mat, D = depth4_mat)
fun_depths_mat <- na.omit(fun_depths_mat)

#run curve
fun_abund <- iNEXT(fun_depths_mat, datatype = "abundance") #takes a while
fun_rarefaction_depth <- ggiNEXT(fun_abund, se = TRUE) + theme_classic2() + scale_color_manual(values = depth_colors) + theme(legend.position = "none")

tiff("Figures/PlotS7C.tiff", width = 4, height = 4, units = 'in', res = 300)
fun_rarefaction_depth
dev.off()

#split datasets into plot specific
fun_ASV_num_plot <- fun_ASV_num
#reorder naming so last letter is for plot
names(fun_ASV_num_plot) <- sapply(names(fun_ASV_num_plot), function(x) paste0(substr(x, nchar(x), nchar(x)), 
                                 substr(x, 1, nchar(x) - 1)))
plot30 <- fun_ASV_num_plot %>% dplyr::select(ends_with("A"))
plot75 <- fun_ASV_num_plot %>% dplyr::select(ends_with("B"))
plot250 <- fun_ASV_num_plot %>% dplyr::select(ends_with("C"))
plot900 <- fun_ASV_num_plot %>% dplyr::select(ends_with("D"))
plot1900 <- fun_ASV_num_plot %>% dplyr::select(ends_with("E"))

#convert to matrix
plot30_mat <- as.matrix(plot30)
plot75_mat <- as.matrix(plot75)
plot250_mat <- as.matrix(plot250)
plot900_mat <- as.matrix(plot900)
plot1900_mat <- as.matrix(plot1900)
#combine to form final matrix
fun_plots_mat <- list(A = plot30_mat, B = plot75_mat, C = plot250_mat, D = plot900_mat, E = plot1900_mat)
fun_plots_mat <- na.omit(fun_plots_mat)

#run curve
fun_abund_plot <- iNEXT(fun_plots_mat, datatype = "abundance") #takes a while
fun_rarefaction_plot <- ggiNEXT(fun_abund_plot, se = TRUE) + theme_classic2() + scale_color_manual(values = plot_colors) + theme(legend.position = "none")

tiff("Figures/PlotS7F.tiff", width = 4, height = 4, units = 'in', res = 300)
fun_rarefaction_plot
dev.off()
```

### Rarefaction curve - archaea

```{r, eval = FALSE}
arc_ASV_num <- arc_ASV[,sapply(arc_ASV, is.numeric)]
str(arc_ASV_num)
names(arc_ASV_num) = gsub(pattern = "ArcTE16S", replacement = "", x = names(arc_ASV_num))

#split datasets into depth specific
depth1 <- arc_ASV_num %>% dplyr::select(ends_with("A"))
depth2 <- arc_ASV_num %>% dplyr::select(ends_with("B"))
depth3 <- arc_ASV_num %>% dplyr::select(ends_with("C"))
depth4 <- arc_ASV_num %>% dplyr::select(ends_with("D"))
#convert to matrix
depth1_mat <- as.matrix(depth1)
depth2_mat <- as.matrix(depth2)
depth3_mat <- as.matrix(depth3)
depth4_mat <- as.matrix(depth4)
#combine to form final matrix
arc_depths_mat <- list(A = depth1_mat, B = depth2_mat, C = depth3_mat, D = depth4_mat)
arc_depths_mat <- na.omit(arc_depths_mat)

#run curve
arc_abund <- iNEXT(arc_depths_mat, datatype = "abundance") #takes a while
arc_rarefaction_depth <- ggiNEXT(arc_abund, se = TRUE) + theme_classic2() + scale_color_manual(values = depth_colors) + theme(legend.position = "none")

tiff("Figures/PlotS7B.tiff", width = 4, height = 4, units = 'in', res = 300)
arc_rarefaction_depth
dev.off()

#split datasets into plot specific
arc_ASV_num_plot <- arc_ASV_num
#reorder naming so last letter is for plot
names(arc_ASV_num_plot) <- sapply(names(arc_ASV_num_plot), function(x) paste0(substr(x, nchar(x), nchar(x)), 
                                 substr(x, 1, nchar(x) - 1)))
plot30 <- arc_ASV_num_plot %>% dplyr::select(ends_with("A"))
plot75 <- arc_ASV_num_plot %>% dplyr::select(ends_with("B"))
plot250 <- arc_ASV_num_plot %>% dplyr::select(ends_with("C"))
plot900 <- arc_ASV_num_plot %>% dplyr::select(ends_with("D"))
plot1900 <- arc_ASV_num_plot %>% dplyr::select(ends_with("E"))

#convert to matrix
plot30_mat <- as.matrix(plot30)
plot75_mat <- as.matrix(plot75)
plot250_mat <- as.matrix(plot250)
plot900_mat <- as.matrix(plot900)
plot1900_mat <- as.matrix(plot1900)
#combine to form final matrix
arc_plots_mat <- list(A = plot30_mat, B = plot75_mat, C = plot250_mat, D = plot900_mat, E = plot1900_mat)
arc_plots_mat <- na.omit(arc_plots_mat)

#run curve
arc_abund_plot <- iNEXT(arc_plots_mat, datatype = "abundance") #takes a while
arc_rarefaction_plot <- ggiNEXT(arc_abund_plot, se = TRUE) + theme_classic2() + scale_color_manual(values = plot_colors) + theme(legend.position = "none")

tiff("Figures/PlotS7E.tiff", width = 4, height = 4, units = 'in', res = 300)
arc_rarefaction_plot
dev.off()

#all rarefaction curves
rarefaction_panel <- (bact_rarefaction_depth | bact_rarefaction_plot) / (arc_rarefaction_depth | arc_rarefaction_plot) / (fun_rarefaction_depth |fun_rarefaction_plot)
```

### GDM (bact)

#### trees (removed)

```{r, eval = FALSE}
#include tree dissimilarity table
tree_counts <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Step1_Tree_family_counts.csv")
tree_counts <- tree_counts %>% filter(Plot == "3" | Plot == "4" | Plot == "5")
#calculate Bray Curtis dissimilarities
tree_abund <- tree_counts %>% dplyr::select(-Transect, -Plot) %>% as.data.frame()
bray_curtis <- vegdist(tree_abund) %>% as.matrix()
tree_dissim <-cbind(tree_counts[1:2], bray_curtis) 
write_csv(tree_dissim, "/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Step2_tree_dissim.csv")

tree_bray <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Final_tree_dissim_all.csv")
bact_tree_bray <- tree_bray %>% dplyr::select(-Sample_Arc, -Sample_ITS) %>% dplyr::rename(Sample = Sample_16S)
samplenames <- as.character(bact_tree_bray[[1]])
colnames(bact_tree_bray)[-1] <- samplenames
bact_tree_bray <- bact_tree_bray %>% dplyr::inner_join(bact_env_data_gdm) %>% select(1:101)

# Get the values from the first column (e.g., "Label")
labels <- unique(bact_tree_bray[[1]])
# Identify which column names (excluding first) match those labels
keep_cols <- colnames(bact_tree_bray) %in% labels
# Always keep the first column
keep_cols[1] <- TRUE
# Subset the dataframe
bact_tree_bray_filt <- bact_tree_bray[, keep_cols]

bact_gdmTab <- formatsitepair(bact_gdmTab_filt, 4, 
                               predData = bact_envTab_filt,
                               siteColumn = "Sample",
                               distPreds = list(as.matrix(bact_tree_bray)))
```

#### actual 

```{r, eval = FALSE}
library(gdm)

latlong <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Coordinates/Plot Coordinates - Clean.csv") %>% dplyr::select(-Sample)
latlong$Transect <- as.factor(latlong$Transect)
latlong$Plot <- as.factor(latlong$Plot)

basalarea <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/env_data_BA.csv") %>% dplyr::rename(Tree_BA = `Stand Basal Area (m2/ha)`)
basalarea$Transect <- as.factor(basalarea$Transect)
basalarea$Plot <- as.factor(basalarea$Plot)
bact_env_data <- bact_env_data %>% dplyr::full_join(basalarea)

str(bact_env_data)
bact_tax_gdm <- bact_phyloseq_filt.df %>% dplyr::select(OTU, Sample, Abundance, Transect, Plot, Depth)
#bact_tax_gdm <- bact_tax_gdm %>% uncount(weights = Abundance)
bact_env_data_gdm <- bact_env_data %>% select(1:17,33)
bact_env_data_gdm <- bact_env_data_gdm %>% na.omit()
bact_gdm <- bact_tax_gdm %>% dplyr::inner_join(bact_env_data_gdm) %>% dplyr::inner_join(latlong)
bact_gdm$Plot <- as.numeric(bact_gdm$Plot)
bact_gdm$Depth <- as.numeric(bact_gdm$Depth)

plotreplacements <- c("1" = "A", 
                  "2" = "B", 
                  "3" = "C", 
                  "4" = "D",
                  "5" = "E",
                  "A" = "30", 
                  "B" = "75", 
                  "C" = "250", 
                  "D" = "900",
                  "E" = "1900")
depthreplacements <- c("1" = "10", 
                  "2" = "50", 
                  "3" = "100", 
                  "4" = "150")


bact_gdm <- bact_gdm %>% 
  mutate(Plot = str_replace_all(Plot, plotreplacements)) %>%
  mutate(Depth = str_replace_all(Depth, depthreplacements))

bact_gdm$Plot <- as.numeric(bact_gdm$Plot)
bact_gdm$Depth <- as.numeric(bact_gdm$Depth)
summary(bact_gdm)

bact_sppTab <- bact_gdm[, c("OTU", "Sample", "Abundance", "Plot", "Depth", "Latitude", "Longitude")]
bact_envTab <- bact_gdm[, c(2, 5:ncol(bact_gdm))]

bact_gdmTab <- formatsitepair(bioData=bact_sppTab, 
                         bioFormat=2, #x-y spp list
                         abundance = TRUE,
                         XColumn="Longitude", 
                         YColumn="Latitude",
                         abundColumn = "Abundance",
                         sppColumn="OTU", 
                         siteColumn="Sample", 
                         predData=bact_envTab)

summary(bact_gdmTab)
histogram(bact_gdmTab$distance) #check distribution of biological dissimilatiry (if all near 1, inflating dissimilarity due to sampling)

bact_gdm.1 <- gdm(data=bact_gdmTab, geo=TRUE)
summary(bact_gdm.1)

#Samples:  3741
#Geographical distance used in model fitting?  TRUE
#NULL Deviance:  489.074
#GDM Deviance:  89.553
#Percent Deviance Explained:  81.689
#Intercept:  0.347

length(bact_gdm.1$predictors) # get ideal of number of panels
#> 17
plot(bact_gdm.1, plot.layout=c(1,1))
#how close the blue points follow line provides info on predictive capacity of the model

#individual variables -- bigger sum of coefficients = greatest influence on predicted dissimilarity
bact_gdm.1.splineDat <- isplineExtract(bact_gdm.1)
str(bact_gdm.1.splineDat)
plot(bact_gdm.1.splineDat$x[,"P"], 
     bact_gdm.1.splineDat$y[,"P"], 
     lwd=3,
     type="l", 
     xlab="Soil P", 
     ylab="Partial ecological distance")

bact_gdm.1.pred <- predict(object=bact_gdm.1, data=bact_gdmTab)

head(bact_gdm.1.pred)
#> 0.3193725 0.3911295 0.4563422 0.4545445 0.4893251 0.4925674

plot(bact_gdmTab$distance, 
     bact_gdm.1.pred, 
     xlab="Observed dissimilarity", 
     ylab="Predicted dissimilarity", 
     xlim=c(0,1), 
     ylim=c(0,1), 
     pch=20, 
     col=rgb(0,0,1,0.5))
lines(c(-1,2), c(-1,2))

#testing statistical significance with backwards selection
gdm.varImp(bact_gdmTab, geo = FALSE, splines = NULL, knots = NULL, predSelect = TRUE)

#filter for significant variables
bact_envTab_filt <- bact_gdm[, c(2, 5:ncol(bact_gdm))] %>% dplyr::select(Sample, Plot, Depth, pH, P, Fe, Tree_BA, Latitude, Longitude)

bact_gdmTab_filt <- formatsitepair(bioData=bact_sppTab, 
                         bioFormat=2, #x-y spp list
                         abundance = TRUE,
                         XColumn="Longitude", 
                         YColumn="Latitude",
                         abundColumn = "Abundance",
                         sppColumn="OTU", 
                         siteColumn="Sample", 
                         predData=bact_envTab_filt)

summary(bact_gdmTab_filt)
histogram(bact_gdmTab_filt$distance) #check distribution of biological dissimilatiry (if all near 1, inflating dissimilarity due to sampling)

bact_gdm.1_filt <- gdm(data=bact_gdmTab_filt, geo=TRUE)
summary(bact_gdm.1_filt)

#Samples:  3741
#Geographical distance used in model fitting?  TRUE
#NULL Deviance:  489.074
#GDM Deviance:  99.477
#Percent Deviance Explained:  79.66
#Intercept:  0.397

length(bact_gdm.1_filt$predictors) # get ideal of number of panels
#> 7
plot(bact_gdm.1_filt, plot.layout=c(3,3))
#how close the blue points follow line provides info on predictive capacity of the model

#individual variables -- bigger sum of coefficients = greatest influence on predicted dissimilarity
par(mfrow=c(1,1), mar = c(5,5,2,2))
#dev.off()
bact_gdm.1.splineDat_filt <- isplineExtract(bact_gdm.1_filt)
str(bact_gdm.1.splineDat_filt)

tiff("Figures/Plot2Gi.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(bact_gdm.1.splineDat_filt$x[,"Tree_BA"], 
     bact_gdm.1.splineDat_filt$y[,"Tree_BA"], 
     lwd=3,
     type="l",
     cex.axis = 1.25,
     cex.lab = 1.5,
     ylim = c(0,0.9),
     xlab="Tree basal area (m^2/ha)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2Gii.tiff", width = 6, height = 4, units = 'in', res = 300)
plot((bact_gdm.1.splineDat_filt$x[,"Plot"]), 
     (bact_gdm.1.splineDat_filt$y[,"Plot"]), 
     lwd=3,
     type="l", 
     cex.axis = 1.25,
     cex.lab = 1.5,
     ylim = c(0,0.9),
     xlab="Log-scale Distance from disturbance (m)", 
     ylab="Partial ecological distance", 
     log = "x")
dev.off()

tiff("Figures/Plot2G3.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(bact_gdm.1.splineDat_filt$x[,"pH"], 
     bact_gdm.1.splineDat_filt$y[,"pH"], 
     lwd=3,
     type="l", 
     cex.axis = 1.25,
     cex.lab = 1.5,
     ylim = c(0,0.9),
     xlab="pH", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2G4.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(bact_gdm.1.splineDat_filt$x[,"Fe"], 
     bact_gdm.1.splineDat_filt$y[,"Fe"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,0.9),
     xlab="Soil Fe (mg/kg soil)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2G5.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(bact_gdm.1.splineDat_filt$x[,"P"], 
     bact_gdm.1.splineDat_filt$y[,"P"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,0.9),
     xlab="Soil P (mg/kg soil)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2G6.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(bact_gdm.1.splineDat_filt$x[,"Geographic"], 
     bact_gdm.1.splineDat_filt$y[,"Geographic"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,0.9),
     xlab="Geographic distance(°)", 
     ylab="Partial ecological distance")
dev.off()

bact_gdm.1.pred_filt <- predict(object=bact_gdm.1_filt, data=bact_gdmTab_filt)

head(bact_gdm.1.pred_filt)
#> 0.3635171 0.4439472 0.4845109 0.4817477 0.5134057 0.5318944

plot(bact_gdmTab_filt$distance, 
     bact_gdm.1.pred_filt, 
     xlab="Observed dissimilarity", 
     ylab="Predicted dissimilarity", 
     xlim=c(0,1), 
     ylim=c(0,1), 
     pch=20, 
     col=rgb(0,0,1,0.5))
lines(c(-1,2), c(-1,2))
```

### GDM (Archaea) 

```{r}
arc_env_data <- arc_env_data %>% dplyr::full_join(basalarea)

str(arc_env_data)
arc_tax_gdm <- arc_phyloseq_filt.df %>% dplyr::select(OTU, Sample, Abundance, Transect, Plot, Depth)
#arc_tax_gdm <- arc_tax_gdm %>% uncount(weights = Abundance)
arc_env_data_gdm <- arc_env_data %>% select(1:17,33)
arc_env_data_gdm <- arc_env_data_gdm %>% na.omit()
arc_gdm <- arc_tax_gdm %>% dplyr::inner_join(arc_env_data_gdm) %>% dplyr::inner_join(latlong)
arc_gdm$Plot <- as.numeric(arc_gdm$Plot)
arc_gdm$Depth <- as.numeric(arc_gdm$Depth)

arc_gdm <- arc_gdm %>% 
  mutate(Plot = str_replace_all(Plot, plotreplacements)) %>%
  mutate(Depth = str_replace_all(Depth, depthreplacements))

arc_gdm$Plot <- as.numeric(arc_gdm$Plot)
arc_gdm$Depth <- as.numeric(arc_gdm$Depth)
summary(arc_gdm)

arc_sppTab <- arc_gdm[, c("OTU", "Sample", "Abundance", "Plot", "Depth", "Latitude", "Longitude")]
arc_envTab <- arc_gdm[, c(2, 5:ncol(arc_gdm))]

arc_gdmTab <- formatsitepair(bioData=arc_sppTab, 
                         bioFormat=2, #x-y spp list
                         abundance = TRUE,
                         XColumn="Longitude", 
                         YColumn="Latitude",
                         abundColumn = "Abundance",
                         sppColumn="OTU", 
                         siteColumn="Sample", 
                         predData=arc_envTab)

summary(arc_gdmTab)
histogram(arc_gdmTab$distance) #check distribution of biological dissimilarity (if all near 1, inflating dissimilarity due to sampling)

arc_gdm.1 <- gdm(data=arc_gdmTab, geo=TRUE)
summary(arc_gdm.1)

#Samples:  2145
#Geographical distance used in model fitting?  TRUE
#NULL Deviance:  387.99
#GDM Deviance:  273.407
#Percent Deviance Explained:  29.532
#Intercept:  0.75

length(arc_gdm.1$predictors) # get ideal of number of panels
#> 17
plot(arc_gdm.1, plot.layout=c(4,4))
#how close the blue points follow line provides info on predictive capacity of the model

#individual variables -- bigger sum of coefficients = greatest influence on predicted dissimilarity
arc_gdm.1.splineDat <- isplineExtract(arc_gdm.1)
str(arc_gdm.1.splineDat)
plot(arc_gdm.1.splineDat$x[,"P"], 
     arc_gdm.1.splineDat$y[,"P"], 
     lwd=3,
     type="l", 
     xlab="Soil P", 
     ylab="Partial ecological distance")

arc_gdm.1.pred <- predict(object=arc_gdm.1, data=arc_gdmTab)

head(arc_gdm.1.pred)
#> 0.5810583 0.6649518 0.6830465 0.7362530 0.7588555 0.8143464

plot(arc_gdmTab$distance, 
     arc_gdm.1.pred, 
     xlab="Observed dissimilarity", 
     ylab="Predicted dissimilarity", 
     xlim=c(0,1), 
     ylim=c(0,1), 
     pch=20, 
     col=rgb(0,0,1,0.5))
lines(c(-1,2), c(-1,2))

#testing statistical significance with backwards selection
gdm.varImp(arc_gdmTab, geo = FALSE, splines = NULL, knots = NULL, predSelect = TRUE)

#filter for significant variables
arc_envTab_filt <- arc_gdm[, c(2, 5:ncol(arc_gdm))] %>% dplyr::select(Sample, Plot, Depth, pH, Ca, Fe, Tree_BA, Latitude, Longitude)

arc_gdmTab_filt <- formatsitepair(bioData=arc_sppTab, 
                         bioFormat=2, #x-y spp list
                         abundance = TRUE,
                         XColumn="Longitude", 
                         YColumn="Latitude",
                         abundColumn = "Abundance",
                         sppColumn="OTU", 
                         siteColumn="Sample", 
                         predData=arc_envTab_filt)

summary(arc_gdmTab_filt)
histogram(arc_gdmTab_filt$distance) #check distribution of biological dissimilatiry (if all near 1, inflating dissimilarity due to sampling)

arc_gdm.1_filt <- gdm(data=arc_gdmTab_filt, geo=TRUE)
summary(arc_gdm.1_filt)

#Samples:  3741
#Geographical distance used in model fitting?  TRUE
#NULL Deviance:  387.99
#GDM Deviance:  290.694
#Percent Deviance Explained:  25.077
#Intercept:  0.85

length(arc_gdm.1_filt$predictors) # get ideal of number of panels
#> 7
plot(arc_gdm.1_filt, plot.layout=c(3,3))
#how close the blue points follow line provides info on predictive capacity of the model

#individual variables -- bigger sum of coefficients = greatest influence on predicted dissimilarity
par(mfrow=c(1,1), mar = c(5,5,2,2))
arc_gdm.1.splineDat_filt <- isplineExtract(arc_gdm.1_filt)
str(arc_gdm.1.splineDat_filt)

tiff("Figures/Plot2H1.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(arc_gdm.1.splineDat_filt$x[,"Fe"], 
     arc_gdm.1.splineDat_filt$y[,"Fe"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,0.9),
     xlab="Soil Fe (mg/kg soil)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2H2.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(arc_gdm.1.splineDat_filt$x[,"Ca"], 
     arc_gdm.1.splineDat_filt$y[,"Ca"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,0.9),
     xlab="Soil Ca (mg/kg soil)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2H3.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(arc_gdm.1.splineDat_filt$x[,"Geographic"], 
     arc_gdm.1.splineDat_filt$y[,"Geographic"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,0.9),
     xlab="Geographic distance(°)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2H4.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(arc_gdm.1.splineDat_filt$x[,"Tree_BA"], 
     arc_gdm.1.splineDat_filt$y[,"Tree_BA"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,0.9),
     xlab="Tree basal area (m^2/ha)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2H5.tiff", width = 6, height = 4, units = 'in', res = 300)
plot((arc_gdm.1.splineDat_filt$x[,"Plot"]), 
     (arc_gdm.1.splineDat_filt$y[,"Plot"]), 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,0.9),
     xlab="Log-scale Distance from disturbance (m)", 
     ylab="Partial ecological distance", 
     log = "x")
dev.off()

tiff("Figures/Plot2H6.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(arc_gdm.1.splineDat_filt$x[,"pH"], 
     arc_gdm.1.splineDat_filt$y[,"pH"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,0.9),
     xlab="pH", 
     ylab="Partial ecological distance")
dev.off()

arc_gdm.1.pred_filt <- predict(object=arc_gdm.1_filt, data=arc_gdmTab_filt)

head(arc_gdm.1.pred_filt)
#> 0.5986171 0.6733333 0.6851608 0.7357696 0.7701597 0.8121601

plot(arc_gdmTab_filt$distance, 
     arc_gdm.1.pred_filt, 
     xlab="Observed dissimilarity", 
     ylab="Predicted dissimilarity", 
     xlim=c(0,1), 
     ylim=c(0,1), 
     pch=20, 
     col=rgb(0,0,1,0.5))
lines(c(-1,2), c(-1,2))
```

### GDM (Fungi) 

```{r}
fun_env_data <- fun_env_data %>% dplyr::full_join(basalarea)

str(fun_env_data)
fun_tax_gdm <- fun_phyloseq_filt.df %>% dplyr::select(OTU, Sample, Abundance, Transect, Plot, Depth)
#fun_tax_gdm <- fun_tax_gdm %>% uncount(weights = Abundance)
fun_env_data_gdm <- fun_env_data %>% select(1:17,33)
fun_env_data_gdm <- fun_env_data_gdm %>% na.omit()
fun_gdm <- fun_tax_gdm %>% dplyr::inner_join(fun_env_data_gdm) %>% dplyr::inner_join(latlong)
fun_gdm$Plot <- as.numeric(fun_gdm$Plot)
fun_gdm$Depth <- as.numeric(fun_gdm$Depth)

fun_gdm <- fun_gdm %>% 
  mutate(Plot = str_replace_all(Plot, plotreplacements)) %>%
  mutate(Depth = str_replace_all(Depth, depthreplacements))

fun_gdm$Plot <- as.numeric(fun_gdm$Plot)
fun_gdm$Depth <- as.numeric(fun_gdm$Depth)
summary(fun_gdm)

fun_sppTab <- fun_gdm[, c("OTU", "Sample", "Abundance", "Plot", "Depth", "Latitude", "Longitude")]
fun_envTab <- fun_gdm[, c(2, 5:ncol(fun_gdm))]

fun_gdmTab <- formatsitepair(bioData=fun_sppTab, 
                         bioFormat=2, #x-y spp list
                         abundance = TRUE,
                         XColumn="Longitude", 
                         YColumn="Latitude",
                         abundColumn = "Abundance",
                         sppColumn="OTU", 
                         siteColumn="Sample", 
                         predData=fun_envTab)

summary(fun_gdmTab)
histogram(fun_gdmTab$distance) #check distribution of biological dissimilarity (if all near 1, inflating dissimilarity due to sampling)

fun_gdm.1 <- gdm(data=fun_gdmTab, geo=TRUE)
summary(fun_gdm.1)

#Samples:  3655
#Geographical distance used in model fitting?  TRUE
#NULL Deviance:  542.069
#GDM Deviance:  213.156
#Percent Deviance Explained:  60.677
#Intercept:  0.795

length(fun_gdm.1$predictors) # get ideal of number of panels
#> 17
plot(fun_gdm.1, plot.layout=c(4,4))
#how close the blue points follow line provides info on predictive capacity of the model

#individual variables -- bigger sum of coefficients = greatest influence on predicted dissimilarity
fun_gdm.1.splineDat <- isplineExtract(fun_gdm.1)
str(fun_gdm.1.splineDat)
plot(fun_gdm.1.splineDat$x[,"P"], 
     fun_gdm.1.splineDat$y[,"P"], 
     lwd=3,
     type="l", 
     xlab="Soil P", 
     ylab="Partial ecological distance")

fun_gdm.1.pred <- predict(object=fun_gdm.1, data=fun_gdmTab)

head(fun_gdm.1.pred)
#> 0.6060471 0.6791542 0.6615581 0.6821011 0.6634309 0.7247219

plot(fun_gdmTab$distance, 
     fun_gdm.1.pred, 
     xlab="Observed dissimilarity", 
     ylab="Predicted dissimilarity", 
     xlim=c(0,1), 
     ylim=c(0,1), 
     pch=20, 
     col=rgb(0,0,1,0.5))
lines(c(-1,2), c(-1,2))

#testing statistical significance with backwards selection
gdm.varImp(fun_gdmTab, geo = FALSE, splines = NULL, knots = NULL, predSelect = TRUE)

#filter for significant variables
fun_envTab_filt <- fun_gdm[, c(2, 5:ncol(fun_gdm))] %>% dplyr::select(Sample, Plot, Depth, pH, Ca, P, Tree_BA, Latitude, Longitude)

fun_gdmTab_filt <- formatsitepair(bioData=fun_sppTab, 
                         bioFormat=2, #x-y spp list
                         abundance = TRUE,
                         XColumn="Longitude", 
                         YColumn="Latitude",
                         abundColumn = "Abundance",
                         sppColumn="OTU", 
                         siteColumn="Sample", 
                         predData=fun_envTab_filt)

summary(fun_gdmTab_filt)
histogram(fun_gdmTab_filt$distance) #check distribution of biological dissimilatiry (if all near 1, inflating dissimilarity due to sampling)

fun_gdm.1_filt <- gdm(data=fun_gdmTab_filt, geo=TRUE)
summary(fun_gdm.1_filt)

#Samples:  3655
#Geographical distance used in model fitting?  TRUE
#NULL Deviance:  542.069
#GDM Deviance:  240.781
#Percent Deviance Explained:  55.581
#Intercept:  1.089

length(fun_gdm.1_filt$predictors) # get ideal of number of panels
#> 7
plot(fun_gdm.1_filt, plot.layout=c(3,3))
#how close the blue points follow line provides info on predictive capacity of the model

#individual variables -- bigger sum of coefficients = greatest influence on predicted dissimilarity
par(mfrow = c(1,1), mar = c(5,5,2,2))
fun_gdm.1.splineDat_filt <- isplineExtract(fun_gdm.1_filt)
str(fun_gdm.1.splineDat_filt)

tiff("Figures/Plot2I1.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(fun_gdm.1.splineDat_filt$x[,"Tree_BA"], 
     fun_gdm.1.splineDat_filt$y[,"Tree_BA"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     ylim = c(0,4),
     type="l", 
     xlab="Tree Basal Area (m^2/ha)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2I2.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(fun_gdm.1.splineDat_filt$x[,"pH"], 
     fun_gdm.1.splineDat_filt$y[,"pH"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,4),
     xlab="pH", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2I3.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(fun_gdm.1.splineDat_filt$x[,"Ca"], 
     fun_gdm.1.splineDat_filt$y[,"Ca"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,4),
     xlab="Soil Ca (mg/kg soil)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2I4.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(fun_gdm.1.splineDat_filt$x[,"P"], 
     fun_gdm.1.splineDat_filt$y[,"P"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,4),
     xlab="Soil P (mg/kg soil)", 
     ylab="Partial ecological distance")
dev.off()

tiff("Figures/Plot2I5.tiff", width = 6, height = 4, units = 'in', res = 300)
plot((fun_gdm.1.splineDat_filt$x[,"Plot"]), 
     (fun_gdm.1.splineDat_filt$y[,"Plot"]), 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,4),
     xlab="Log-scale Distance from disturbance (m)", 
     ylab="Partial ecological distance", 
     log = "x")
dev.off()

tiff("Figures/Plot2I6.tiff", width = 6, height = 4, units = 'in', res = 300)
plot(fun_gdm.1.splineDat_filt$x[,"Depth"], 
     fun_gdm.1.splineDat_filt$y[,"Depth"], 
     lwd=3,
     cex.axis = 1.25,
     cex.lab = 1.5,
     type="l", 
     ylim = c(0,4),
     xlab="Soil Depth (cm)", 
     ylab="Partial ecological distance")
dev.off()

fun_gdm.1.pred_filt <- predict(object=fun_gdm.1_filt, data=fun_gdmTab_filt)

head(fun_gdm.1.pred_filt)
#> 0.6989254 0.7779328 0.7393854 0.7499415 0.7415189 0.8205603

plot(fun_gdmTab_filt$distance, 
     fun_gdm.1.pred_filt, 
     xlab="Observed dissimilarity", 
     ylab="Predicted dissimilarity", 
     xlim=c(0,1), 
     ylim=c(0,1), 
     pch=20, 
     col=rgb(0,0,1,0.5))
lines(c(-1,2), c(-1,2))
```

## WGCNA (compiled)

### import normalised data from DESeq

```{r, eval = FALSE}
#pkgs <- c("preprocessCore", "impute", "AnnotationDbi", "GO.db")
#for (pkg in pkgs) {
  #if (!requireNamespace(pkg, quietly = TRUE))
   # BiocManager::install(pkg)
#}

library(WGCNA)
library(magrittr)      # provides the %>% operator
library(igraph)

bact_expr_normalized_df <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/expr_normalized_df.csv") %>% mutate(
    Sample = gsub("TE16S","", Sample)) %>% mutate(OTU = gsub("ASV", "bact_ASV", OTU))
arc_expr_normalized_df <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/expr_normalized_df.csv") %>% mutate(
    Sample = gsub("ArcTE16S","", Sample)) %>% mutate(OTU = gsub("ASV", "arc_ASV", OTU))
fun_expr_normalized_df <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_ITS/Post_tax/expr_normalized_df.csv") %>% mutate(
    Sample = gsub("TEITS","", Sample)) %>% mutate(OTU = gsub("ASV", "fun_ASV", OTU))

all_expr_normalized_df <- rbind(bact_expr_normalized_df, arc_expr_normalized_df, fun_expr_normalized_df)
```

### filter for specific plot/depth

```{r, eval = FALSE}
#working_nodes <- all_expr_normalized_df
#filtered for bottom two layers
patterns = c("..C", "..D")
working_nodes <- filter(all_expr_normalized_df, grepl(paste(patterns, collapse="|"), all_expr_normalized_df$Sample))
#filter for nonforested
patterns = c(".A.", ".B.")
working_nodes <- filter(working_nodes, grepl(paste(patterns, collapse="|"), working_nodes$Sample))
write_csv(working_nodes, "/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/working_nodes.csv")

#pivot wider
working_nodes <- working_nodes %>% select(-Transect, - Plot, -Depth) %>% pivot_wider(names_from = Sample, values_from = value) %>% column_to_rownames("OTU")
working_nodes[is.na(working_nodes)] <- -0.1638618
working_nodes <- as.matrix(working_nodes)
```

### Transpose data/prep data for WGCNA

```{r, eval = FALSE}
input_mat = t(working_nodes)

allowWGCNAThreads()          # allow multi-threading (optional)

# Choose a set of soft-thresholding powers
powers = c(c(1:10), seq(from = 12, to = 20, by = 2))

# Call the network topology analysis function
sft = pickSoftThreshold(
  input_mat,             # <= Input data
  #blockSize = 30,
  powerVector = powers,
  verbose = 5
  )

par(mfrow = c(1,2));
cex1 = 0.9;

plot(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     xlab = "Soft Threshold (power)",
     ylab = "Scale Free Topology Model Fit, signed R^2",
     main = paste("Scale independence")
)
text(sft$fitIndices[, 1],
     -sign(sft$fitIndices[, 3]) * sft$fitIndices[, 2],
     labels = powers, cex = cex1, col = "red"
)
abline(h = 0.90, col = "red")
plot(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     xlab = "Soft Threshold (power)",
     ylab = "Mean Connectivity",
     type = "n",
     main = paste("Mean connectivity")
)
text(sft$fitIndices[, 1],
     sft$fitIndices[, 5],
     labels = powers,
     cex = cex1, col = "red")
```

Pick a soft threshold power near the curve of the plot, so here we could pick 7, 8 or 9. We’ll pick 8 but feel free to experiment with other powers to see how it affects your results. 

Now we can create the network using the blockwiseModules command. The blockwiseModule may take a while to run, since it is constructing the TOM (topological overlap matrix) and several other steps. While it runs, take a look at the blockwiseModule documentation (link to vignette) for more information on the parameters. How might you change the parameters to get more or less modules?

```{r, eval = FALSE}
picked_power = 6
#top two layers
#for forested = 6
#for nonforested = 6

#bottom two layers
#for forested = 6
#for nonforested = 6

temp_cor <- cor       
cor <- WGCNA::cor         # Force it to use WGCNA cor function (fix a namespace conflict issue)
netwk <- blockwiseModules(input_mat,                # <= input here

                          # == Adjacency Function ==
                          power = picked_power,                # <= power here
                          networkType = "signed",

                          # == Tree and Block Options ==
                          deepSplit = 2,
                          pamRespectsDendro = F,
                          # detectCutHeight = 0.75,
                          minModuleSize = 10,
                          maxBlockSize = 4000,

                          # == Module Adjustments ==
                          reassignThreshold = 0,
                          mergeCutHeight = 0.25,

                          # == TOM == Archive the run results in TOM file (saves time)
                          saveTOMs = T,
                          saveTOMFileBase = "ER",

                          # == Output Options
                          numericLabels = T,
                          verbose = 3)
cor <- temp_cor     # Return cor function to original namespace
```

take a look at modules

```{r, eval = FALSE}
# Convert labels to colors for plotting
mergedColors = labels2colors(netwk$colors)
# Plot the dendrogram and the module colors underneath
plotDendroAndColors(
  netwk$dendrograms[[1]],
  mergedColors[netwk$blockGenes[[1]]],
  "Module colors",
  dendroLabels = FALSE,
  hang = 0.03,
  addGuide = TRUE,
  guideHang = 0.05 )
```

### Relate Module (cluster) Assignments to Treatment Groups

```{r, eval = FALSE}
module_df <- data.frame(
  OTU = names(netwk$colors),
  colors = labels2colors(netwk$colors)
)

write_csv(module_df,
            file = "/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/gene_modules.csv")
```

We have written out a tab delimited file listing the genes and their modules. However, we need to figure out which modules are associated with each trait/treatment group. WGCNA will calcuate an Eigangene (hypothetical central gene) for each module, so it easier to determine if modules are associated with different treatments.

```{r, eval = FALSE}
# Get Module Eigengenes per cluster
MEs0 <- moduleEigengenes(input_mat, mergedColors)$eigengenes

# Reorder modules so similar modules are next to each other
MEs0 <- orderMEs(MEs0)
module_order = names(MEs0) %>% gsub("ME","", .)

# Add treatment names
MEs0$treatment = row.names(MEs0)

# tidy & plot data
mME = MEs0 %>%
  pivot_longer(-treatment) %>%
  mutate(
    name = gsub("ME", "", name),
    name = factor(name, levels = module_order)
  )

mME %>% ggplot(., aes(x=treatment, y=name, fill=value)) +
  geom_tile() +
  theme_bw() +
  scale_fill_gradient2(
    low = "blue",
    high = "red",
    mid = "white",
    midpoint = 0,
    limit = c(-1,1)) +
  theme(axis.text.x = element_text(angle=90)) +
  labs(title = "Module-trait Relationships", y = "Modules", fill="corr")
```

### Examine expression profiles

```{r, eval = FALSE}
# pick out a few modules of interest here
#modules_of_interest = c("grey", "brown", "turquoise", "yellow", "green", "blue")
modules_of_interest = c("grey", "turquoise", "blue")

# Pull out list of genes in that module
submod = module_df %>%
  subset(colors %in% modules_of_interest)

row.names(module_df) = module_df$OTU

subexpr = working_nodes[submod$OTU,]

submod_df = data.frame(subexpr) %>%
  mutate(
    OTU = row.names(.)
  ) %>%
  pivot_longer(-OTU) %>%
  mutate(
    module = module_df[OTU,]$colors
  )

submod_df %>% ggplot(., aes(x=name, y=value, group=OTU)) +
  geom_line(aes(color = module),
            alpha = 0.2) +
  theme_bw() +
  theme(
    axis.text.x = element_text(angle = 90)
  ) +
  facet_grid(rows = vars(module)) +
  labs(x = "treatment",
       y = "normalized expression")
```

### Generate p values

```{r, eval = FALSE}
genes_of_interest = module_df %>%
  subset(colors %in% modules_of_interest)

# Calculate expression data for genes of interest
expr_of_interest = working_nodes[genes_of_interest$OTU,]

# Check for zero variance genes
expr_of_interest_filtered <- expr_of_interest[apply(expr_of_interest, 1, sd) > 0, ]

#calculate p values
# Calculate correlation after filtering
correlation_matrix <- cor(t(expr_of_interest_filtered))

# View the correlation matrix
head(correlation_matrix)

# Number of observations (genes)
n_genes <- nrow(expr_of_interest_filtered)

# Calculate the p-values for each correlation coefficient
p_value_matrix <- corPvalueStudent(correlation_matrix, n_genes)

p_value_matrix <- data.frame(p_value_matrix) %>%
  mutate(
    gene1 = row.names(.)
  ) %>%
  pivot_longer(-gene1) %>%
  dplyr::rename(gene2 = name, pvalue = value) %>%
  unique() %>%
  subset(!(gene1==gene2))

# View the p-value matrix
head(p_value_matrix)
```

#### BH adjustment

```{r, eval = FALSE}
#sort by lowest to highest
p_value_matrix <- p_value_matrix %>% arrange(pvalue)

#create "rank"
p_value_matrix <- p_value_matrix %>% mutate(rank = row_number())
n_obs <- nrow(p_value_matrix)

#adjust p values
p_value_matrix$adj_pvalue <- p_value_matrix$pvalue * n_obs / p_value_matrix$rank

#clean table
p_value_matrix <- p_value_matrix %>% dplyr::select(gene1, gene2, adj_pvalue)
```

### Create edge list

```{r, eval = FALSE}
TOM = TOMsimilarityFromExpr(t(expr_of_interest_filtered),
                            power = picked_power)

row.names(TOM) = row.names(expr_of_interest_filtered)
colnames(TOM) = row.names(expr_of_interest_filtered)

edge_list = data.frame(TOM) %>%
  mutate(
    gene1 = row.names(.)
  ) %>%
  pivot_longer(-gene1) %>%
  dplyr::rename(gene2 = name, correlation = value) %>%
  unique() %>%
  subset(!(gene1==gene2)) %>%
  mutate(
    module1 = module_df[gene1,]$colors,
    module2 = module_df[gene2,]$colors
  ) %>% full_join(p_value_matrix)

#select p value cut off
edge_list <- edge_list %>% filter(adj_pvalue <= 0.001)

head(edge_list)
write_csv(edge_list, file = "/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/edgelist.csv")
```

## prep for Gephi

### read in data

```{r, eval = FALSE}
# first import in abundance data

# Bacteria
bact_counts <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/bact_counts.csv") %>% mutate(OTU = gsub("ASV", "bact_ASV", OTU))
bact_summarized_nodes <- bact_counts %>% select(-Transect, -Plot, -Depth, -Sample) %>% group_by(OTU) %>% mutate(Abundance = sum(value)) %>% select(-value) %>% unique()

# Archaea
arc_counts <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/arc_counts.csv") %>% mutate(OTU = gsub("ASV", "arc_ASV", OTU))
arc_summarized_nodes <- arc_counts %>% select(-Transect, -Plot, -Depth, -Sample) %>% group_by(OTU) %>% mutate(Abundance = sum(value)) %>% select(-value) %>% unique()

# Fungi
fun_counts <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_ITS/Post_tax/fun_counts.csv") %>% mutate(OTU = gsub("ASV", "fun_ASV", OTU))
fun_summarized_nodes <- fun_counts %>% select(-Transect, -Plot, -Depth, -Sample) %>% group_by(OTU) %>% mutate(Abundance = sum(value)) %>% select(-value) %>% unique()

#combine into all summarised nodes
all_summarized_nodes <- rbind(bact_summarized_nodes, arc_summarized_nodes, fun_summarized_nodes)

#then combine with modules
modules <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/gene_modules.csv") %>% dplyr::rename(module = "colors")
all_summarized_nodes <- modules %>% right_join(all_summarized_nodes)

#then combine with taxa
#bacteria
bact_phyloseq_rare <- readRDS("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_16S/Post_tax/physeq_rarefied_16S")
bact_phyloseq_rare.df <- psmelt(bact_phyloseq_rare)
bact_phyloseq_rare.df <- bact_phyloseq_rare.df %>% subset(Kingdom == "Bacteria") #filter only bacteria
bact_taxa_table <-  bact_phyloseq_rare.df %>% select(OTU, Phylum, Class, Order, Family, Genus) %>% unique() %>% mutate(OTU = gsub("ASV", "bact_ASV", OTU))
#archaea
arc_phyloseq_rare <- readRDS("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_Archaea/Post_tax/Arch_physeq_rare.RDS")
arc_phyloseq_rare.df <- psmelt(arc_phyloseq_rare)
arc_phyloseq_rare.df <- arc_phyloseq_rare.df %>% subset(Kingdom == "Archaea") #filter only archaea
arc_taxa_table <-  arc_phyloseq_rare.df %>% select(OTU, Phylum, Class, Order, Family, Genus) %>% unique() %>% mutate(OTU = gsub("ASV", "arc_ASV", OTU))
#fungi
fun_phyloseq_rare <- readRDS("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/TE_ITS/Post_tax/physeq_rarefied_ITS.Rds")
fun_phyloseq_rare.df <- psmelt(fun_phyloseq_rare)
fun_taxa_table <-  fun_phyloseq_rare.df %>% select(OTU, Phylum, Class, Order, Family, Genus) %>% unique() %>% mutate(OTU = gsub("ASV", "fun_ASV", OTU))

#all taxa
all_taxa_table <- rbind(bact_taxa_table, arc_taxa_table, fun_taxa_table)

#final combination
all_summarized_nodes <- all_taxa_table %>% right_join(all_summarized_nodes)
```

```{r, eval = FALSE}
links <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/edgelist.csv") #edgelist from WGCNA

#remove direction
links$edge_id <- paste(links$gene1, links$gene2, sep = " ")
links$edge_id2 <- paste(links$gene2, links$gene1, sep = " ")

links <- links %>%
  mutate(pair = pmin(edge_id, edge_id2),  # Sort values in each row
         pair2 = pmax(edge_id, edge_id2)) %>%
  distinct(pair, pair2, .keep_all = TRUE) %>%  # Keep only distinct pairs
  select(-pair, -pair2, edge_id, edge_id2)

#filter links to only have links between prevalent nodes
OTU <- all_summarized_nodes$OTU
OTU <- as.data.frame(OTU)
OTU <- OTU %>% dplyr::rename(gene1 = OTU)
links <- links %>% inner_join(OTU)
OTU <- OTU %>% dplyr::rename(gene2 = gene1)
links <- links %>% inner_join(OTU)
```

### convert data to igraph object

```{r, eval = FALSE}
net <- graph_from_data_frame(d=links, vertices=all_summarized_nodes, directed=F) 

# We can access the nodes, edges, and their attributes:
#E(net)$weight <- E(net)$correlation *100
#E(net)$weight
E(net)$correlation
V(net)

#plot(net)
#gsize(net)
```

### convert stuff to gephi

```{r, eval = FALSE}
#devtools::install_github("RMHogervorst/gephi")
library(gephi)

gephi_write_edges(net, "/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Gephi/Edgelists/bottom_nonforested3075.csv")

edges <- read_csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Gephi/Edgelists/bottom_nonforested3075.csv") 
head(edges)

nodes <- edges %>% dplyr::rename(OTU = Source) %>% left_join(all_taxa_table) %>% dplyr::rename(Id = OTU) %>% remove_rownames() %>% dplyr::select(Id, Phylum, Class, Order, Family, Genus) %>% unique()
head(nodes)
write.csv(nodes, "/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Gephi/Nodes/bottom_nonforested3075.csv", row.names = FALSE)
```

## stats for networks

### proportion of kingdoms

```{r}
topnonforest <- read.csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Gephi/Exported_tables/top_open.csv") %>% separate(ID, c("Kingdom", "ASV"), extra = "merge")
topnonforest$Depth <- "Top"
topnonforest$Plot <- "Open"
topnonforest_sum <- topnonforest %>%
  count(Kingdom) %>%
  mutate(perc = n/sum(n)) %>%
  select(-n)
topnonforest_sum$Depth <- "Top"
topnonforest_sum$Plot <- "Open"
#topnonforest_sum <- topnonforest_sum %>% pivot_wider(values_from = perc, names_from = Kingdom)

topforest <- read.csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Gephi/Exported_tables/top_forest.csv") %>% separate(ID, c("Kingdom", "ASV"), extra = "merge")
topforest$Depth <- "Top"
topforest$Plot <- "Forested"
topforest_sum <- topforest %>%
  count(Kingdom) %>%
  mutate(perc = n/sum(n)) %>%
  select(-n)
topforest_sum$Depth <- "Top"
topforest_sum$Plot <- "Forested"
#topforest_sum <- topforest_sum %>% pivot_wider(values_from = perc, names_from = Kingdom)

bottomnonforest <- read.csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Gephi/Exported_tables/bottom_open.csv") %>% separate(ID, c("Kingdom", "ASV"), extra = "merge")
bottomnonforest$Depth <- "Bottom"
bottomnonforest$Plot <- "Open"
bottomnonforest_sum <- bottomnonforest %>%
  count(Kingdom) %>%
  mutate(perc = n/sum(n)) %>%
  select(-n)
bottomnonforest_sum$Depth <- "Bottom"
bottomnonforest_sum$Plot <- "Open"
#bottomnonforest_sum <- bottomnonforest_sum %>% pivot_wider(values_from = perc, names_from = Kingdom)

bottomforest <- read.csv("/Users/Ecolabs/Desktop/SCELSCE_Sequencing/TE_Degradation_Plots/Gephi/Exported_tables/bottom_forest.csv") %>% separate(ID, c("Kingdom", "ASV"), extra = "merge")
bottomforest$Depth <- "Bottom"
bottomforest$Plot <- "Forested"
bottomforest_sum <- bottomforest %>%
  count(Kingdom) %>%
  mutate(perc = n/sum(n)) %>%
  select(-n)
bottomforest_sum$Depth <- "Bottom"
bottomforest_sum$Plot <- "Forested"
#bottomforest_sum <- bottomforest_sum %>% pivot_wider(values_from = perc, names_from = Kingdom)

network_proportion <- rbind(topnonforest_sum, topforest_sum, bottomnonforest_sum, bottomforest_sum)
network_proportion$Plot <- factor(network_proportion$Plot, levels = c("Open", "Forested"))
network_proportion$Depth <- factor(network_proportion$Depth, levels = c("Top", "Bottom"))

network_prop <- ggplot(network_proportion, aes(fill=Kingdom, y=perc, x=Depth)) + geom_bar(position="fill", stat="identity") + theme_classic() + facet_grid(~Plot) + ylab("Relative Abundance") + scale_fill_manual(values = c("#FAB255", "#0F7BA2", "#43B284"))
```

### stats graphs

#### set up

```{r}
network_stats <- rbind(topnonforest, topforest, bottomnonforest, bottomforest)
network_stats$Plot <- factor(network_stats$Plot, levels = c("Open", "Forested"))
network_stats$Depth <- factor(network_stats$Depth, levels = c("Top", "Bottom"))
network_stats <- network_stats %>% filter(Kingdom != c("arc"))
network_stats$Kingdom <-gsub(pattern = "bact", replacement = "Bacteria", x = network_stats$Kingdom)
network_stats$Kingdom <-gsub(pattern = "fun", replacement = "Fungi", x = network_stats$Kingdom)

boxplot(network_stats$Betweenness.Centrality)
network_stats$Betweenness.Centrality[network_stats$Betweenness.Centrality > 200] <- NA
boxplot(network_stats$Degree)
boxplot(network_stats$Closeness.Centrality)

#degree
degree_plot <- ggplot() +
  geom_boxplot(aes(x = Depth, y = Degree, fill = Plot), width = 0.8, size = 1, alpha = 0.8,
              data = network_stats) + facet_wrap(~Kingdom) +
  labs(y = "Degree", x = "Depth") + scale_fill_manual(values = c("#0F7BA2", "#43B284")) + theme_classic() +
  theme(legend.position = "none")

#closeness centrality
closeness_plot <- ggplot() +
  geom_boxplot(aes(x = Depth, y = Closeness.Centrality, fill = Plot), width = 0.8, size = 1, alpha = 0.8,
             data = network_stats) + facet_wrap(~Kingdom) +
  labs(y = "Closeness Centrality", x = "Depth") + scale_fill_manual(values = c("#0F7BA2", "#43B284")) + theme_classic() + theme(legend.position = "none")

#betweenness centrality
betweenness_plot <- ggplot() +
  geom_boxplot(aes(x = Depth, y = Betweenness.Centrality, fill = Plot), width = 0.8, size = 1, alpha = 0.8, data = network_stats) + facet_wrap(~Kingdom) +
  labs(y = "Betweeness Centrality", x = "Depth") + scale_fill_manual(values = c("#0F7BA2", "#43B284")) + theme_classic()

network_panel <- (network_prop | degree_plot) / (closeness_plot | betweenness_plot)

##ANOVAs!!
#degree
degree_anova <- aov((Degree) ~ Plot*Kingdom + Depth, data = network_stats)
check_model(degree_anova)
anova(degree_anova)
emmeans(degree_anova, list(pairwise ~Plot*Kingdom), adjust = "tukey")

#closeness
closeness_anova <- aov((Closeness.Centrality) ~ Plot*Depth+Kingdom, data = network_stats)
check_model(closeness_anova)
anova(closeness_anova)
emmeans(closeness_anova, list(pairwise ~Plot*Depth+Kingdom), adjust = "tukey")

#betweenness
betweenness_anova <- aov((Betweenness.Centrality) ~ Plot*Depth*Kingdom, data = network_stats)
check_model(betweenness_anova)
anova(betweenness_anova)
emmeans(betweenness_anova, list(pairwise ~Plot*Depth*Kingdom), adjust = "tukey")
```

## TITAN2 (bact)

```{r}
library(TITAN2)


#abundance of major fungal families
bact_pseq_fam <- aggregate_rare(bact_pseq, level = "Family", detection = 1/100, prevalence = 2/100)
bact_fam_merge <- psmelt(bact_pseq_fam) #create dataframe from phyloseq 
bact_df_fam <- bact_fam_merge %>% dplyr::select(Sample, Family, Abundance, Plot, Depth)
unique(bact_df_fam$Family)
bact_df_fam <- bact_df_fam %>% group_by(Family) %>%
  pivot_wider(names_from = Family, values_from = Abundance) %>% dplyr::select(-Other)

#prep bact taxa data
bact_soil_data <- bact_env_data %>% dplyr::select(1:4,9,11,13,14,17,18:23,25)
bact_TITAN <- bact_df_fam %>% full_join(bact_soil_data)
bact_TITAN <- bact_TITAN %>% na.omit()
bact_TITAN_taxa <- bact_TITAN %>% dplyr::select(1,3:80) %>% column_to_rownames("Sample")


#prep bact taxa data
#data(glades.taxa) #example
bact_TITAN_taxa <- bact_major_merge %>% full_join(bact_env_data) %>% dplyr::select(c(20, 6:19),)
bact_TITAN_taxa[is.na(bact_TITAN_taxa)] <- 0

#prep environmental gradient
#data(glades.env) #example
bact_TITAN_env <- bact_env_data %>% dplyr::select(c(1, 3:17, 33),)
bact_TITAN_env <- bact_TITAN_env[complete.cases(bact_TITAN_env),] 
bact_TITAN_env$Plot <- as.numeric(bact_TITAN_env$Plot)
bact_TITAN_env$Depth <- as.numeric(bact_TITAN_env$Depth)



#change plot and depth to numeric gradients
plotreplacements <- c("1" = "A", "2" = "B", "3" = "C", "4" = "D", "5" = "E",
                  "A" = "30", "B" = "75", "C" = "250", "D" = "900", "E" = "1900")
depthreplacements <- c("1" = "10", "2" = "50", "3" = "100", "4" = "150")

bact_TITAN_env <- bact_TITAN_env %>% 
  mutate(Plot = str_replace_all(Plot, plotreplacements)) %>%
  mutate(Depth = str_replace_all(Depth, depthreplacements))

bact_TITAN_env$Plot <- as.numeric(bact_TITAN_env$Plot)
bact_TITAN_env$Depth <- as.numeric(bact_TITAN_env$Depth)

#filter out for only samples that are retained (taxa)
bact_TITAN_taxa <- bact_TITAN_taxa %>% dplyr::inner_join(bact_TITAN_env) %>% dplyr::select(c(1:15),) %>% column_to_rownames("Sample")

#pH
bact_TITAN_env_pH <- bact_TITAN_env %>% dplyr::select(Sample, pH) %>% remove_rownames() %>% column_to_rownames("Sample")
#run titan() function
bact_TITAN <- titan(bact_TITAN_env_pH, bact_TITAN_taxa)
```




### TITAN2 (fungi)

```{r}
#abundance of major fungal families
bact_pseq_fam <- aggregate_rare(bact_pseq, level = "Family", detection = 1/100, prevalence = 2/100)
bact_fam_merge <- psmelt(bact_pseq_fam) #create dataframe from phyloseq 
bact_df_fam <- bact_fam_merge %>% dplyr::select(Sample, Family, Abundance, FT)
unique(bact_df_fam$Family)
bact_df_fam <- bact_df_fam %>% group_by(Family) %>%
  pivot_wider(names_from = Family, values_from = Abundance) %>% dplyr::select(-Other)

#prep bact taxa data
bact_soil_data <- bact_env_data %>% dplyr::select(1:4,9,11,13,14,17,18:23,25)
bact_TITAN <- bact_df_fam %>% full_join(bact_soil_data)
bact_TITAN <- bact_TITAN %>% na.omit()
bact_TITAN_taxa <- fun_TITAN %>% dplyr::select(1,3:80) %>% column_to_rownames("Sample")

#prep environmental gradient
fun_TITAN_env <- fun_TITAN %>% dplyr::select(1,2, 81:94) %>% dplyr::select(-Plot)
fun_TITAN_env$FT <- as.numeric(fun_TITAN_env$FT)
fun_TITAN_env$Sample_1 <- gsub(pattern = "ITS", replacement = "", x = fun_TITAN_env$Sample)
soil_fertility <- microbe_env %>% dplyr::select(Sample, soil_fertility) %>% dplyr::rename(Sample_1 = Sample)
fun_TITAN_env <- fun_TITAN_env %>% inner_join(soil_fertility)



#soil fertility gradient
fun_TITAN_env_soil <- fun_TITAN_env %>% dplyr::select(Sample, soil_fertility) %>% remove_rownames() %>% column_to_rownames("Sample")
#run titan() function -- this takes a while
#fun_TITAN_soil <- titan(fun_TITAN_env_soil, fun_TITAN_taxa)

fun_TITAN_soil$sumz.cp
head(fun_TITAN_soil$sppmax)
summary(fun_TITAN_soil)

#visualise along gradient (all taxa) -- z+ is positive change and vice versa
#plot_sumz_density(fun_TITAN_soil, xlabel = expression(paste("Soil Fe (mg/kg soil)")))
##same thing as above but points instead of shaded
plot_sumz_density(fun_TITAN_soil, ribbon = FALSE, points = TRUE, xlabel = expression(paste("Soil Fertility Gradient")))
##same thing as above but for one side instead of both
#plot_sumz_density(fun_TITAN_soil, ribbon = TRUE, points = FALSE, sumz1 = FALSE, change_points = FALSE, xlabel = expression(paste("Soil Fertility Gradient")))
##same as above with a cumulative frequency
#plot_sumz(fun_TITAN_soil, filter = TRUE)

##pure+reliable indicator taxa 
plot_taxa_ridges(fun_TITAN_soil, axis.text.y = 8, linewidth = 1, vline_width = 1, n_ytaxa = 15, rel_heights = c(0.5,1), xlabel = expression(paste("Soil Fertility Gradient")))
##limiting the amount of taxa on y axis
#plot_taxa_ridges(fun_TITAN_soil, xlabel = expression(paste("Soil Fertility Gradient")), n_ytaxa = 50)
##this is the same again but the old function
#plot_taxa(fun_TITAN_soil, xlabel = "Soil Fertility Gradient")

##looking at taxon specific changes (for pure+reliable taxa)
plot_cps(fun_TITAN_soil) #generally want to scan the distributions
## to look at one specific taxa more up close
par(mfrow=c(1,1))
plot_cps(fun_TITAN_soil, taxaID = "Acidothermaceae", cp.trace = TRUE, xlabel = "Soil Fertility Gradient")
plot_cps(fun_TITAN_soil, taxa.dist = FALSE, xlabel = "Soil Fertility Gradient")





#Fe (basing this off GDM)
fun_TITAN_env_Fe <- fun_TITAN_env %>% dplyr::select(Sample, sFe) %>% remove_rownames() %>% column_to_rownames("Sample")
#run titan() function -- this takes a while
#fun_TITAN <- titan(fun_TITAN_env_Fe, fun_TITAN_taxa)

fun_TITAN$sumz.cp
head(fun_TITAN$sppmax)
summary(fun_TITAN)

#visualise along gradient (all taxa) -- z+ is positive change and vice versa
plot_sumz_density(fun_TITAN, xlabel = expression(paste("Soil Fe (mg/kg soil)")))
##same thing as above but points instead of shaded
plot_sumz_density(fun_TITAN, ribbon = FALSE, points = TRUE, , xlabel = expression(paste("Soil Fe (mg/kg soil)")))
##same thing as above but for one side instead of both
plot_sumz_density(fun_TITAN, ribbon = TRUE, points = FALSE, sumz1 = FALSE, change_points = FALSE, xlabel = expression(paste("Soil Fe (mg/kg soil)")))
##same as above with a cumulative frequency
#plot_sumz(fun_TITAN, filter = TRUE)

##pure+reliable indicator taxa 
plot_taxa_ridges(fun_TITAN, axis.text.y = 8, xlabel = expression(paste("Soil Fe (mg/kg soil)")))
# > There are 16 indicator taxa of 90 possible for plotting -- ONLY 15 taxa
##limiting the amount of taxa on y axis
#plot_taxa_ridges(fun_TITAN, xlabel = expression(paste("Soil Fe (mg/kg soil)")), n_ytaxa = 50)
##this is the same again but the old function
#plot_taxa(fun_TITAN, xlabel = "Soil Fe (mg/kg soil)")

##looking at taxon specific changes (for pure+reliable taxa)
plot_cps(fun_TITAN) #generally want to scan the distributions
## to look at one specific taxa more up close
par(mfrow=c(1,1))
plot_cps(fun_TITAN, taxaID = "Trichosporonaceae", cp.trace = TRUE, xlabel = "Soil Fe (mg/kg soil)")
plot_cps(fun_TITAN, taxa.dist = FALSE, xlabel = "Soil Fe (mg/kg soil)")



#Al (basing this off GDM)
fun_TITAN_env_Al <- fun_TITAN_env %>% dplyr::select(Sample, sAl) %>% remove_rownames() %>% column_to_rownames("Sample")
#run titan() function -- this takes a while
fun_TITAN_Al <- titan(fun_TITAN_env_Al, fun_TITAN_taxa)

fun_TITAN_Al$sumz.cp
head(fun_TITAN_Al$sppmax)
summary(fun_TITAN_Al)

#visualise along gradient (all taxa) -- z+ is positive change and vice versa
#plot_sumz_density(fun_TITAN_Al, xlabel = expression(paste("Soil Al (mg/kg soil)")))
##same thing as above but points instead of shaded
plot_sumz_density(fun_TITAN_Al, ribbon = FALSE, points = TRUE, xlabel = expression(paste("Soil Al (mg/kg soil)")))
##same thing as above but for one side instead of both
plot_sumz_density(fun_TITAN_Al, ribbon = TRUE, points = FALSE, sumz1 = FALSE, change_points = FALSE, xlabel = expression(paste("Soil Al (mg/kg soil)")))
##same as above with a cumulative frequency
#plot_sumz(fun_TITAN_Al, filter = TRUE)

##pure+reliable indicator taxa 
plot_taxa_ridges(fun_TITAN_Al, axis.text.y = 8, xlabel = expression(paste("Soil Al (mg/kg soil)")))
# > There are 16 indicator taxa of 90 possible for plotting -- ONLY 15 taxa
##limiting the amount of taxa on y axis
#plot_taxa_ridges(fun_TITAN_Al, xlabel = expression(paste("Soil Al (mg/kg soil)")), n_ytaxa = 50)
##this is the same again but the old function
#plot_taxa(fun_TITAN_Al, xlabel = "Soil Al (mg/kg soil)")

##looking at taxon specific changes (for pure+reliable taxa)
plot_cps(fun_TITAN_Al) #generally want to scan the distributions
## to look at one specific taxa more up close
par(mfrow=c(1,1))
plot_cps(fun_TITAN_Al, taxaID = "Burkholderiaceae", cp.trace = TRUE, xlabel = "Soil Al (mg/kg soil)")
plot_cps(fun_TITAN_Al, taxa.dist = FALSE, xlabel = "Soil Al (mg/kg soil)")
```

